{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "Include the intercept to the model because\n",
    "expression data, remove the s,d \n",
    "expression should be float\n",
    "beta in linear compared to beta in quad\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import math\n",
    "import numpy as np\n",
    "import os\n",
    "from collections import Counter\n",
    "import pandas as pd\n",
    "import random\n",
    "import shutil\n",
    "import statsmodels.api as sm\n",
    "import statsmodels.formula.api as smf\n",
    "from statsmodels.stats.anova import anova_lm\n",
    "import sys\n",
    "\n",
    "\n",
    "def Regression(data,norm=True, minsamples=120, linear=True):\n",
    "    \"\"\"\n",
    "    Perform non linear regression, return beta, p_beta, alpha, p_alpha, se_alpha, se_beta\n",
    "    data.columns has to have [allele1='x1', allele2='x2', gene_expression='expr'] not normalized\n",
    "    \"\"\"\n",
    "    if data.shape[0] <= minsamples: \n",
    "        return None, None, None, None, None, None\n",
    "    m = np.mean(list(locus_str[\"x1\"].astype(int))+list(locus_str[\"x2\"].astype(int))) \n",
    "    sd= math.sqrt(np.var(locus_str[\"x1\"].astype(int) + locus_str[\"x2\"].astype(int) ))\n",
    "    if norm:\n",
    "        data['x1'] = ZNorm(data['x1'].astype(int), sd, m)\n",
    "        data['x2'] = ZNorm(data['x2'].astype(int), sd, m)\n",
    "        E=data['expr'].astype(float)\n",
    "        data['expr'] = ZNorm(E, np.mean(E), math.sqrt(np.var(E)))\n",
    "        if data['x1'].isnull().all()  or data['x2'].isnull().all() or data['expr'].isnull().all(): \n",
    "            return None, None, None, None, None, None  \n",
    "    data['X']=data['x1']+data['x2']\n",
    "    data['X2']=data['x1']**2 + data['x2']**2\n",
    "    model = sm.ols(formula = 'expr ~ X + X2', data = data).fit()\n",
    "    #model = sm.OLS(Locus_detail[['expr']],Locus_detail[['X','X2']] , missing='drop').fit()\n",
    "    #print(data.head(3), '\\n..(1)..\\n',data.head(3))\n",
    "    mod_ols = sm.ols(formula = 'expr ~ X + X2', data = data).fit()\n",
    "    alpha = list(mod_ols.params)[1]\n",
    "    beta = list(mod_ols.params)[2]\n",
    "    alpha_p = list(mod_ols.pvalues)[1]\n",
    "    beta_p  = list(mod_ols.pvalues)[2]\n",
    "    alpha_se=list(mod_ols.bse)[1]\n",
    "    beta_se=list(mod_ols.bse)[2]\n",
    "    quad_rsq = mod_ols.rsquared\n",
    "    if linear: \n",
    "        #print(data.head(3), '\\n..(2)..\\n',data.head(3))\n",
    "        X=data['X'].astype(int).values\n",
    "        mod_ols1 = sm.OLS(data['expr'], X, missing='drop').fit()\n",
    "        pval = mod_ols1.pvalues[0]\n",
    "        slope = mod_ols1.params[0]\n",
    "        err = mod_ols1.bse[0]\n",
    "        lin_rsq = mod_ols1.rsquared\n",
    "        anova_results = anova_lm(mod_ols, mod_ols1)\n",
    "        anova_pval = anova_results[\"Pr(>F)\"].values[1]\n",
    "        delta_aic = mod_ols.aic - mod_ols1.aic\n",
    "        \n",
    "        return alpha, alpha_se, alpha_p, beta, beta_se, beta_p, slope, err, pval,quad_rsq, lin_rsq, delta_aic, anova_pval\n",
    "    else:\n",
    "        return alpha, alpha_se, alpha_p, beta, beta_se, beta_p\n",
    "    \n",
    "    \n",
    "\n",
    "\n",
    "def ZNorm(vals,m=None,sd=None):\n",
    "    if m is None:\n",
    "        m = np.mean(vals)\n",
    "        sd = math.sqrt(np.var(vals))\n",
    "    if sd == 0: return None\n",
    "    return [(item-m)/sd for item in vals]\n",
    "\n",
    "def LinearRegression(data, Y, norm=False, minsamples=0, alleles=False):\n",
    "    \"\"\"\n",
    "    Perform linear regression, return beta, beta_se, p\n",
    "    \"\"\"\n",
    "\n",
    "    if norm:\n",
    "        if alleles:\n",
    "            data['x1']=data['x1'].astype(float) ; data['x2']=data['x2'].astype(float)\n",
    "            data['x1+x2'] = data[['x1', 'x2']].sum(axis=1)\n",
    "            X = ZNorm(data['x1+x2'], None, None)\n",
    "        else:\n",
    "            X = ZNorm(data) \n",
    "            Y = ZNorm(Y, None, None)\n",
    "        if X is None or Y is None: return None, None, None\n",
    "        if np.var(X)==0: return None, None, None\n",
    "        if len(X) <= minsamples: return None, None, None\n",
    "    else:\n",
    "        X = data\n",
    "    mod_ols = sm.OLS(Y, X, missing='drop')\n",
    "    res_ols = mod_ols.fit()\n",
    "    #print(res_ols.pvalues)\n",
    "    pval = res_ols.pvalues[0]\n",
    "    #print 'P-value: ', pval \n",
    "    slope = res_ols.params[0]\n",
    "    err = res_ols.bse[0]\n",
    "    return res_ols, slope, err, pval\n",
    "\n",
    "def QuadraticRegression(data,norm=True, minsamples=120):\n",
    "    \"\"\"\n",
    "    Perform non linear regression, return beta, p_beta, alpha, p_alpha, se_alpha, se_beta\n",
    "    data.columns has to have [allele1='x1', allele2='x2', gene_expression='expr'] not normalized\n",
    "    \"\"\"\n",
    "    if data.shape[0] <= minsamples: \n",
    "        return None, None, None, None, None, None, None\n",
    "    m = np.mean(list(locus_str[\"x1\"].astype(int))+list(locus_str[\"x2\"].astype(int))) \n",
    "    sd= math.sqrt(np.var(locus_str[\"x1\"].astype(int) + locus_str[\"x2\"].astype(int) ))\n",
    "    if norm:\n",
    "        data.loc[:,'x1'] =ZNorm(data['x1'].astype(int), sd, m)\n",
    "        data.loc[:,'x2'] =ZNorm(data['x2'].astype(int), sd, m)       \n",
    "        data.loc[:,'expr'] =ZNorm(data['expr'].astype(int))\n",
    "        if data['x1'].isnull().all()  or data['x2'].isnull().all() or data['expr'].isnull().all(): \n",
    "            return None, None, None, None, None, None, None  \n",
    "        \n",
    "    data['X']=data['x1']+data['x2']\n",
    "    data['X2']=data['x1']**2 + data['x2']**2\n",
    "    \n",
    "    mod_ols = sm.OLS('expr ,X ,X2', data = data).fit()\n",
    "    #mod_ols = smf.ols(formula = 'expr ~ X + X2', data = data).fit()\n",
    "    print(mod_ols.summary())\n",
    "    alpha = list(mod_ols.params)[1]\n",
    "    beta = list(mod_ols.params)[2]\n",
    "    alpha_p = list(mod_ols.pvalues)[1]\n",
    "    beta_p  = list(mod_ols.pvalues)[2]\n",
    "    alpha_se=list(mod_ols.bse)[1]\n",
    "    beta_se=list(mod_ols.bse)[2]\n",
    "    return mod_ols, alpha, alpha_se, alpha_p, beta, beta_se, beta_p\n",
    "\n",
    "def Modelcompare(model1, model2):\n",
    "    \"\"\"\n",
    "    Performs ANOVA test to compares 2 models \n",
    "    Outputs rsquares , delta AIC (aic_model1- aic_model2) and pvalue\n",
    "    \"\"\"\n",
    "    rsq1 = model1.rsquared\n",
    "    rsq2 = model2.rsquared\n",
    "    anova_output = sm.stats.anova_lm(model1 , model2)\n",
    "    #print(anova_output, anova_output[\"Pr(>F)\"])\n",
    "    anova_pval = anova_output[\"Pr(>F)\"].values[1]\n",
    "    delta_aic = model1.aic - model2.aic  \n",
    "    return rsq1, rsq2, delta_aic, anova_pval\n",
    "\n",
    "STRGTFILE= '/storage/szfeupe/Runs/650GTEx_estr/Genotypes/Allele_Gentypes.table'\n",
    "EXPRESSION='/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/Adipose-Subcutaneous/Corr_Expr.csv'\n",
    "EXPRANNOT='/storage/resources/dbase/human/hg19/gencode_gene_annotations_hg19.csv'\n",
    "distfromgene = 100000\n",
    "CHROM='chr1'\n",
    "NORM=True"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Load STRs\n",
      "Load expression\n",
      "Load annotation\n",
      "There are (3051, 272) samples\n"
     ]
    }
   ],
   "source": [
    "# Load STR genotypes\n",
    "print(\"Load STRs\")\n",
    "strgt = pd.read_csv(STRGTFILE, sep=\"\\t\", low_memory=False)\n",
    "strgt = strgt[strgt[\"chrom\"] == CHROM]\n",
    "print(\"Load expression\")\n",
    "expr = pd.read_csv(EXPRESSION)\n",
    "print(\"Load annotation\")\n",
    "expr_annot = pd.read_csv(EXPRANNOT)\n",
    "expr_annot.index = expr_annot[\"probe.id\"].values\n",
    "expr_annot = expr_annot[expr_annot[\"gene.chr\"] == CHROM]\n",
    "expr_annot = expr_annot.loc[[item for item in expr.columns if item in expr_annot.index],:]\n",
    "# Restrict to STR samples\n",
    "str_samples = list(set(strgt.columns[2:].values))\n",
    "samples_to_remove = []\n",
    "for item in str_samples:\n",
    "    if item not in expr.index: samples_to_remove.append(item) #str_samples.remove(item)\n",
    "for item in samples_to_remove: str_samples.remove(item)\n",
    "expr = expr.loc[str_samples,:]\n",
    "strgt = strgt[['chrom','start']+str_samples]\n",
    "print(\"There are %s samples\"%str(strgt.shape))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " Getting data for ENSG00000158109.10\n",
      "ENSG00000158109.10 \t 3541566 \t 3546691 \t 3 STRs tested \n",
      "\n",
      "chr1\tENSG00000158109.10\t3501711\tNone\tNone\tNone\tNone\tNone\tNone\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib64/python3.4/site-packages/pandas/core/indexing.py:537: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[item] = s\n",
      "/usr/lib64/python3.4/site-packages/pandas/core/indexing.py:621: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[item_labels[indexer[info_axis]]] = value\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "unrecognized data structures: <class 'str'> / <class 'NoneType'>",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-8679a5b5576a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     30\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     31\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mLocus_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'expr'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'x1'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'x2'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 32\u001b[0;31m         \u001b[0mquad_model\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malpha\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malpha_se\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0malpha_pval\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbeta\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbeta_se\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbeta_pval\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mQuadraticRegression\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mnorm\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mNORM\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mminsamples\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m120\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     33\u001b[0m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mLocus_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'x1'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'x2'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     34\u001b[0m \u001b[0;31m#        lin_model, slope, err, pval = LinearRegression(data, locus_y[\"expr\"].values, norm=NORM, minsamples=MINSAMPLES, alleles=ALLELE)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-7-3ed9b6e199c5>\u001b[0m in \u001b[0;36mQuadraticRegression\u001b[0;34m(data, norm, minsamples)\u001b[0m\n\u001b[1;32m    113\u001b[0m     \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'X2'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'x1'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0;36m2\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'x2'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    114\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 115\u001b[0;31m     \u001b[0mmod_ols\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msm\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mOLS\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'expr ,X ,X2'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    116\u001b[0m     \u001b[0;31m#mod_ols = smf.ols(formula = 'expr ~ X + X2', data = data).fit()\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    117\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmod_ols\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msummary\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/regression/linear_model.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, endog, exog, missing, hasconst, **kwargs)\u001b[0m\n\u001b[1;32m    629\u001b[0m                  **kwargs):\n\u001b[1;32m    630\u001b[0m         super(OLS, self).__init__(endog, exog, missing=missing,\n\u001b[0;32m--> 631\u001b[0;31m                                   hasconst=hasconst, **kwargs)\n\u001b[0m\u001b[1;32m    632\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0;34m\"weights\"\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_init_keys\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    633\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_init_keys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mremove\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"weights\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/regression/linear_model.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, endog, exog, weights, missing, hasconst, **kwargs)\u001b[0m\n\u001b[1;32m    524\u001b[0m             \u001b[0mweights\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mweights\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msqueeze\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    525\u001b[0m         super(WLS, self).__init__(endog, exog, missing=missing,\n\u001b[0;32m--> 526\u001b[0;31m                                   weights=weights, hasconst=hasconst, **kwargs)\n\u001b[0m\u001b[1;32m    527\u001b[0m         \u001b[0mnobs\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexog\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    528\u001b[0m         \u001b[0mweights\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mweights\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/regression/linear_model.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, endog, exog, **kwargs)\u001b[0m\n\u001b[1;32m     93\u001b[0m     \"\"\"\n\u001b[1;32m     94\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 95\u001b[0;31m         \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mRegressionModel\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     96\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_data_attr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mextend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'pinv_wexog'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'wendog'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'wexog'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'weights'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     97\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/base/model.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, endog, exog, **kwargs)\u001b[0m\n\u001b[1;32m    210\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    211\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 212\u001b[0;31m         \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mLikelihoodModel\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    213\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0minitialize\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    214\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/base/model.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, endog, exog, **kwargs)\u001b[0m\n\u001b[1;32m     61\u001b[0m         \u001b[0mhasconst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpop\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'hasconst'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     62\u001b[0m         self.data = self._handle_data(endog, exog, missing, hasconst,\n\u001b[0;32m---> 63\u001b[0;31m                                       **kwargs)\n\u001b[0m\u001b[1;32m     64\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mk_constant\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mk_constant\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexog\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mexog\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/base/model.py\u001b[0m in \u001b[0;36m_handle_data\u001b[0;34m(self, endog, exog, missing, hasconst, **kwargs)\u001b[0m\n\u001b[1;32m     86\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     87\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_handle_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmissing\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhasconst\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 88\u001b[0;31m         \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhandle_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmissing\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mhasconst\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     89\u001b[0m         \u001b[0;31m# kwargs arrays could have changed, easier to just attach here\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     90\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mkey\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mkwargs\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/base/data.py\u001b[0m in \u001b[0;36mhandle_data\u001b[0;34m(endog, exog, missing, hasconst, **kwargs)\u001b[0m\n\u001b[1;32m    626\u001b[0m         \u001b[0mexog\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0masarray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mexog\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    627\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 628\u001b[0;31m     \u001b[0mklass\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhandle_data_class_factory\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mendog\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mexog\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    629\u001b[0m     return klass(endog, exog=exog, missing=missing, hasconst=hasconst,\n\u001b[1;32m    630\u001b[0m                  **kwargs)\n",
      "\u001b[0;32m/usr/lib64/python3.4/site-packages/statsmodels/base/data.py\u001b[0m in \u001b[0;36mhandle_data_class_factory\u001b[0;34m(endog, exog)\u001b[0m\n\u001b[1;32m    615\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    616\u001b[0m         raise ValueError('unrecognized data structures: %s / %s' %\n\u001b[0;32m--> 617\u001b[0;31m                          (type(endog), type(exog)))\n\u001b[0m\u001b[1;32m    618\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mklass\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    619\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: unrecognized data structures: <class 'str'> / <class 'NoneType'>"
     ]
    }
   ],
   "source": [
    "expr_annot = expr_annot.loc[expr_annot.index == 'ENSG00000158109.10']\n",
    "NORM=True; ALLELE=True\n",
    "MINSAMPLES=150\n",
    "#for i in range(150,expr_annot.shape[0],1):\n",
    "for i in range(0,1,1):\n",
    "    gene = expr_annot.index.values[i]\n",
    "    print(\" Getting data for %s\"%gene)\n",
    "    start = expr_annot[\"gene.start\"].values[i]\n",
    "    end = expr_annot[\"gene.stop\"].values[i]\n",
    "    cis_strs = strgt[(strgt[\"start\"] >= (start-distfromgene)) & (strgt[\"start\"] <= (end+distfromgene))]\n",
    "    print(gene, '\\t',start,'\\t',end,'\\t',\"%s STRs tested \\n\"%str(cis_strs.shape[0]))\n",
    "    y = pd.DataFrame({\"expr\":list(expr.loc[:, gene])})\n",
    "    y.index = str_samples\n",
    "    \n",
    "    for j in range(cis_strs.shape[0]):\n",
    "        locus_str = cis_strs.iloc[[j],:][str_samples].transpose()\n",
    "        locus_str.index = str_samples\n",
    "        locus_str.columns = [\"STR_%s\"%(cis_strs[\"start\"].values[j])]\n",
    "        test_str=locus_str.columns[0]\n",
    "        str_start = cis_strs[\"start\"].values[j]\n",
    "        locus_str['x1'] = locus_str[test_str].apply(lambda x: x.split(',')[0] )\n",
    "        locus_str['x2'] = locus_str[test_str].apply(lambda x: x.split(',')[1] )\n",
    "        samples_to_keep = [str_samples[k] for k in range(len(str_samples)) if (str(locus_str['x1'].values[k]) != \"NA\")and(str(locus_str['x2'].values[k]) != \"NA\")]   \n",
    "        locus_str = locus_str.loc[samples_to_keep,:]        \n",
    "        locus_y = y.loc[samples_to_keep,:]            \n",
    "        Locus_data = locus_str.join(locus_y)\n",
    "        locus_str = locus_str.join(locus_y)[['expr','x1','x2']]\n",
    "        #print(locus_str.head(3), '\\n...\\n',Locus_detail.head(3))\n",
    "        # Run regression\n",
    "\n",
    "        data = Locus_data[['expr','x1','x2']]\n",
    "        quad_model, alpha, alpha_se, alpha_pval, beta, beta_se, beta_pval=QuadraticRegression(data,norm=NORM, minsamples=120)\n",
    "        data = Locus_data[['x1','x2']]\n",
    "#        lin_model, slope, err, pval = LinearRegression(data, locus_y[\"expr\"].values, norm=NORM, minsamples=MINSAMPLES, alleles=ALLELE)\n",
    "#        quad_rsq, lin_rsq, delta_aic, anova_pval = Modelcompare(lin_model , quad_model)\n",
    "#        print(\"\\t\".join(map(str, [CHROM, gene, test_str, str_start,alpha, alpha_se, alpha_pval, beta, beta_se, beta_pval, slope, err, pval,quad_rsq, lin_rsq, delta_aic, anova_pval] ) )+\"\\n\")                \n",
    "        print(\"\\t\".join(map(str, [CHROM, gene, str_start,alpha, alpha_se, alpha_pval, beta, beta_se, beta_pval] ) )+\"\\n\")                \n",
    "        \n",
    "    break\n",
    "#Locus_detail "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib64/python3.4/site-packages/pandas/core/indexing.py:537: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy\n",
      "  self.obj[item] = s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(None, None, None, None, None, None, None)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data = Locus_data[['expr','x1','x2']]\n",
    "data\n",
    "QuadraticRegression(data,norm=NORM, minsamples=120)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene.chr</th>\n",
       "      <th>gene.start</th>\n",
       "      <th>gene.stop</th>\n",
       "      <th>gene.id</th>\n",
       "      <th>gene.strand</th>\n",
       "      <th>probe.chr</th>\n",
       "      <th>probe.start</th>\n",
       "      <th>probe.stop</th>\n",
       "      <th>probe.id</th>\n",
       "      <th>probe.strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ENSG00000184731.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>38814</td>\n",
       "      <td>46870</td>\n",
       "      <td>ENSG00000184731.5</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>38814</td>\n",
       "      <td>46870</td>\n",
       "      <td>ENSG00000184731.5</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000035115.17</th>\n",
       "      <td>chr2</td>\n",
       "      <td>217730</td>\n",
       "      <td>266398</td>\n",
       "      <td>ENSG00000035115.17</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>217730</td>\n",
       "      <td>266398</td>\n",
       "      <td>ENSG00000035115.17</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143727.11</th>\n",
       "      <td>chr2</td>\n",
       "      <td>264393</td>\n",
       "      <td>278283</td>\n",
       "      <td>ENSG00000143727.11</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>264393</td>\n",
       "      <td>278283</td>\n",
       "      <td>ENSG00000143727.11</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000189292.11</th>\n",
       "      <td>chr2</td>\n",
       "      <td>279558</td>\n",
       "      <td>288851</td>\n",
       "      <td>ENSG00000189292.11</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>279558</td>\n",
       "      <td>288851</td>\n",
       "      <td>ENSG00000189292.11</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000151353.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>667335</td>\n",
       "      <td>676704</td>\n",
       "      <td>ENSG00000151353.10</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>667335</td>\n",
       "      <td>676704</td>\n",
       "      <td>ENSG00000151353.10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172554.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>946554</td>\n",
       "      <td>1371385</td>\n",
       "      <td>ENSG00000172554.7</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>946554</td>\n",
       "      <td>1371385</td>\n",
       "      <td>ENSG00000172554.7</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115705.16</th>\n",
       "      <td>chr2</td>\n",
       "      <td>1377995</td>\n",
       "      <td>1547483</td>\n",
       "      <td>ENSG00000115705.16</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>1377995</td>\n",
       "      <td>1547483</td>\n",
       "      <td>ENSG00000115705.16</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000130508.6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>1635659</td>\n",
       "      <td>1748624</td>\n",
       "      <td>ENSG00000130508.6</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>1635659</td>\n",
       "      <td>1748624</td>\n",
       "      <td>ENSG00000130508.6</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000186487.13</th>\n",
       "      <td>chr2</td>\n",
       "      <td>1792885</td>\n",
       "      <td>2335032</td>\n",
       "      <td>ENSG00000186487.13</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>1792885</td>\n",
       "      <td>2335032</td>\n",
       "      <td>ENSG00000186487.13</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000032389.8</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3192696</td>\n",
       "      <td>3381653</td>\n",
       "      <td>ENSG00000032389.8</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3192696</td>\n",
       "      <td>3381653</td>\n",
       "      <td>ENSG00000032389.8</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171853.11</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3383446</td>\n",
       "      <td>3488865</td>\n",
       "      <td>ENSG00000171853.11</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3383446</td>\n",
       "      <td>3488865</td>\n",
       "      <td>ENSG00000171853.11</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000182551.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3501200</td>\n",
       "      <td>3521518</td>\n",
       "      <td>ENSG00000182551.9</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3501200</td>\n",
       "      <td>3521518</td>\n",
       "      <td>ENSG00000182551.9</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171865.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3592383</td>\n",
       "      <td>3606081</td>\n",
       "      <td>ENSG00000171865.5</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3592383</td>\n",
       "      <td>3606081</td>\n",
       "      <td>ENSG00000171865.5</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000171863.8</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3622795</td>\n",
       "      <td>3628509</td>\n",
       "      <td>ENSG00000171863.8</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3622795</td>\n",
       "      <td>3628509</td>\n",
       "      <td>ENSG00000171863.8</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000118004.13</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3642426</td>\n",
       "      <td>3692048</td>\n",
       "      <td>ENSG00000118004.13</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3642426</td>\n",
       "      <td>3692048</td>\n",
       "      <td>ENSG00000118004.13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000151360.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3705785</td>\n",
       "      <td>3750174</td>\n",
       "      <td>ENSG00000151360.5</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3705785</td>\n",
       "      <td>3750174</td>\n",
       "      <td>ENSG00000151360.5</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000214866.3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>3751453</td>\n",
       "      <td>3836122</td>\n",
       "      <td>ENSG00000214866.3</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>3751453</td>\n",
       "      <td>3836122</td>\n",
       "      <td>ENSG00000214866.3</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000134326.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>6980701</td>\n",
       "      <td>7006766</td>\n",
       "      <td>ENSG00000134326.7</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>6980701</td>\n",
       "      <td>7006766</td>\n",
       "      <td>ENSG00000134326.7</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000134321.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>7005937</td>\n",
       "      <td>7038370</td>\n",
       "      <td>ENSG00000134321.7</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>7005937</td>\n",
       "      <td>7038370</td>\n",
       "      <td>ENSG00000134321.7</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000151692.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>7057523</td>\n",
       "      <td>7208417</td>\n",
       "      <td>ENSG00000151692.10</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>7057523</td>\n",
       "      <td>7208417</td>\n",
       "      <td>ENSG00000151692.10</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115738.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>8818984</td>\n",
       "      <td>8824583</td>\n",
       "      <td>ENSG00000115738.5</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>8818984</td>\n",
       "      <td>8824583</td>\n",
       "      <td>ENSG00000115738.5</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000134313.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>8865408</td>\n",
       "      <td>8977760</td>\n",
       "      <td>ENSG00000134313.10</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>8865408</td>\n",
       "      <td>8977760</td>\n",
       "      <td>ENSG00000134313.10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000143797.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>8992820</td>\n",
       "      <td>9143942</td>\n",
       "      <td>ENSG00000143797.7</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>8992820</td>\n",
       "      <td>9143942</td>\n",
       "      <td>ENSG00000143797.7</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000151693.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9346894</td>\n",
       "      <td>9543603</td>\n",
       "      <td>ENSG00000151693.5</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9346894</td>\n",
       "      <td>9543603</td>\n",
       "      <td>ENSG00000151693.5</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000119185.8</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9545813</td>\n",
       "      <td>9563676</td>\n",
       "      <td>ENSG00000119185.8</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9545813</td>\n",
       "      <td>9563676</td>\n",
       "      <td>ENSG00000119185.8</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000119203.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9563697</td>\n",
       "      <td>9613230</td>\n",
       "      <td>ENSG00000119203.9</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9563697</td>\n",
       "      <td>9613230</td>\n",
       "      <td>ENSG00000119203.9</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000134330.14</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9613787</td>\n",
       "      <td>9636672</td>\n",
       "      <td>ENSG00000134330.14</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9613787</td>\n",
       "      <td>9636672</td>\n",
       "      <td>ENSG00000134330.14</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000151694.8</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9629872</td>\n",
       "      <td>9695921</td>\n",
       "      <td>ENSG00000151694.8</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9629872</td>\n",
       "      <td>9695921</td>\n",
       "      <td>ENSG00000151694.8</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000134308.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9724101</td>\n",
       "      <td>9771143</td>\n",
       "      <td>ENSG00000134308.9</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9724101</td>\n",
       "      <td>9771143</td>\n",
       "      <td>ENSG00000134308.9</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115750.12</th>\n",
       "      <td>chr2</td>\n",
       "      <td>9983483</td>\n",
       "      <td>10074545</td>\n",
       "      <td>ENSG00000115750.12</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>9983483</td>\n",
       "      <td>10074545</td>\n",
       "      <td>ENSG00000115750.12</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000218416.3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241388835</td>\n",
       "      <td>241396131</td>\n",
       "      <td>ENSG00000218416.3</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241388835</td>\n",
       "      <td>241396131</td>\n",
       "      <td>ENSG00000218416.3</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000144504.11</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241418839</td>\n",
       "      <td>241508626</td>\n",
       "      <td>ENSG00000144504.11</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241418839</td>\n",
       "      <td>241508626</td>\n",
       "      <td>ENSG00000144504.11</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188542.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241499471</td>\n",
       "      <td>241503431</td>\n",
       "      <td>ENSG00000188542.5</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241499471</td>\n",
       "      <td>241503431</td>\n",
       "      <td>ENSG00000188542.5</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000142327.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241505221</td>\n",
       "      <td>241520789</td>\n",
       "      <td>ENSG00000142327.7</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241505221</td>\n",
       "      <td>241520789</td>\n",
       "      <td>ENSG00000142327.7</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000142330.15</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241526133</td>\n",
       "      <td>241557122</td>\n",
       "      <td>ENSG00000142330.15</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241526133</td>\n",
       "      <td>241557122</td>\n",
       "      <td>ENSG00000142330.15</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000178623.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241544848</td>\n",
       "      <td>241570676</td>\n",
       "      <td>ENSG00000178623.7</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241544848</td>\n",
       "      <td>241570676</td>\n",
       "      <td>ENSG00000178623.7</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000130294.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241653181</td>\n",
       "      <td>241759725</td>\n",
       "      <td>ENSG00000130294.10</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241653181</td>\n",
       "      <td>241759725</td>\n",
       "      <td>ENSG00000130294.10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172482.4</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241807896</td>\n",
       "      <td>241818537</td>\n",
       "      <td>ENSG00000172482.4</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241807896</td>\n",
       "      <td>241818537</td>\n",
       "      <td>ENSG00000172482.4</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000172478.13</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241825465</td>\n",
       "      <td>241836306</td>\n",
       "      <td>ENSG00000172478.13</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241825465</td>\n",
       "      <td>241836306</td>\n",
       "      <td>ENSG00000172478.13</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000226321.3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241858202</td>\n",
       "      <td>241932728</td>\n",
       "      <td>ENSG00000226321.3</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241858202</td>\n",
       "      <td>241932728</td>\n",
       "      <td>ENSG00000226321.3</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000162804.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>241938255</td>\n",
       "      <td>242034011</td>\n",
       "      <td>ENSG00000162804.9</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>241938255</td>\n",
       "      <td>242034011</td>\n",
       "      <td>ENSG00000162804.9</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000122085.12</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242011584</td>\n",
       "      <td>242041747</td>\n",
       "      <td>ENSG00000122085.12</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242011584</td>\n",
       "      <td>242041747</td>\n",
       "      <td>ENSG00000122085.12</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115687.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242045514</td>\n",
       "      <td>242088919</td>\n",
       "      <td>ENSG00000115687.9</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242045514</td>\n",
       "      <td>242088919</td>\n",
       "      <td>ENSG00000115687.9</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115685.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242088991</td>\n",
       "      <td>242123067</td>\n",
       "      <td>ENSG00000115685.10</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242088991</td>\n",
       "      <td>242123067</td>\n",
       "      <td>ENSG00000115685.10</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000146205.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242127924</td>\n",
       "      <td>242164791</td>\n",
       "      <td>ENSG00000146205.9</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242127924</td>\n",
       "      <td>242164791</td>\n",
       "      <td>ENSG00000146205.9</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115677.12</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242166679</td>\n",
       "      <td>242256476</td>\n",
       "      <td>ENSG00000115677.12</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242166679</td>\n",
       "      <td>242256476</td>\n",
       "      <td>ENSG00000115677.12</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000168385.13</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242254515</td>\n",
       "      <td>242293442</td>\n",
       "      <td>ENSG00000168385.13</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242254515</td>\n",
       "      <td>242293442</td>\n",
       "      <td>ENSG00000168385.13</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000006607.9</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242295658</td>\n",
       "      <td>242431094</td>\n",
       "      <td>ENSG00000006607.9</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242295658</td>\n",
       "      <td>242431094</td>\n",
       "      <td>ENSG00000006607.9</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000115694.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242432089</td>\n",
       "      <td>242449145</td>\n",
       "      <td>ENSG00000115694.10</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242432089</td>\n",
       "      <td>242449145</td>\n",
       "      <td>ENSG00000115694.10</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000176720.3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242498136</td>\n",
       "      <td>242513546</td>\n",
       "      <td>ENSG00000176720.3</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242498136</td>\n",
       "      <td>242513546</td>\n",
       "      <td>ENSG00000176720.3</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000176946.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242523820</td>\n",
       "      <td>242576627</td>\n",
       "      <td>ENSG00000176946.7</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242523820</td>\n",
       "      <td>242576627</td>\n",
       "      <td>ENSG00000176946.7</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000168397.12</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242576865</td>\n",
       "      <td>242613272</td>\n",
       "      <td>ENSG00000168397.12</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242576865</td>\n",
       "      <td>242613272</td>\n",
       "      <td>ENSG00000168397.12</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000168393.8</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242615157</td>\n",
       "      <td>242626406</td>\n",
       "      <td>ENSG00000168393.8</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242615157</td>\n",
       "      <td>242626406</td>\n",
       "      <td>ENSG00000168393.8</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000168395.10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242641455</td>\n",
       "      <td>242668893</td>\n",
       "      <td>ENSG00000168395.10</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242641455</td>\n",
       "      <td>242668893</td>\n",
       "      <td>ENSG00000168395.10</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000180902.12</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242673994</td>\n",
       "      <td>242708226</td>\n",
       "      <td>ENSG00000180902.12</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242673994</td>\n",
       "      <td>242708226</td>\n",
       "      <td>ENSG00000180902.12</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000154252.11</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242716240</td>\n",
       "      <td>242743623</td>\n",
       "      <td>ENSG00000154252.11</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242716240</td>\n",
       "      <td>242743623</td>\n",
       "      <td>ENSG00000154252.11</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000204099.7</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242750902</td>\n",
       "      <td>242758739</td>\n",
       "      <td>ENSG00000204099.7</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242750902</td>\n",
       "      <td>242758739</td>\n",
       "      <td>ENSG00000204099.7</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188389.6</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242792033</td>\n",
       "      <td>242801060</td>\n",
       "      <td>ENSG00000188389.6</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242792033</td>\n",
       "      <td>242801060</td>\n",
       "      <td>ENSG00000188389.6</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000188011.5</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242811752</td>\n",
       "      <td>242815975</td>\n",
       "      <td>ENSG00000188011.5</td>\n",
       "      <td>+</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242811752</td>\n",
       "      <td>242815975</td>\n",
       "      <td>ENSG00000188011.5</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ENSG00000216921.3</th>\n",
       "      <td>chr2</td>\n",
       "      <td>242836136</td>\n",
       "      <td>242844702</td>\n",
       "      <td>ENSG00000216921.3</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>242836136</td>\n",
       "      <td>242844702</td>\n",
       "      <td>ENSG00000216921.3</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1111 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   gene.chr  gene.start  gene.stop             gene.id  \\\n",
       "ENSG00000184731.5      chr2       38814      46870   ENSG00000184731.5   \n",
       "ENSG00000035115.17     chr2      217730     266398  ENSG00000035115.17   \n",
       "ENSG00000143727.11     chr2      264393     278283  ENSG00000143727.11   \n",
       "ENSG00000189292.11     chr2      279558     288851  ENSG00000189292.11   \n",
       "ENSG00000151353.10     chr2      667335     676704  ENSG00000151353.10   \n",
       "ENSG00000172554.7      chr2      946554    1371385   ENSG00000172554.7   \n",
       "ENSG00000115705.16     chr2     1377995    1547483  ENSG00000115705.16   \n",
       "ENSG00000130508.6      chr2     1635659    1748624   ENSG00000130508.6   \n",
       "ENSG00000186487.13     chr2     1792885    2335032  ENSG00000186487.13   \n",
       "ENSG00000032389.8      chr2     3192696    3381653   ENSG00000032389.8   \n",
       "ENSG00000171853.11     chr2     3383446    3488865  ENSG00000171853.11   \n",
       "ENSG00000182551.9      chr2     3501200    3521518   ENSG00000182551.9   \n",
       "ENSG00000171865.5      chr2     3592383    3606081   ENSG00000171865.5   \n",
       "ENSG00000171863.8      chr2     3622795    3628509   ENSG00000171863.8   \n",
       "ENSG00000118004.13     chr2     3642426    3692048  ENSG00000118004.13   \n",
       "ENSG00000151360.5      chr2     3705785    3750174   ENSG00000151360.5   \n",
       "ENSG00000214866.3      chr2     3751453    3836122   ENSG00000214866.3   \n",
       "ENSG00000134326.7      chr2     6980701    7006766   ENSG00000134326.7   \n",
       "ENSG00000134321.7      chr2     7005937    7038370   ENSG00000134321.7   \n",
       "ENSG00000151692.10     chr2     7057523    7208417  ENSG00000151692.10   \n",
       "ENSG00000115738.5      chr2     8818984    8824583   ENSG00000115738.5   \n",
       "ENSG00000134313.10     chr2     8865408    8977760  ENSG00000134313.10   \n",
       "ENSG00000143797.7      chr2     8992820    9143942   ENSG00000143797.7   \n",
       "ENSG00000151693.5      chr2     9346894    9543603   ENSG00000151693.5   \n",
       "ENSG00000119185.8      chr2     9545813    9563676   ENSG00000119185.8   \n",
       "ENSG00000119203.9      chr2     9563697    9613230   ENSG00000119203.9   \n",
       "ENSG00000134330.14     chr2     9613787    9636672  ENSG00000134330.14   \n",
       "ENSG00000151694.8      chr2     9629872    9695921   ENSG00000151694.8   \n",
       "ENSG00000134308.9      chr2     9724101    9771143   ENSG00000134308.9   \n",
       "ENSG00000115750.12     chr2     9983483   10074545  ENSG00000115750.12   \n",
       "...                     ...         ...        ...                 ...   \n",
       "ENSG00000218416.3      chr2   241388835  241396131   ENSG00000218416.3   \n",
       "ENSG00000144504.11     chr2   241418839  241508626  ENSG00000144504.11   \n",
       "ENSG00000188542.5      chr2   241499471  241503431   ENSG00000188542.5   \n",
       "ENSG00000142327.7      chr2   241505221  241520789   ENSG00000142327.7   \n",
       "ENSG00000142330.15     chr2   241526133  241557122  ENSG00000142330.15   \n",
       "ENSG00000178623.7      chr2   241544848  241570676   ENSG00000178623.7   \n",
       "ENSG00000130294.10     chr2   241653181  241759725  ENSG00000130294.10   \n",
       "ENSG00000172482.4      chr2   241807896  241818537   ENSG00000172482.4   \n",
       "ENSG00000172478.13     chr2   241825465  241836306  ENSG00000172478.13   \n",
       "ENSG00000226321.3      chr2   241858202  241932728   ENSG00000226321.3   \n",
       "ENSG00000162804.9      chr2   241938255  242034011   ENSG00000162804.9   \n",
       "ENSG00000122085.12     chr2   242011584  242041747  ENSG00000122085.12   \n",
       "ENSG00000115687.9      chr2   242045514  242088919   ENSG00000115687.9   \n",
       "ENSG00000115685.10     chr2   242088991  242123067  ENSG00000115685.10   \n",
       "ENSG00000146205.9      chr2   242127924  242164791   ENSG00000146205.9   \n",
       "ENSG00000115677.12     chr2   242166679  242256476  ENSG00000115677.12   \n",
       "ENSG00000168385.13     chr2   242254515  242293442  ENSG00000168385.13   \n",
       "ENSG00000006607.9      chr2   242295658  242431094   ENSG00000006607.9   \n",
       "ENSG00000115694.10     chr2   242432089  242449145  ENSG00000115694.10   \n",
       "ENSG00000176720.3      chr2   242498136  242513546   ENSG00000176720.3   \n",
       "ENSG00000176946.7      chr2   242523820  242576627   ENSG00000176946.7   \n",
       "ENSG00000168397.12     chr2   242576865  242613272  ENSG00000168397.12   \n",
       "ENSG00000168393.8      chr2   242615157  242626406   ENSG00000168393.8   \n",
       "ENSG00000168395.10     chr2   242641455  242668893  ENSG00000168395.10   \n",
       "ENSG00000180902.12     chr2   242673994  242708226  ENSG00000180902.12   \n",
       "ENSG00000154252.11     chr2   242716240  242743623  ENSG00000154252.11   \n",
       "ENSG00000204099.7      chr2   242750902  242758739   ENSG00000204099.7   \n",
       "ENSG00000188389.6      chr2   242792033  242801060   ENSG00000188389.6   \n",
       "ENSG00000188011.5      chr2   242811752  242815975   ENSG00000188011.5   \n",
       "ENSG00000216921.3      chr2   242836136  242844702   ENSG00000216921.3   \n",
       "\n",
       "                   gene.strand probe.chr  probe.start  probe.stop  \\\n",
       "ENSG00000184731.5            -      chr2        38814       46870   \n",
       "ENSG00000035115.17           -      chr2       217730      266398   \n",
       "ENSG00000143727.11           +      chr2       264393      278283   \n",
       "ENSG00000189292.11           -      chr2       279558      288851   \n",
       "ENSG00000151353.10           -      chr2       667335      676704   \n",
       "ENSG00000172554.7            +      chr2       946554     1371385   \n",
       "ENSG00000115705.16           +      chr2      1377995     1547483   \n",
       "ENSG00000130508.6            -      chr2      1635659     1748624   \n",
       "ENSG00000186487.13           -      chr2      1792885     2335032   \n",
       "ENSG00000032389.8            -      chr2      3192696     3381653   \n",
       "ENSG00000171853.11           +      chr2      3383446     3488865   \n",
       "ENSG00000182551.9            -      chr2      3501200     3521518   \n",
       "ENSG00000171865.5            -      chr2      3592383     3606081   \n",
       "ENSG00000171863.8            +      chr2      3622795     3628509   \n",
       "ENSG00000118004.13           +      chr2      3642426     3692048   \n",
       "ENSG00000151360.5            +      chr2      3705785     3750174   \n",
       "ENSG00000214866.3            +      chr2      3751453     3836122   \n",
       "ENSG00000134326.7            -      chr2      6980701     7006766   \n",
       "ENSG00000134321.7            +      chr2      7005937     7038370   \n",
       "ENSG00000151692.10           +      chr2      7057523     7208417   \n",
       "ENSG00000115738.5            +      chr2      8818984     8824583   \n",
       "ENSG00000134313.10           -      chr2      8865408     8977760   \n",
       "ENSG00000143797.7            -      chr2      8992820     9143942   \n",
       "ENSG00000151693.5            +      chr2      9346894     9543603   \n",
       "ENSG00000119185.8            -      chr2      9545813     9563676   \n",
       "ENSG00000119203.9            +      chr2      9563697     9613230   \n",
       "ENSG00000134330.14           +      chr2      9613787     9636672   \n",
       "ENSG00000151694.8            -      chr2      9629872     9695921   \n",
       "ENSG00000134308.9            -      chr2      9724101     9771143   \n",
       "ENSG00000115750.12           +      chr2      9983483    10074545   \n",
       "...                        ...       ...          ...         ...   \n",
       "ENSG00000218416.3            -      chr2    241388835   241396131   \n",
       "ENSG00000144504.11           -      chr2    241418839   241508626   \n",
       "ENSG00000188542.5            +      chr2    241499471   241503431   \n",
       "ENSG00000142327.7            +      chr2    241505221   241520789   \n",
       "ENSG00000142330.15           +      chr2    241526133   241557122   \n",
       "ENSG00000178623.7            +      chr2    241544848   241570676   \n",
       "ENSG00000130294.10           -      chr2    241653181   241759725   \n",
       "ENSG00000172482.4            +      chr2    241807896   241818537   \n",
       "ENSG00000172478.13           -      chr2    241825465   241836306   \n",
       "ENSG00000226321.3            +      chr2    241858202   241932728   \n",
       "ENSG00000162804.9            +      chr2    241938255   242034011   \n",
       "ENSG00000122085.12           -      chr2    242011584   242041747   \n",
       "ENSG00000115687.9            -      chr2    242045514   242088919   \n",
       "ENSG00000115685.10           +      chr2    242088991   242123067   \n",
       "ENSG00000146205.9            +      chr2    242127924   242164791   \n",
       "ENSG00000115677.12           -      chr2    242166679   242256476   \n",
       "ENSG00000168385.13           +      chr2    242254515   242293442   \n",
       "ENSG00000006607.9            +      chr2    242295658   242431094   \n",
       "ENSG00000115694.10           -      chr2    242432089   242449145   \n",
       "ENSG00000176720.3            +      chr2    242498136   242513546   \n",
       "ENSG00000176946.7            -      chr2    242523820   242576627   \n",
       "ENSG00000168397.12           +      chr2    242576865   242613272   \n",
       "ENSG00000168393.8            -      chr2    242615157   242626406   \n",
       "ENSG00000168395.10           +      chr2    242641455   242668893   \n",
       "ENSG00000180902.12           +      chr2    242673994   242708226   \n",
       "ENSG00000154252.11           +      chr2    242716240   242743623   \n",
       "ENSG00000204099.7            +      chr2    242750902   242758739   \n",
       "ENSG00000188389.6            -      chr2    242792033   242801060   \n",
       "ENSG00000188011.5            +      chr2    242811752   242815975   \n",
       "ENSG00000216921.3            -      chr2    242836136   242844702   \n",
       "\n",
       "                              probe.id probe.strand  \n",
       "ENSG00000184731.5    ENSG00000184731.5            -  \n",
       "ENSG00000035115.17  ENSG00000035115.17            -  \n",
       "ENSG00000143727.11  ENSG00000143727.11            +  \n",
       "ENSG00000189292.11  ENSG00000189292.11            -  \n",
       "ENSG00000151353.10  ENSG00000151353.10            -  \n",
       "ENSG00000172554.7    ENSG00000172554.7            +  \n",
       "ENSG00000115705.16  ENSG00000115705.16            +  \n",
       "ENSG00000130508.6    ENSG00000130508.6            -  \n",
       "ENSG00000186487.13  ENSG00000186487.13            -  \n",
       "ENSG00000032389.8    ENSG00000032389.8            -  \n",
       "ENSG00000171853.11  ENSG00000171853.11            +  \n",
       "ENSG00000182551.9    ENSG00000182551.9            -  \n",
       "ENSG00000171865.5    ENSG00000171865.5            -  \n",
       "ENSG00000171863.8    ENSG00000171863.8            +  \n",
       "ENSG00000118004.13  ENSG00000118004.13            +  \n",
       "ENSG00000151360.5    ENSG00000151360.5            +  \n",
       "ENSG00000214866.3    ENSG00000214866.3            +  \n",
       "ENSG00000134326.7    ENSG00000134326.7            -  \n",
       "ENSG00000134321.7    ENSG00000134321.7            +  \n",
       "ENSG00000151692.10  ENSG00000151692.10            +  \n",
       "ENSG00000115738.5    ENSG00000115738.5            +  \n",
       "ENSG00000134313.10  ENSG00000134313.10            -  \n",
       "ENSG00000143797.7    ENSG00000143797.7            -  \n",
       "ENSG00000151693.5    ENSG00000151693.5            +  \n",
       "ENSG00000119185.8    ENSG00000119185.8            -  \n",
       "ENSG00000119203.9    ENSG00000119203.9            +  \n",
       "ENSG00000134330.14  ENSG00000134330.14            +  \n",
       "ENSG00000151694.8    ENSG00000151694.8            -  \n",
       "ENSG00000134308.9    ENSG00000134308.9            -  \n",
       "ENSG00000115750.12  ENSG00000115750.12            +  \n",
       "...                                ...          ...  \n",
       "ENSG00000218416.3    ENSG00000218416.3            -  \n",
       "ENSG00000144504.11  ENSG00000144504.11            -  \n",
       "ENSG00000188542.5    ENSG00000188542.5            +  \n",
       "ENSG00000142327.7    ENSG00000142327.7            +  \n",
       "ENSG00000142330.15  ENSG00000142330.15            +  \n",
       "ENSG00000178623.7    ENSG00000178623.7            +  \n",
       "ENSG00000130294.10  ENSG00000130294.10            -  \n",
       "ENSG00000172482.4    ENSG00000172482.4            +  \n",
       "ENSG00000172478.13  ENSG00000172478.13            -  \n",
       "ENSG00000226321.3    ENSG00000226321.3            +  \n",
       "ENSG00000162804.9    ENSG00000162804.9            +  \n",
       "ENSG00000122085.12  ENSG00000122085.12            -  \n",
       "ENSG00000115687.9    ENSG00000115687.9            -  \n",
       "ENSG00000115685.10  ENSG00000115685.10            +  \n",
       "ENSG00000146205.9    ENSG00000146205.9            +  \n",
       "ENSG00000115677.12  ENSG00000115677.12            -  \n",
       "ENSG00000168385.13  ENSG00000168385.13            +  \n",
       "ENSG00000006607.9    ENSG00000006607.9            +  \n",
       "ENSG00000115694.10  ENSG00000115694.10            -  \n",
       "ENSG00000176720.3    ENSG00000176720.3            +  \n",
       "ENSG00000176946.7    ENSG00000176946.7            -  \n",
       "ENSG00000168397.12  ENSG00000168397.12            +  \n",
       "ENSG00000168393.8    ENSG00000168393.8            -  \n",
       "ENSG00000168395.10  ENSG00000168395.10            +  \n",
       "ENSG00000180902.12  ENSG00000180902.12            +  \n",
       "ENSG00000154252.11  ENSG00000154252.11            +  \n",
       "ENSG00000204099.7    ENSG00000204099.7            +  \n",
       "ENSG00000188389.6    ENSG00000188389.6            -  \n",
       "ENSG00000188011.5    ENSG00000188011.5            +  \n",
       "ENSG00000216921.3    ENSG00000216921.3            -  \n",
       "\n",
       "[1111 rows x 10 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expr_annot\n",
    "#expr_annot.loc[expr_annot.index =='ENSG00000143727.11']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
