{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.4/site-packages/IPython/core/interactiveshell.py:2785: DtypeWarning: Columns (12,23,27) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Body mass index 841\n",
      "Schizophrenia 899\n",
      "Type 2 diabetes 396\n",
      "Inflammatory bowel disease 319\n",
      "Rheumatoid arthritis 309\n",
      "Obesity-related traits 957\n",
      "Height 823\n",
      "Post bronchodilator FEV1/FVC ratio in COPD 473\n",
      "IgG glycosylation 699\n",
      "Post bronchodilator FEV1 1182\n",
      "Post bronchodilator FEV1/FVC ratio 1631\n",
      "Crohn's disease 390\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import re\n",
    "\n",
    "#Causality\n",
    "Causal_scores = pd.read_csv(\"/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/Merged_Best_causality.Table\")\n",
    "\n",
    "#gwas STRs (in literature)\n",
    "STR_gwas_list = ['JARID2','TH','ATXN1','GCLC','NUMBL','PHOX2B','CCK','CETP','CNBP','CYP19A1','GC','GRIK1','LEP','MIF','MOG','PTHR1','TCF7L2','TFAP2B','TNFRSF8','AFF2','AR','ARX','ASCL1','ATN1','ATXN2','ATXN3','ATXN7','ATXN8OS','CACNA1A','CBL','CNDP1','CNR1','DIP2B','DMPK','FMR1','FOXC1','FRA10A','FXN','GCLC','GSPT1','HD','HOXA13','HOXD13','JPH3','KCNN3','MECP2B','MICA','NCOA3','NOS1','PABPN1','PAX7','PCSK9','PHOX2B','POLG','PPP2R2B','RENL','RETN','RUNX2','SIX5','TBP','TGFBR1','VLDLR']\n",
    "#gwas file -- downloaded from \n",
    "gwas = pd.read_csv(\"/storage/szfeupe/Runs/650GTEx_estr/gwas/gwas-catalog-associations.tsv\", sep='\\t') \n",
    "gwas.columns\n",
    "######\n",
    "selectedcolumns=[ 'DISEASE/TRAIT',  'REGION', 'CHR_ID', 'CHR_POS','REPORTED GENE(S)', 'MAPPED_GENE', 'SNP_GENE_IDS', 'UPSTREAM_GENE_DISTANCE', 'DOWNSTREAM_GENE_DISTANCE', 'STRONGEST SNP-RISK ALLELE', 'SNPS',  'CONTEXT', 'INTERGENIC', 'RISK ALLELE FREQUENCY', 'P-VALUE', 'PVALUE_MLOG', 'OR or BETA']\n",
    "for i in set(list(gwas['DISEASE/TRAIT'])):\n",
    "    if list(gwas['DISEASE/TRAIT']).count(i)>300:\n",
    "        print(i, list(gwas['DISEASE/TRAIT']).count(i))\n",
    "        \n",
    "selectedcolumns = ['DISEASE/TRAIT',  'REGION', 'CHR_ID', 'CHR_POS','REPORTED GENE(S)', 'MAPPED_GENE','SNP_GENE_IDS', 'SNPS',  'CONTEXT', 'INTERGENIC', 'RISK ALLELE FREQUENCY', 'P-VALUE', 'PVALUE_MLOG' ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1312\n",
      "946\n"
     ]
    }
   ],
   "source": [
    "#Get genes associated with trait from gwas studies (SNPs)\n",
    "\n",
    "trait='Schizophrenia'\n",
    "condition = gwas.loc[gwas['DISEASE/TRAIT']==trait][selectedcolumns].sort_values(\"CHR_ID\")\n",
    "genes = []\n",
    "for g in [x.split('-') for x in list(condition['MAPPED_GENE'].astype(str))]:\n",
    "    genes=genes+g\n",
    "print(len(genes))\n",
    "genes = [x.strip() for x in genes if 'LOC' not in x]\n",
    "print(len(genes))\n",
    "condition['MAPPED_GENE']=condition['MAPPED_GENE'].astype(str).apply(lambda x: x.split('-')+['*'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(88752, 29)\n",
      "946\n",
      "430\n",
      "Number of tests  3634\n",
      "Number of gwas eGenes for schizophrenia:  190\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chrom</th>\n",
       "      <th>str.start</th>\n",
       "      <th>E.tissues</th>\n",
       "      <th>str.end</th>\n",
       "      <th>period</th>\n",
       "      <th>motif</th>\n",
       "      <th>gene.name</th>\n",
       "      <th>str.len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000177674.11</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11793667</td>\n",
       "      <td>1.0</td>\n",
       "      <td>11793684</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>AGTRAP</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>chr1</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000121766.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>31672838</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31672886</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>ZCCHC17</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000121766.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>31685250</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31685267</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>ZCCHC17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000142949.12</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44025261</td>\n",
       "      <td>1.0</td>\n",
       "      <td>44025272</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000120341.14</td>\n",
       "      <td>chr1</td>\n",
       "      <td>177982159</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177982169</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>SEC16B</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ENSG00000120341.14</td>\n",
       "      <td>chr1</td>\n",
       "      <td>178037855</td>\n",
       "      <td>1.0</td>\n",
       "      <td>178037909</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>SEC16B</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ENSG00000120341.14</td>\n",
       "      <td>chr1</td>\n",
       "      <td>177918519</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177918565</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>SEC16B</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ENSG00000079277.15</td>\n",
       "      <td>chr1</td>\n",
       "      <td>47014814</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47014832</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>MKNK1</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000079277.15</td>\n",
       "      <td>chr1</td>\n",
       "      <td>47071277</td>\n",
       "      <td>3.0</td>\n",
       "      <td>47071296</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>MKNK1</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ENSG00000134698.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>36270731</td>\n",
       "      <td>3.0</td>\n",
       "      <td>36270758</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>ENSG00000134698.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>36284227</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36284305</td>\n",
       "      <td>4</td>\n",
       "      <td>AAGG</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>ENSG00000143622.6</td>\n",
       "      <td>chr1</td>\n",
       "      <td>155915072</td>\n",
       "      <td>4.0</td>\n",
       "      <td>155915114</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>RIT1</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSG00000117020.12</td>\n",
       "      <td>chr1</td>\n",
       "      <td>243861275</td>\n",
       "      <td>1.0</td>\n",
       "      <td>243861289</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ENSG00000184144.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>205049654</td>\n",
       "      <td>1.0</td>\n",
       "      <td>205049690</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>CNTN2</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ENSG00000184144.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>205094969</td>\n",
       "      <td>2.0</td>\n",
       "      <td>205095016</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>CNTN2</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ENSG00000117569.14</td>\n",
       "      <td>chr1</td>\n",
       "      <td>97211622</td>\n",
       "      <td>1.0</td>\n",
       "      <td>97211654</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>PTBP2</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ENSG00000117707.11</td>\n",
       "      <td>chr1</td>\n",
       "      <td>214137474</td>\n",
       "      <td>1.0</td>\n",
       "      <td>214137525</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>PROX1</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ENSG00000134198.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>115517955</td>\n",
       "      <td>1.0</td>\n",
       "      <td>115517980</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>TSPAN2</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ENSG00000162426.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8373565</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8373584</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>SLC45A1</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>ENSG00000142599.13</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8661796</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8661856</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>RERE</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>ENSG00000142599.13</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8695997</td>\n",
       "      <td>3.0</td>\n",
       "      <td>8696063</td>\n",
       "      <td>5</td>\n",
       "      <td>AAAAT</td>\n",
       "      <td>RERE</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>ENSG00000152382.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>166861300</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166861329</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>TADA1</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>ENSG00000152382.5</td>\n",
       "      <td>chr1</td>\n",
       "      <td>166865783</td>\n",
       "      <td>4.0</td>\n",
       "      <td>166865802</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>TADA1</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>ENSG00000143195.8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>166865783</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166865802</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>ILDR2</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>ENSG00000143195.8</td>\n",
       "      <td>chr1</td>\n",
       "      <td>167033526</td>\n",
       "      <td>2.0</td>\n",
       "      <td>167033573</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAG</td>\n",
       "      <td>ILDR2</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENSG00000143382.9</td>\n",
       "      <td>chr1</td>\n",
       "      <td>150552235</td>\n",
       "      <td>1.0</td>\n",
       "      <td>150552290</td>\n",
       "      <td>6</td>\n",
       "      <td>CCCCGG</td>\n",
       "      <td>ADAMTSL4</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>ENSG00000162909.13</td>\n",
       "      <td>chr1</td>\n",
       "      <td>223894939</td>\n",
       "      <td>2.0</td>\n",
       "      <td>223895002</td>\n",
       "      <td>2</td>\n",
       "      <td>AG</td>\n",
       "      <td>CAPN2</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>ENSG00000143575.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>154200020</td>\n",
       "      <td>1.0</td>\n",
       "      <td>154200040</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>HAX1</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENSG00000160685.9</td>\n",
       "      <td>chr1</td>\n",
       "      <td>155027695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>155027718</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>246</th>\n",
       "      <td>ENSG00000135250.12</td>\n",
       "      <td>chr7</td>\n",
       "      <td>104694906</td>\n",
       "      <td>1.0</td>\n",
       "      <td>104694978</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>SRPK2</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>247</th>\n",
       "      <td>ENSG00000135250.12</td>\n",
       "      <td>chr7</td>\n",
       "      <td>105009568</td>\n",
       "      <td>2.0</td>\n",
       "      <td>105009610</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>SRPK2</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>248</th>\n",
       "      <td>ENSG00000105928.9</td>\n",
       "      <td>chr7</td>\n",
       "      <td>24894109</td>\n",
       "      <td>1.0</td>\n",
       "      <td>24894167</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>DFNA5</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>ENSG00000105928.9</td>\n",
       "      <td>chr7</td>\n",
       "      <td>24684115</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24684142</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>DFNA5</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>ENSG00000261115.1</td>\n",
       "      <td>chr7</td>\n",
       "      <td>140676038</td>\n",
       "      <td>4.0</td>\n",
       "      <td>140676092</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251</th>\n",
       "      <td>ENSG00000105866.9</td>\n",
       "      <td>chr7</td>\n",
       "      <td>21508137</td>\n",
       "      <td>1.0</td>\n",
       "      <td>21508154</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>SP4</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>252</th>\n",
       "      <td>ENSG00000105926.11</td>\n",
       "      <td>chr7</td>\n",
       "      <td>24684115</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24684142</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>MPP6</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253</th>\n",
       "      <td>ENSG00000105926.11</td>\n",
       "      <td>chr7</td>\n",
       "      <td>24693013</td>\n",
       "      <td>2.0</td>\n",
       "      <td>24693024</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>MPP6</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>254</th>\n",
       "      <td>ENSG00000131558.10</td>\n",
       "      <td>chr7</td>\n",
       "      <td>133347370</td>\n",
       "      <td>3.0</td>\n",
       "      <td>133347389</td>\n",
       "      <td>3</td>\n",
       "      <td>AAC</td>\n",
       "      <td>EXOC4</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>255</th>\n",
       "      <td>ENSG00000131558.10</td>\n",
       "      <td>chr7</td>\n",
       "      <td>133532122</td>\n",
       "      <td>4.0</td>\n",
       "      <td>133532149</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>EXOC4</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>256</th>\n",
       "      <td>ENSG00000146776.10</td>\n",
       "      <td>chr7</td>\n",
       "      <td>105357112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>105357167</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>ATXN7L1</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>257</th>\n",
       "      <td>ENSG00000183166.6</td>\n",
       "      <td>chr7</td>\n",
       "      <td>71845653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71845729</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>CALN1</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>258</th>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>chr7</td>\n",
       "      <td>110591595</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110591632</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>259</th>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>chr7</td>\n",
       "      <td>110602497</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110602508</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260</th>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>chr7</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>ENSG00000185274.7</td>\n",
       "      <td>chr7</td>\n",
       "      <td>70681544</td>\n",
       "      <td>4.0</td>\n",
       "      <td>70681616</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>WBSCR17</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>262</th>\n",
       "      <td>ENSG00000186472.15</td>\n",
       "      <td>chr7</td>\n",
       "      <td>82731428</td>\n",
       "      <td>1.0</td>\n",
       "      <td>82731451</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>PCLO</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263</th>\n",
       "      <td>ENSG00000196335.8</td>\n",
       "      <td>chr7</td>\n",
       "      <td>23880976</td>\n",
       "      <td>3.0</td>\n",
       "      <td>23881006</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>STK31</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>264</th>\n",
       "      <td>ENSG00000196335.8</td>\n",
       "      <td>chr7</td>\n",
       "      <td>23913814</td>\n",
       "      <td>1.0</td>\n",
       "      <td>23913888</td>\n",
       "      <td>6</td>\n",
       "      <td>AGAGAT</td>\n",
       "      <td>STK31</td>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>265</th>\n",
       "      <td>ENSG00000165046.8</td>\n",
       "      <td>chr8</td>\n",
       "      <td>38299881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38299913</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>LETM2</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>ENSG00000165046.8</td>\n",
       "      <td>chr8</td>\n",
       "      <td>38304617</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38304670</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>LETM2</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>ENSG00000153317.10</td>\n",
       "      <td>chr8</td>\n",
       "      <td>131345163</td>\n",
       "      <td>1.0</td>\n",
       "      <td>131345173</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>ASAP1</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>268</th>\n",
       "      <td>ENSG00000153317.10</td>\n",
       "      <td>chr8</td>\n",
       "      <td>131430786</td>\n",
       "      <td>1.0</td>\n",
       "      <td>131430825</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>ASAP1</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>ENSG00000173273.11</td>\n",
       "      <td>chr8</td>\n",
       "      <td>9467805</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9467819</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>TNKS</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270</th>\n",
       "      <td>ENSG00000175806.10</td>\n",
       "      <td>chr8</td>\n",
       "      <td>10307052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10307110</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271</th>\n",
       "      <td>ENSG00000107282.5</td>\n",
       "      <td>chr9</td>\n",
       "      <td>72341943</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72341968</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>APBA1</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>272</th>\n",
       "      <td>ENSG00000136828.14</td>\n",
       "      <td>chr9</td>\n",
       "      <td>129666377</td>\n",
       "      <td>1.0</td>\n",
       "      <td>129666400</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>RALGPS1</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>273</th>\n",
       "      <td>ENSG00000136828.14</td>\n",
       "      <td>chr9</td>\n",
       "      <td>129833393</td>\n",
       "      <td>3.0</td>\n",
       "      <td>129833410</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>RALGPS1</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>274</th>\n",
       "      <td>ENSG00000153707.11</td>\n",
       "      <td>chr9</td>\n",
       "      <td>9595947</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9595963</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>275</th>\n",
       "      <td>ENSG00000174482.6</td>\n",
       "      <td>chr9</td>\n",
       "      <td>28708760</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28708840</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>LINGO2</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>276 rows × 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   gene chrom  str.start  E.tissues    str.end  period  \\\n",
       "0    ENSG00000177674.11  chr1   11793667        1.0   11793684       1   \n",
       "1    ENSG00000054282.11  chr1  243394402       12.0  243394412       1   \n",
       "2    ENSG00000121766.10  chr1   31672838        1.0   31672886       2   \n",
       "3    ENSG00000121766.10  chr1   31685250        1.0   31685267       1   \n",
       "4    ENSG00000142949.12  chr1   44025261        1.0   44025272       2   \n",
       "5    ENSG00000120341.14  chr1  177982159        1.0  177982169       1   \n",
       "6    ENSG00000120341.14  chr1  178037855        1.0  178037909       2   \n",
       "7    ENSG00000120341.14  chr1  177918519        1.0  177918565       2   \n",
       "8    ENSG00000079277.15  chr1   47014814        1.0   47014832       1   \n",
       "9    ENSG00000079277.15  chr1   47071277        3.0   47071296       1   \n",
       "10   ENSG00000134698.10  chr1   36270731        3.0   36270758       1   \n",
       "11   ENSG00000134698.10  chr1   36284227        1.0   36284305       4   \n",
       "12    ENSG00000143622.6  chr1  155915072        4.0  155915114       2   \n",
       "13   ENSG00000117020.12  chr1  243861275        1.0  243861289       1   \n",
       "14    ENSG00000184144.5  chr1  205049654        1.0  205049690       2   \n",
       "15    ENSG00000184144.5  chr1  205094969        2.0  205095016       2   \n",
       "16   ENSG00000117569.14  chr1   97211622        1.0   97211654       1   \n",
       "17   ENSG00000117707.11  chr1  214137474        1.0  214137525       2   \n",
       "18    ENSG00000134198.5  chr1  115517955        1.0  115517980       2   \n",
       "19   ENSG00000162426.10  chr1    8373565        1.0    8373584       1   \n",
       "20   ENSG00000142599.13  chr1    8661796        2.0    8661856       2   \n",
       "21   ENSG00000142599.13  chr1    8695997        3.0    8696063       5   \n",
       "22    ENSG00000152382.5  chr1  166861300        1.0  166861329       3   \n",
       "23    ENSG00000152382.5  chr1  166865783        4.0  166865802       2   \n",
       "24    ENSG00000143195.8  chr1  166865783        1.0  166865802       2   \n",
       "25    ENSG00000143195.8  chr1  167033526        2.0  167033573       4   \n",
       "26    ENSG00000143382.9  chr1  150552235        1.0  150552290       6   \n",
       "27   ENSG00000162909.13  chr1  223894939        2.0  223895002       2   \n",
       "28   ENSG00000143575.10  chr1  154200020        1.0  154200040       1   \n",
       "29    ENSG00000160685.9  chr1  155027695        1.0  155027718       3   \n",
       "..                  ...   ...        ...        ...        ...     ...   \n",
       "246  ENSG00000135250.12  chr7  104694906        1.0  104694978       4   \n",
       "247  ENSG00000135250.12  chr7  105009568        2.0  105009610       3   \n",
       "248   ENSG00000105928.9  chr7   24894109        1.0   24894167       2   \n",
       "249   ENSG00000105928.9  chr7   24684115        2.0   24684142       2   \n",
       "250   ENSG00000261115.1  chr7  140676038        4.0  140676092       2   \n",
       "251   ENSG00000105866.9  chr7   21508137        1.0   21508154       1   \n",
       "252  ENSG00000105926.11  chr7   24684115        2.0   24684142       2   \n",
       "253  ENSG00000105926.11  chr7   24693013        2.0   24693024       1   \n",
       "254  ENSG00000131558.10  chr7  133347370        3.0  133347389       3   \n",
       "255  ENSG00000131558.10  chr7  133532122        4.0  133532149       1   \n",
       "256  ENSG00000146776.10  chr7  105357112        1.0  105357167       2   \n",
       "257   ENSG00000183166.6  chr7   71845653        1.0   71845729       4   \n",
       "258   ENSG00000184903.5  chr7  110591595        1.0  110591632       2   \n",
       "259   ENSG00000184903.5  chr7  110602497        1.0  110602508       1   \n",
       "260   ENSG00000184903.5  chr7  111191552        1.0  111191588       2   \n",
       "261   ENSG00000185274.7  chr7   70681544        4.0   70681616       2   \n",
       "262  ENSG00000186472.15  chr7   82731428        1.0   82731451       4   \n",
       "263   ENSG00000196335.8  chr7   23880976        3.0   23881006       2   \n",
       "264   ENSG00000196335.8  chr7   23913814        1.0   23913888       6   \n",
       "265   ENSG00000165046.8  chr8   38299881        1.0   38299913       2   \n",
       "266   ENSG00000165046.8  chr8   38304617        1.0   38304670       2   \n",
       "267  ENSG00000153317.10  chr8  131345163        1.0  131345173       1   \n",
       "268  ENSG00000153317.10  chr8  131430786        1.0  131430825       4   \n",
       "269  ENSG00000173273.11  chr8    9467805        1.0    9467819       2   \n",
       "270  ENSG00000175806.10  chr8   10307052        1.0   10307110       2   \n",
       "271   ENSG00000107282.5  chr9   72341943        1.0   72341968       2   \n",
       "272  ENSG00000136828.14  chr9  129666377        1.0  129666400       1   \n",
       "273  ENSG00000136828.14  chr9  129833393        3.0  129833410       2   \n",
       "274  ENSG00000153707.11  chr9    9595947        1.0    9595963       1   \n",
       "275   ENSG00000174482.6  chr9   28708760        1.0   28708840       4   \n",
       "\n",
       "      motif gene.name  str.len  \n",
       "0         A    AGTRAP       17  \n",
       "1         A   SDCCAG8       10  \n",
       "2        AC   ZCCHC17       48  \n",
       "3         A   ZCCHC17       17  \n",
       "4        AC     PTPRF       11  \n",
       "5         A    SEC16B       10  \n",
       "6        AC    SEC16B       54  \n",
       "7        AC    SEC16B       46  \n",
       "8         A     MKNK1       18  \n",
       "9         A     MKNK1       19  \n",
       "10        A      AGO4       27  \n",
       "11     AAGG      AGO4       78  \n",
       "12       AC      RIT1       42  \n",
       "13        A      AKT3       14  \n",
       "14       AC     CNTN2       36  \n",
       "15       AC     CNTN2       47  \n",
       "16        A     PTBP2       32  \n",
       "17       AT     PROX1       51  \n",
       "18       AC    TSPAN2       25  \n",
       "19        A   SLC45A1       19  \n",
       "20       AC      RERE       60  \n",
       "21    AAAAT      RERE       66  \n",
       "22      AAT     TADA1       29  \n",
       "23       AT     TADA1       19  \n",
       "24       AT     ILDR2       19  \n",
       "25     AAAG     ILDR2       47  \n",
       "26   CCCCGG  ADAMTSL4       55  \n",
       "27       AG     CAPN2       63  \n",
       "28        A      HAX1       20  \n",
       "29      AAT    ZBTB7B       23  \n",
       "..      ...       ...      ...  \n",
       "246    AGAT     SRPK2       72  \n",
       "247     AAT     SRPK2       42  \n",
       "248      AC     DFNA5       58  \n",
       "249      AT     DFNA5       27  \n",
       "250      AC  TMEM178B       54  \n",
       "251       A       SP4       17  \n",
       "252      AT      MPP6       27  \n",
       "253       A      MPP6       11  \n",
       "254     AAC     EXOC4       19  \n",
       "255       A     EXOC4       27  \n",
       "256      AC   ATXN7L1       55  \n",
       "257    AGAT     CALN1       76  \n",
       "258      AC    IMMP2L       37  \n",
       "259       A    IMMP2L       11  \n",
       "260      AC    IMMP2L       36  \n",
       "261      AC   WBSCR17       72  \n",
       "262    AAAT      PCLO       23  \n",
       "263      AC     STK31       30  \n",
       "264  AGAGAT     STK31       74  \n",
       "265      AC     LETM2       32  \n",
       "266      AC     LETM2       53  \n",
       "267       A     ASAP1       10  \n",
       "268    AAAT     ASAP1       39  \n",
       "269      AT      TNKS       14  \n",
       "270      AC      MSRA       58  \n",
       "271      AC     APBA1       25  \n",
       "272       A   RALGPS1       23  \n",
       "273      AT   RALGPS1       17  \n",
       "274       A     PTPRD       16  \n",
       "275    AGAT    LINGO2       80  \n",
       "\n",
       "[276 rows x 9 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Exploring STRs in GTEx\n",
    "merged_t = pd.read_csv('/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/LR_SummaryTest_Table.tsv', sep='\\t')\n",
    "summary = pd.read_csv('/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/Nerve-Tibial/Master.table', sep='\\t')\n",
    "names = summary[['gene', 'gene.name']].copy().drop_duplicates()\n",
    "print(summary.shape)\n",
    "gwas_s=genes\n",
    "print(len(gwas_s))\n",
    "gwas_genes = list(set(list(summary.loc[summary['gene.name'].isin(gwas_s)]['gene'])))\n",
    "print(len(gwas_genes))\n",
    "numb_tests = merged_t.loc[merged_t['gene'].isin(gwas_genes)]\n",
    "print('Number of tests ', numb_tests.shape[0])\n",
    "gwas_egenes = numb_tests.loc[numb_tests['E.tissues']>=1]\n",
    "print('Number of gwas eGenes for schizophrenia: ',len(set(list(gwas_egenes['gene']))))\n",
    "Finalset = gwas_egenes[['gene','chrom','str.start','E.tissues','str.end','period','motif']].merge(names)\n",
    "Finalset['str.len']=Finalset['str.end'].astype(int)-Finalset['str.start'].astype(int)\n",
    "#In gwas\n",
    "condition['in.estr']=condition['MAPPED_GENE'].apply(lambda x: 1 if (x[0] in genes) or (x[1] in genes) else 0)\n",
    "Finalset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(502, 14)\n",
      "(246, 12)\n",
      "(246, 12)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp.start</th>\n",
       "      <th>gene.name</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>gene</th>\n",
       "      <th>str.start</th>\n",
       "      <th>E.tissues</th>\n",
       "      <th>str.end</th>\n",
       "      <th>period</th>\n",
       "      <th>motif</th>\n",
       "      <th>str.len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35818353</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>rs11264213</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000134698.10</td>\n",
       "      <td>36270731</td>\n",
       "      <td>3.0</td>\n",
       "      <td>36270758</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>35818353</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>rs11264213</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000134698.10</td>\n",
       "      <td>36284227</td>\n",
       "      <td>1.0</td>\n",
       "      <td>36284305</td>\n",
       "      <td>4</td>\n",
       "      <td>AAGG</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243500591</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>rs14403</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSG00000117020.12</td>\n",
       "      <td>243861275</td>\n",
       "      <td>1.0</td>\n",
       "      <td>243861289</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243527189</td>\n",
       "      <td>AKT3</td>\n",
       "      <td>rs13376709</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000117020.12</td>\n",
       "      <td>243861275</td>\n",
       "      <td>1.0</td>\n",
       "      <td>243861289</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155906822</td>\n",
       "      <td>RIT1</td>\n",
       "      <td>rs3856261</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000143622.6</td>\n",
       "      <td>155915072</td>\n",
       "      <td>4.0</td>\n",
       "      <td>155915114</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243286579</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs3818802</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243381525</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs1538774</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243445665</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs6703335</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243391803</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs77149735</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243391803</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs77149735</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8329994</td>\n",
       "      <td>SLC45A1</td>\n",
       "      <td>rs12142465</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000162426.10</td>\n",
       "      <td>8373565</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8373584</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1</td>\n",
       "      <td>230136877</td>\n",
       "      <td>GALNT2</td>\n",
       "      <td>rs11807834</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000143641.8</td>\n",
       "      <td>230243323</td>\n",
       "      <td>1.0</td>\n",
       "      <td>230243334</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>115077013</td>\n",
       "      <td>TSPAN2</td>\n",
       "      <td>rs55675132</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000134198.5</td>\n",
       "      <td>115517955</td>\n",
       "      <td>1.0</td>\n",
       "      <td>115517980</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31359023</td>\n",
       "      <td>ZCCHC17</td>\n",
       "      <td>rs7551345</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000121766.10</td>\n",
       "      <td>31672838</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31672886</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>chr1</td>\n",
       "      <td>31359023</td>\n",
       "      <td>ZCCHC17</td>\n",
       "      <td>rs7551345</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000121766.10</td>\n",
       "      <td>31685250</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31685267</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>204630167</td>\n",
       "      <td>LRRN2</td>\n",
       "      <td>rs7551222</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000170382.7</td>\n",
       "      <td>204564282</td>\n",
       "      <td>1.0</td>\n",
       "      <td>204564326</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>chr1</td>\n",
       "      <td>223719125</td>\n",
       "      <td>CAPN2</td>\n",
       "      <td>rs7539624</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000162909.13</td>\n",
       "      <td>223894939</td>\n",
       "      <td>2.0</td>\n",
       "      <td>223895002</td>\n",
       "      <td>2</td>\n",
       "      <td>AG</td>\n",
       "      <td>63</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>chr1</td>\n",
       "      <td>213974546</td>\n",
       "      <td>PROX1</td>\n",
       "      <td>rs7529073</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000117707.11</td>\n",
       "      <td>214137474</td>\n",
       "      <td>1.0</td>\n",
       "      <td>214137525</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>chr1</td>\n",
       "      <td>43572014</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>rs3001723</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000142949.12</td>\n",
       "      <td>44025261</td>\n",
       "      <td>1.0</td>\n",
       "      <td>44025272</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>chr1</td>\n",
       "      <td>46579541</td>\n",
       "      <td>MKNK1</td>\n",
       "      <td>rs11211309</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000079277.15</td>\n",
       "      <td>47014814</td>\n",
       "      <td>1.0</td>\n",
       "      <td>47014832</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>chr1</td>\n",
       "      <td>46579541</td>\n",
       "      <td>MKNK1</td>\n",
       "      <td>rs11211309</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000079277.15</td>\n",
       "      <td>47071277</td>\n",
       "      <td>3.0</td>\n",
       "      <td>47071296</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155017821</td>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs2242194</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSG00000160685.9</td>\n",
       "      <td>155027695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>155027718</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150567705</td>\n",
       "      <td>ADAMTSL4</td>\n",
       "      <td>rs72700829</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000143382.9</td>\n",
       "      <td>150552235</td>\n",
       "      <td>1.0</td>\n",
       "      <td>150552290</td>\n",
       "      <td>6</td>\n",
       "      <td>CCCCGG</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>205066327</td>\n",
       "      <td>CNTN2</td>\n",
       "      <td>rs16937</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184144.5</td>\n",
       "      <td>205049654</td>\n",
       "      <td>1.0</td>\n",
       "      <td>205049690</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>205066327</td>\n",
       "      <td>CNTN2</td>\n",
       "      <td>rs16937</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184144.5</td>\n",
       "      <td>205094969</td>\n",
       "      <td>2.0</td>\n",
       "      <td>205095016</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8362616</td>\n",
       "      <td>RERE</td>\n",
       "      <td>rs2252865</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000142599.13</td>\n",
       "      <td>8661796</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8661856</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>60</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8362616</td>\n",
       "      <td>RERE</td>\n",
       "      <td>rs2252865</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000142599.13</td>\n",
       "      <td>8695997</td>\n",
       "      <td>3.0</td>\n",
       "      <td>8696063</td>\n",
       "      <td>5</td>\n",
       "      <td>AAAAT</td>\n",
       "      <td>66</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>chr10</td>\n",
       "      <td>18456176</td>\n",
       "      <td>CACNB2</td>\n",
       "      <td>rs7893279</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165995.14</td>\n",
       "      <td>18834646</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18834657</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr10</td>\n",
       "      <td>18456176</td>\n",
       "      <td>CACNB2</td>\n",
       "      <td>rs7893279</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165995.14</td>\n",
       "      <td>18834646</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18834657</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>chr10</td>\n",
       "      <td>18445599</td>\n",
       "      <td>CACNB2</td>\n",
       "      <td>rs17691888</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165995.14</td>\n",
       "      <td>18834646</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18834657</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>chr7</td>\n",
       "      <td>105288617</td>\n",
       "      <td>SRPK2</td>\n",
       "      <td>rs6466055</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000135250.12</td>\n",
       "      <td>104694906</td>\n",
       "      <td>1.0</td>\n",
       "      <td>104694978</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>217</th>\n",
       "      <td>chr7</td>\n",
       "      <td>105288617</td>\n",
       "      <td>SRPK2</td>\n",
       "      <td>rs6466055</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000135250.12</td>\n",
       "      <td>105009568</td>\n",
       "      <td>2.0</td>\n",
       "      <td>105009610</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>chr7</td>\n",
       "      <td>72286331</td>\n",
       "      <td>CALN1</td>\n",
       "      <td>rs12699131</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000183166.6</td>\n",
       "      <td>71845653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71845729</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>219</th>\n",
       "      <td>chr7</td>\n",
       "      <td>72374423</td>\n",
       "      <td>CALN1</td>\n",
       "      <td>rs1914395</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000183166.6</td>\n",
       "      <td>71845653</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71845729</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>chr7</td>\n",
       "      <td>105670146</td>\n",
       "      <td>ATXN7L1</td>\n",
       "      <td>rs12540205</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000146776.10</td>\n",
       "      <td>105357112</td>\n",
       "      <td>1.0</td>\n",
       "      <td>105357167</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>chr7</td>\n",
       "      <td>21480594</td>\n",
       "      <td>SP4</td>\n",
       "      <td>rs73060317</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000105866.9</td>\n",
       "      <td>21508137</td>\n",
       "      <td>1.0</td>\n",
       "      <td>21508154</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>chr7</td>\n",
       "      <td>82815056</td>\n",
       "      <td>PCLO</td>\n",
       "      <td>rs2189246</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000186472.15</td>\n",
       "      <td>82731428</td>\n",
       "      <td>1.0</td>\n",
       "      <td>82731451</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>223</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111135343</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs73201076</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>110591595</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110591632</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>224</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111135343</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs73201076</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>110602497</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110602508</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111135343</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs73201076</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>110591595</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110591632</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>227</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>110602497</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110602508</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>110591595</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110591632</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>37</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>230</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>110602497</td>\n",
       "      <td>1.0</td>\n",
       "      <td>110602508</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>232</th>\n",
       "      <td>chr7</td>\n",
       "      <td>141271019</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>rs76405237</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000261115.1</td>\n",
       "      <td>140676038</td>\n",
       "      <td>4.0</td>\n",
       "      <td>140676092</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>233</th>\n",
       "      <td>chr8</td>\n",
       "      <td>10165428</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>rs7017212</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000175806.10</td>\n",
       "      <td>10307052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10307110</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234</th>\n",
       "      <td>chr8</td>\n",
       "      <td>10175915</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>rs73191547</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000175806.10</td>\n",
       "      <td>10307052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10307110</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>chr8</td>\n",
       "      <td>10392538</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>rs12541491</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000175806.10</td>\n",
       "      <td>10307052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10307110</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>chr8</td>\n",
       "      <td>38390788</td>\n",
       "      <td>LETM2</td>\n",
       "      <td>rs112537273</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165046.8</td>\n",
       "      <td>38299881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38299913</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>chr8</td>\n",
       "      <td>38390788</td>\n",
       "      <td>LETM2</td>\n",
       "      <td>rs112537273</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165046.8</td>\n",
       "      <td>38304617</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38304670</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>238</th>\n",
       "      <td>chr8</td>\n",
       "      <td>130140419</td>\n",
       "      <td>ASAP1</td>\n",
       "      <td>rs4733770</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000153317.10</td>\n",
       "      <td>131345163</td>\n",
       "      <td>1.0</td>\n",
       "      <td>131345173</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>239</th>\n",
       "      <td>chr8</td>\n",
       "      <td>130140419</td>\n",
       "      <td>ASAP1</td>\n",
       "      <td>rs4733770</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000153317.10</td>\n",
       "      <td>131430786</td>\n",
       "      <td>1.0</td>\n",
       "      <td>131430825</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>240</th>\n",
       "      <td>chr9</td>\n",
       "      <td>29161220</td>\n",
       "      <td>LINGO2</td>\n",
       "      <td>rs10123671</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>ENSG00000174482.6</td>\n",
       "      <td>28708760</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28708840</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>241</th>\n",
       "      <td>chr9</td>\n",
       "      <td>28891819</td>\n",
       "      <td>LINGO2</td>\n",
       "      <td>rs10812882</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>ENSG00000174482.6</td>\n",
       "      <td>28708760</td>\n",
       "      <td>1.0</td>\n",
       "      <td>28708840</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>242</th>\n",
       "      <td>chr9</td>\n",
       "      <td>69472320</td>\n",
       "      <td>APBA1</td>\n",
       "      <td>rs4744901</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000107282.5</td>\n",
       "      <td>72341943</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72341968</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>243</th>\n",
       "      <td>chr9</td>\n",
       "      <td>10255540</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>rs4740446</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000153707.11</td>\n",
       "      <td>9595947</td>\n",
       "      <td>1.0</td>\n",
       "      <td>9595963</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>244</th>\n",
       "      <td>chr9</td>\n",
       "      <td>126978004</td>\n",
       "      <td>RALGPS1</td>\n",
       "      <td>rs11790388</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000136828.14</td>\n",
       "      <td>129666377</td>\n",
       "      <td>1.0</td>\n",
       "      <td>129666400</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>245</th>\n",
       "      <td>chr9</td>\n",
       "      <td>126978004</td>\n",
       "      <td>RALGPS1</td>\n",
       "      <td>rs11790388</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000136828.14</td>\n",
       "      <td>129833393</td>\n",
       "      <td>3.0</td>\n",
       "      <td>129833410</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>246 rows × 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom  snp.start gene.name         SNPS  \\\n",
       "0     chr1   35818353      AGO4   rs11264213   \n",
       "1     chr1   35818353      AGO4   rs11264213   \n",
       "2     chr1  243500591      AKT3      rs14403   \n",
       "3     chr1  243527189      AKT3   rs13376709   \n",
       "4     chr1  155906822      RIT1    rs3856261   \n",
       "5     chr1  243286579   SDCCAG8    rs3818802   \n",
       "6     chr1  243381525   SDCCAG8    rs1538774   \n",
       "7     chr1  243445665   SDCCAG8    rs6703335   \n",
       "8     chr1  243391803   SDCCAG8   rs77149735   \n",
       "9     chr1  243391803   SDCCAG8   rs77149735   \n",
       "10    chr1    8329994   SLC45A1   rs12142465   \n",
       "11    chr1  230136877    GALNT2   rs11807834   \n",
       "12    chr1  115077013    TSPAN2   rs55675132   \n",
       "13    chr1   31359023   ZCCHC17    rs7551345   \n",
       "14    chr1   31359023   ZCCHC17    rs7551345   \n",
       "15    chr1  204630167     LRRN2    rs7551222   \n",
       "16    chr1  223719125     CAPN2    rs7539624   \n",
       "17    chr1  213974546     PROX1    rs7529073   \n",
       "18    chr1   43572014     PTPRF    rs3001723   \n",
       "19    chr1   46579541     MKNK1   rs11211309   \n",
       "20    chr1   46579541     MKNK1   rs11211309   \n",
       "21    chr1  155017821    ZBTB7B    rs2242194   \n",
       "22    chr1  150567705  ADAMTSL4   rs72700829   \n",
       "23    chr1  205066327     CNTN2      rs16937   \n",
       "24    chr1  205066327     CNTN2      rs16937   \n",
       "25    chr1    8362616      RERE    rs2252865   \n",
       "26    chr1    8362616      RERE    rs2252865   \n",
       "27   chr10   18456176    CACNB2    rs7893279   \n",
       "28   chr10   18456176    CACNB2    rs7893279   \n",
       "29   chr10   18445599    CACNB2   rs17691888   \n",
       "..     ...        ...       ...          ...   \n",
       "216   chr7  105288617     SRPK2    rs6466055   \n",
       "217   chr7  105288617     SRPK2    rs6466055   \n",
       "218   chr7   72286331     CALN1   rs12699131   \n",
       "219   chr7   72374423     CALN1    rs1914395   \n",
       "220   chr7  105670146   ATXN7L1   rs12540205   \n",
       "221   chr7   21480594       SP4   rs73060317   \n",
       "222   chr7   82815056      PCLO    rs2189246   \n",
       "223   chr7  111135343    IMMP2L   rs73201076   \n",
       "224   chr7  111135343    IMMP2L   rs73201076   \n",
       "225   chr7  111135343    IMMP2L   rs73201076   \n",
       "226   chr7  111258859    IMMP2L   rs13240464   \n",
       "227   chr7  111258859    IMMP2L   rs13240464   \n",
       "228   chr7  111258859    IMMP2L   rs13240464   \n",
       "229   chr7  111258859    IMMP2L   rs13240464   \n",
       "230   chr7  111258859    IMMP2L   rs13240464   \n",
       "231   chr7  111258859    IMMP2L   rs13240464   \n",
       "232   chr7  141271019  TMEM178B   rs76405237   \n",
       "233   chr8   10165428      MSRA    rs7017212   \n",
       "234   chr8   10175915      MSRA   rs73191547   \n",
       "235   chr8   10392538      MSRA   rs12541491   \n",
       "236   chr8   38390788     LETM2  rs112537273   \n",
       "237   chr8   38390788     LETM2  rs112537273   \n",
       "238   chr8  130140419     ASAP1    rs4733770   \n",
       "239   chr8  130140419     ASAP1    rs4733770   \n",
       "240   chr9   29161220    LINGO2   rs10123671   \n",
       "241   chr9   28891819    LINGO2   rs10812882   \n",
       "242   chr9   69472320     APBA1    rs4744901   \n",
       "243   chr9   10255540     PTPRD    rs4740446   \n",
       "244   chr9  126978004   RALGPS1   rs11790388   \n",
       "245   chr9  126978004   RALGPS1   rs11790388   \n",
       "\n",
       "                                CONTEXT                gene  str.start  \\\n",
       "0                        intron_variant  ENSG00000134698.10   36270731   \n",
       "1                        intron_variant  ENSG00000134698.10   36284227   \n",
       "2                   3_prime_UTR_variant  ENSG00000117020.12  243861275   \n",
       "3                        intron_variant  ENSG00000117020.12  243861275   \n",
       "4                        intron_variant   ENSG00000143622.6  155915072   \n",
       "5                        intron_variant  ENSG00000054282.11  243394402   \n",
       "6                        intron_variant  ENSG00000054282.11  243394402   \n",
       "7                        intron_variant  ENSG00000054282.11  243394402   \n",
       "8                        intron_variant  ENSG00000054282.11  243394402   \n",
       "9                        intron_variant  ENSG00000054282.11  243394402   \n",
       "10                       intron_variant  ENSG00000162426.10    8373565   \n",
       "11                       intron_variant   ENSG00000143641.8  230243323   \n",
       "12                       intron_variant   ENSG00000134198.5  115517955   \n",
       "13                       intron_variant  ENSG00000121766.10   31672838   \n",
       "14                       intron_variant  ENSG00000121766.10   31685250   \n",
       "15                       intron_variant   ENSG00000170382.7  204564282   \n",
       "16                       intron_variant  ENSG00000162909.13  223894939   \n",
       "17                       intron_variant  ENSG00000117707.11  214137474   \n",
       "18                       intron_variant  ENSG00000142949.12   44025261   \n",
       "19                       intron_variant  ENSG00000079277.15   47014814   \n",
       "20                       intron_variant  ENSG00000079277.15   47071277   \n",
       "21                  3_prime_UTR_variant   ENSG00000160685.9  155027695   \n",
       "22                       intron_variant   ENSG00000143382.9  150552235   \n",
       "23                       intron_variant   ENSG00000184144.5  205049654   \n",
       "24                       intron_variant   ENSG00000184144.5  205094969   \n",
       "25                       intron_variant  ENSG00000142599.13    8661796   \n",
       "26                       intron_variant  ENSG00000142599.13    8695997   \n",
       "27                       intron_variant  ENSG00000165995.14   18834646   \n",
       "28                       intron_variant  ENSG00000165995.14   18834646   \n",
       "29                       intron_variant  ENSG00000165995.14   18834646   \n",
       "..                                  ...                 ...        ...   \n",
       "216                      intron_variant  ENSG00000135250.12  104694906   \n",
       "217                      intron_variant  ENSG00000135250.12  105009568   \n",
       "218                      intron_variant   ENSG00000183166.6   71845653   \n",
       "219                      intron_variant   ENSG00000183166.6   71845653   \n",
       "220                      intron_variant  ENSG00000146776.10  105357112   \n",
       "221                      intron_variant   ENSG00000105866.9   21508137   \n",
       "222                      intron_variant  ENSG00000186472.15   82731428   \n",
       "223                      intron_variant   ENSG00000184903.5  110591595   \n",
       "224                      intron_variant   ENSG00000184903.5  110602497   \n",
       "225                      intron_variant   ENSG00000184903.5  111191552   \n",
       "226                      intron_variant   ENSG00000184903.5  110591595   \n",
       "227                      intron_variant   ENSG00000184903.5  110602497   \n",
       "228                      intron_variant   ENSG00000184903.5  111191552   \n",
       "229                      intron_variant   ENSG00000184903.5  110591595   \n",
       "230                      intron_variant   ENSG00000184903.5  110602497   \n",
       "231                      intron_variant   ENSG00000184903.5  111191552   \n",
       "232                      intron_variant   ENSG00000261115.1  140676038   \n",
       "233                      intron_variant  ENSG00000175806.10   10307052   \n",
       "234                      intron_variant  ENSG00000175806.10   10307052   \n",
       "235                      intron_variant  ENSG00000175806.10   10307052   \n",
       "236                      intron_variant   ENSG00000165046.8   38299881   \n",
       "237                      intron_variant   ENSG00000165046.8   38304617   \n",
       "238                      intron_variant  ENSG00000153317.10  131345163   \n",
       "239                      intron_variant  ENSG00000153317.10  131430786   \n",
       "240                  intergenic_variant   ENSG00000174482.6   28708760   \n",
       "241               upstream_gene_variant   ENSG00000174482.6   28708760   \n",
       "242                      intron_variant   ENSG00000107282.5   72341943   \n",
       "243                      intron_variant  ENSG00000153707.11    9595947   \n",
       "244  non_coding_transcript_exon_variant  ENSG00000136828.14  129666377   \n",
       "245  non_coding_transcript_exon_variant  ENSG00000136828.14  129833393   \n",
       "\n",
       "     E.tissues    str.end  period   motif  str.len  \n",
       "0          3.0   36270758       1       A       27  \n",
       "1          1.0   36284305       4    AAGG       78  \n",
       "2          1.0  243861289       1       A       14  \n",
       "3          1.0  243861289       1       A       14  \n",
       "4          4.0  155915114       2      AC       42  \n",
       "5         12.0  243394412       1       A       10  \n",
       "6         12.0  243394412       1       A       10  \n",
       "7         12.0  243394412       1       A       10  \n",
       "8         12.0  243394412       1       A       10  \n",
       "9         12.0  243394412       1       A       10  \n",
       "10         1.0    8373584       1       A       19  \n",
       "11         1.0  230243334       1       A       11  \n",
       "12         1.0  115517980       2      AC       25  \n",
       "13         1.0   31672886       2      AC       48  \n",
       "14         1.0   31685267       1       A       17  \n",
       "15         1.0  204564326       3     AAT       44  \n",
       "16         2.0  223895002       2      AG       63  \n",
       "17         1.0  214137525       2      AT       51  \n",
       "18         1.0   44025272       2      AC       11  \n",
       "19         1.0   47014832       1       A       18  \n",
       "20         3.0   47071296       1       A       19  \n",
       "21         1.0  155027718       3     AAT       23  \n",
       "22         1.0  150552290       6  CCCCGG       55  \n",
       "23         1.0  205049690       2      AC       36  \n",
       "24         2.0  205095016       2      AC       47  \n",
       "25         2.0    8661856       2      AC       60  \n",
       "26         3.0    8696063       5   AAAAT       66  \n",
       "27         1.0   18834657       1       A       11  \n",
       "28         1.0   18834657       1       A       11  \n",
       "29         1.0   18834657       1       A       11  \n",
       "..         ...        ...     ...     ...      ...  \n",
       "216        1.0  104694978       4    AGAT       72  \n",
       "217        2.0  105009610       3     AAT       42  \n",
       "218        1.0   71845729       4    AGAT       76  \n",
       "219        1.0   71845729       4    AGAT       76  \n",
       "220        1.0  105357167       2      AC       55  \n",
       "221        1.0   21508154       1       A       17  \n",
       "222        1.0   82731451       4    AAAT       23  \n",
       "223        1.0  110591632       2      AC       37  \n",
       "224        1.0  110602508       1       A       11  \n",
       "225        1.0  111191588       2      AC       36  \n",
       "226        1.0  110591632       2      AC       37  \n",
       "227        1.0  110602508       1       A       11  \n",
       "228        1.0  111191588       2      AC       36  \n",
       "229        1.0  110591632       2      AC       37  \n",
       "230        1.0  110602508       1       A       11  \n",
       "231        1.0  111191588       2      AC       36  \n",
       "232        4.0  140676092       2      AC       54  \n",
       "233        1.0   10307110       2      AC       58  \n",
       "234        1.0   10307110       2      AC       58  \n",
       "235        1.0   10307110       2      AC       58  \n",
       "236        1.0   38299913       2      AC       32  \n",
       "237        1.0   38304670       2      AC       53  \n",
       "238        1.0  131345173       1       A       10  \n",
       "239        1.0  131430825       4    AAAT       39  \n",
       "240        1.0   28708840       4    AGAT       80  \n",
       "241        1.0   28708840       4    AGAT       80  \n",
       "242        1.0   72341968       2      AC       25  \n",
       "243        1.0    9595963       1       A       16  \n",
       "244        1.0  129666400       1       A       23  \n",
       "245        3.0  129833410       2      AT       17  \n",
       "\n",
       "[246 rows x 12 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "newselct =condition.loc[condition['in.estr']==1]\n",
    "print(newselct.shape)\n",
    "newselct = newselct.dropna(subset=['CHR_ID'])\n",
    "newselct['chrom']='chr'+newselct['CHR_ID'].astype(str)\n",
    "newselct['snp.start']=newselct['CHR_POS'].astype(int)\n",
    "newselct['gene.name']=newselct['MAPPED_GENE'].apply(lambda x: str(x[0]))\n",
    "data = Finalset.merge(newselct[['chrom', 'snp.start','gene.name','SNPS', \"CONTEXT\"]],on=['chrom','gene.name'])\n",
    "print(data.shape)\n",
    "data = newselct[['chrom', 'snp.start','gene.name','SNPS', \"CONTEXT\"]].merge(Finalset, on=['chrom','gene.name'])\n",
    "print(data.shape)\n",
    "data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(246, 14)\n",
      "There are 134 eGenes with associates eSNPs for schizophrenia \n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>snp.start</th>\n",
       "      <th>gene.name</th>\n",
       "      <th>SNPS</th>\n",
       "      <th>CONTEXT</th>\n",
       "      <th>gene</th>\n",
       "      <th>str.start</th>\n",
       "      <th>E.tissues</th>\n",
       "      <th>str.end</th>\n",
       "      <th>period</th>\n",
       "      <th>motif</th>\n",
       "      <th>str.len</th>\n",
       "      <th>distance</th>\n",
       "      <th>within_100k</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155906822</td>\n",
       "      <td>RIT1</td>\n",
       "      <td>rs3856261</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000143622.6</td>\n",
       "      <td>155915072</td>\n",
       "      <td>4.0</td>\n",
       "      <td>155915114</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>42</td>\n",
       "      <td>8250</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243381525</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs1538774</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>12877</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243445665</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs6703335</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>-51263</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243391803</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs77149735</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>2599</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1</td>\n",
       "      <td>243391803</td>\n",
       "      <td>SDCCAG8</td>\n",
       "      <td>rs77149735</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000054282.11</td>\n",
       "      <td>243394402</td>\n",
       "      <td>12.0</td>\n",
       "      <td>243394412</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>2599</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr1</td>\n",
       "      <td>8329994</td>\n",
       "      <td>SLC45A1</td>\n",
       "      <td>rs12142465</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000162426.10</td>\n",
       "      <td>8373565</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8373584</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>19</td>\n",
       "      <td>43571</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>chr1</td>\n",
       "      <td>204630167</td>\n",
       "      <td>LRRN2</td>\n",
       "      <td>rs7551222</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000170382.7</td>\n",
       "      <td>204564282</td>\n",
       "      <td>1.0</td>\n",
       "      <td>204564326</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>44</td>\n",
       "      <td>-65885</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>chr1</td>\n",
       "      <td>155017821</td>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>rs2242194</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSG00000160685.9</td>\n",
       "      <td>155027695</td>\n",
       "      <td>1.0</td>\n",
       "      <td>155027718</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>23</td>\n",
       "      <td>9874</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150567705</td>\n",
       "      <td>ADAMTSL4</td>\n",
       "      <td>rs72700829</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000143382.9</td>\n",
       "      <td>150552235</td>\n",
       "      <td>1.0</td>\n",
       "      <td>150552290</td>\n",
       "      <td>6</td>\n",
       "      <td>CCCCGG</td>\n",
       "      <td>55</td>\n",
       "      <td>-15470</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>chr1</td>\n",
       "      <td>205066327</td>\n",
       "      <td>CNTN2</td>\n",
       "      <td>rs16937</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184144.5</td>\n",
       "      <td>205049654</td>\n",
       "      <td>1.0</td>\n",
       "      <td>205049690</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "      <td>-16673</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1</td>\n",
       "      <td>205066327</td>\n",
       "      <td>CNTN2</td>\n",
       "      <td>rs16937</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184144.5</td>\n",
       "      <td>205094969</td>\n",
       "      <td>2.0</td>\n",
       "      <td>205095016</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>47</td>\n",
       "      <td>28642</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>chr10</td>\n",
       "      <td>3779369</td>\n",
       "      <td>KLF6</td>\n",
       "      <td>rs17731</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSG00000067082.10</td>\n",
       "      <td>3809510</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3809548</td>\n",
       "      <td>2</td>\n",
       "      <td>AG</td>\n",
       "      <td>38</td>\n",
       "      <td>30141</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17163076</td>\n",
       "      <td>PIK3C2A</td>\n",
       "      <td>rs2008905</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000011405.9</td>\n",
       "      <td>17086239</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17086310</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>71</td>\n",
       "      <td>-76837</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17163076</td>\n",
       "      <td>PIK3C2A</td>\n",
       "      <td>rs2008905</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000011405.9</td>\n",
       "      <td>17197998</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17198018</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>20</td>\n",
       "      <td>34922</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17138601</td>\n",
       "      <td>PIK3C2A</td>\n",
       "      <td>rs4356203</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000011405.9</td>\n",
       "      <td>17086239</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17086310</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>71</td>\n",
       "      <td>-52362</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>chr11</td>\n",
       "      <td>17138601</td>\n",
       "      <td>PIK3C2A</td>\n",
       "      <td>rs4356203</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000011405.9</td>\n",
       "      <td>17197998</td>\n",
       "      <td>1.0</td>\n",
       "      <td>17198018</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>20</td>\n",
       "      <td>59397</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>chr11</td>\n",
       "      <td>8877853</td>\n",
       "      <td>ST5</td>\n",
       "      <td>rs2455601</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000166444.13</td>\n",
       "      <td>8839810</td>\n",
       "      <td>2.0</td>\n",
       "      <td>8839832</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>22</td>\n",
       "      <td>-38043</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>chr11</td>\n",
       "      <td>134111180</td>\n",
       "      <td>JAM3</td>\n",
       "      <td>rs1267813</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000166086.8</td>\n",
       "      <td>134082270</td>\n",
       "      <td>2.0</td>\n",
       "      <td>134082304</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>34</td>\n",
       "      <td>-28910</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>chr12</td>\n",
       "      <td>29719385</td>\n",
       "      <td>TMTC1</td>\n",
       "      <td>rs16934812</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000133687.11</td>\n",
       "      <td>29645262</td>\n",
       "      <td>1.0</td>\n",
       "      <td>29645273</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>-74123</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>chr16</td>\n",
       "      <td>71362758</td>\n",
       "      <td>CALB2</td>\n",
       "      <td>rs62055045</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000172137.14</td>\n",
       "      <td>71335226</td>\n",
       "      <td>1.0</td>\n",
       "      <td>71335284</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>58</td>\n",
       "      <td>-27532</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>chr16</td>\n",
       "      <td>89806562</td>\n",
       "      <td>FANCA</td>\n",
       "      <td>rs12449000</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000187741.10</td>\n",
       "      <td>89856806</td>\n",
       "      <td>5.0</td>\n",
       "      <td>89856849</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>43</td>\n",
       "      <td>50244</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>chr16</td>\n",
       "      <td>68155437</td>\n",
       "      <td>NFATC3</td>\n",
       "      <td>rs8044995</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000072736.14</td>\n",
       "      <td>68202375</td>\n",
       "      <td>1.0</td>\n",
       "      <td>68202403</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>28</td>\n",
       "      <td>46938</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>chr16</td>\n",
       "      <td>68155437</td>\n",
       "      <td>NFATC3</td>\n",
       "      <td>rs8044995</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000072736.14</td>\n",
       "      <td>68202375</td>\n",
       "      <td>1.0</td>\n",
       "      <td>68202403</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>28</td>\n",
       "      <td>46938</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>chr16</td>\n",
       "      <td>89543010</td>\n",
       "      <td>SPG7</td>\n",
       "      <td>rs34753377</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000197912.9</td>\n",
       "      <td>89528428</td>\n",
       "      <td>2.0</td>\n",
       "      <td>89528511</td>\n",
       "      <td>6</td>\n",
       "      <td>AGAGGG</td>\n",
       "      <td>83</td>\n",
       "      <td>-14582</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>chr16</td>\n",
       "      <td>74533177</td>\n",
       "      <td>GLG1</td>\n",
       "      <td>rs71391092</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000090863.7</td>\n",
       "      <td>74624846</td>\n",
       "      <td>1.0</td>\n",
       "      <td>74624859</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>13</td>\n",
       "      <td>91669</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>chr16</td>\n",
       "      <td>4447450</td>\n",
       "      <td>DNAJA3</td>\n",
       "      <td>rs6500602</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000103423.9</td>\n",
       "      <td>4533162</td>\n",
       "      <td>3.0</td>\n",
       "      <td>4533225</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>63</td>\n",
       "      <td>85712</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>chr17</td>\n",
       "      <td>2305605</td>\n",
       "      <td>SRR</td>\n",
       "      <td>rs4523957</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000167720.8</td>\n",
       "      <td>2282198</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2282280</td>\n",
       "      <td>6</td>\n",
       "      <td>AGCCTC</td>\n",
       "      <td>82</td>\n",
       "      <td>-23407</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>chr17</td>\n",
       "      <td>2305605</td>\n",
       "      <td>SRR</td>\n",
       "      <td>rs4523957</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000167720.8</td>\n",
       "      <td>2282198</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2282280</td>\n",
       "      <td>6</td>\n",
       "      <td>AGCCTC</td>\n",
       "      <td>82</td>\n",
       "      <td>-23407</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>chr17</td>\n",
       "      <td>2305605</td>\n",
       "      <td>SRR</td>\n",
       "      <td>rs4523957</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000167720.8</td>\n",
       "      <td>2282198</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2282280</td>\n",
       "      <td>6</td>\n",
       "      <td>AGCCTC</td>\n",
       "      <td>82</td>\n",
       "      <td>-23407</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>chr17</td>\n",
       "      <td>19250104</td>\n",
       "      <td>EPN2</td>\n",
       "      <td>rs4273100</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000072134.11</td>\n",
       "      <td>19310367</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19310411</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>44</td>\n",
       "      <td>60263</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>157</th>\n",
       "      <td>chr22</td>\n",
       "      <td>19960814</td>\n",
       "      <td>COMT</td>\n",
       "      <td>rs4646312</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000093010.7</td>\n",
       "      <td>19903852</td>\n",
       "      <td>1.0</td>\n",
       "      <td>19903868</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>16</td>\n",
       "      <td>-56962</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>161</th>\n",
       "      <td>chr22</td>\n",
       "      <td>20146827</td>\n",
       "      <td>ZDHHC8</td>\n",
       "      <td>rs1974653</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSG00000099904.11</td>\n",
       "      <td>20063280</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20063291</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>11</td>\n",
       "      <td>-83547</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>chr22</td>\n",
       "      <td>20146827</td>\n",
       "      <td>ZDHHC8</td>\n",
       "      <td>rs1974653</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>ENSG00000099904.11</td>\n",
       "      <td>20070890</td>\n",
       "      <td>1.0</td>\n",
       "      <td>20070904</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>14</td>\n",
       "      <td>-75937</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>169</th>\n",
       "      <td>chr3</td>\n",
       "      <td>2506102</td>\n",
       "      <td>CNTN4</td>\n",
       "      <td>rs17194490</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000144619.10</td>\n",
       "      <td>2492052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2492087</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>35</td>\n",
       "      <td>-14050</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>171</th>\n",
       "      <td>chr3</td>\n",
       "      <td>2506102</td>\n",
       "      <td>CNTN4</td>\n",
       "      <td>rs17194490</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000144619.10</td>\n",
       "      <td>2492052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2492087</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>35</td>\n",
       "      <td>-14050</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>177</th>\n",
       "      <td>chr3</td>\n",
       "      <td>124364323</td>\n",
       "      <td>KALRN</td>\n",
       "      <td>rs504918</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000160145.11</td>\n",
       "      <td>124393851</td>\n",
       "      <td>1.0</td>\n",
       "      <td>124393870</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>19</td>\n",
       "      <td>29528</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>chr3</td>\n",
       "      <td>124364323</td>\n",
       "      <td>KALRN</td>\n",
       "      <td>rs504918</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000160145.11</td>\n",
       "      <td>124299795</td>\n",
       "      <td>1.0</td>\n",
       "      <td>124299820</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>25</td>\n",
       "      <td>-64528</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>184</th>\n",
       "      <td>chr3</td>\n",
       "      <td>128174008</td>\n",
       "      <td>EEFSEC</td>\n",
       "      <td>rs2811474</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000132394.6</td>\n",
       "      <td>128087279</td>\n",
       "      <td>1.0</td>\n",
       "      <td>128087308</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>29</td>\n",
       "      <td>-86729</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>185</th>\n",
       "      <td>chr3</td>\n",
       "      <td>52359426</td>\n",
       "      <td>DNAH1</td>\n",
       "      <td>rs2306864</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000114841.13</td>\n",
       "      <td>52374425</td>\n",
       "      <td>1.0</td>\n",
       "      <td>52374465</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>40</td>\n",
       "      <td>14999</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>188</th>\n",
       "      <td>chr4</td>\n",
       "      <td>2393570</td>\n",
       "      <td>ZFYVE28</td>\n",
       "      <td>rs959770</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000159733.9</td>\n",
       "      <td>2448084</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2448136</td>\n",
       "      <td>3</td>\n",
       "      <td>AAT</td>\n",
       "      <td>52</td>\n",
       "      <td>54514</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>197</th>\n",
       "      <td>chr6</td>\n",
       "      <td>28259826</td>\n",
       "      <td>NKAPL</td>\n",
       "      <td>rs1635</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSG00000189134.3</td>\n",
       "      <td>28168887</td>\n",
       "      <td>2.0</td>\n",
       "      <td>28168972</td>\n",
       "      <td>4</td>\n",
       "      <td>AGAT</td>\n",
       "      <td>85</td>\n",
       "      <td>-90939</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32203298</td>\n",
       "      <td>NOTCH4</td>\n",
       "      <td>rs3132935</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000204301.5</td>\n",
       "      <td>32182244</td>\n",
       "      <td>1.0</td>\n",
       "      <td>32182262</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>18</td>\n",
       "      <td>-21054</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32203298</td>\n",
       "      <td>NOTCH4</td>\n",
       "      <td>rs3132935</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000204301.5</td>\n",
       "      <td>32271092</td>\n",
       "      <td>2.0</td>\n",
       "      <td>32271102</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>67794</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32203298</td>\n",
       "      <td>NOTCH4</td>\n",
       "      <td>rs3132935</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000204301.5</td>\n",
       "      <td>32182244</td>\n",
       "      <td>1.0</td>\n",
       "      <td>32182262</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>18</td>\n",
       "      <td>-21054</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32203298</td>\n",
       "      <td>NOTCH4</td>\n",
       "      <td>rs3132935</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>ENSG00000204301.5</td>\n",
       "      <td>32271092</td>\n",
       "      <td>2.0</td>\n",
       "      <td>32271102</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>67794</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32205216</td>\n",
       "      <td>NOTCH4</td>\n",
       "      <td>rs3131296</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000204301.5</td>\n",
       "      <td>32182244</td>\n",
       "      <td>1.0</td>\n",
       "      <td>32182262</td>\n",
       "      <td>2</td>\n",
       "      <td>AT</td>\n",
       "      <td>18</td>\n",
       "      <td>-22972</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>chr6</td>\n",
       "      <td>32205216</td>\n",
       "      <td>NOTCH4</td>\n",
       "      <td>rs3131296</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000204301.5</td>\n",
       "      <td>32271092</td>\n",
       "      <td>2.0</td>\n",
       "      <td>32271102</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>10</td>\n",
       "      <td>65876</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>208</th>\n",
       "      <td>chr6</td>\n",
       "      <td>33773939</td>\n",
       "      <td>LEMD2</td>\n",
       "      <td>rs4711350</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000161904.7</td>\n",
       "      <td>33801317</td>\n",
       "      <td>1.0</td>\n",
       "      <td>33801379</td>\n",
       "      <td>6</td>\n",
       "      <td>AGAGGG</td>\n",
       "      <td>62</td>\n",
       "      <td>27378</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>chr6</td>\n",
       "      <td>30197496</td>\n",
       "      <td>TRIM26</td>\n",
       "      <td>rs2523722</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000234127.4</td>\n",
       "      <td>30189240</td>\n",
       "      <td>5.0</td>\n",
       "      <td>30189277</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>37</td>\n",
       "      <td>-8256</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>210</th>\n",
       "      <td>chr6</td>\n",
       "      <td>30206354</td>\n",
       "      <td>TRIM26</td>\n",
       "      <td>rs2021722</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000234127.4</td>\n",
       "      <td>30189240</td>\n",
       "      <td>5.0</td>\n",
       "      <td>30189277</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>37</td>\n",
       "      <td>-17114</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>212</th>\n",
       "      <td>chr6</td>\n",
       "      <td>28338894</td>\n",
       "      <td>ZSCAN31</td>\n",
       "      <td>rs13217619</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000235109.3</td>\n",
       "      <td>28280272</td>\n",
       "      <td>9.0</td>\n",
       "      <td>28280298</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>26</td>\n",
       "      <td>-58622</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>213</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31635190</td>\n",
       "      <td>PRRC2A</td>\n",
       "      <td>rs1046089</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>ENSG00000204469.8</td>\n",
       "      <td>31577420</td>\n",
       "      <td>1.0</td>\n",
       "      <td>31577462</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>42</td>\n",
       "      <td>-57770</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>221</th>\n",
       "      <td>chr7</td>\n",
       "      <td>21480594</td>\n",
       "      <td>SP4</td>\n",
       "      <td>rs73060317</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000105866.9</td>\n",
       "      <td>21508137</td>\n",
       "      <td>1.0</td>\n",
       "      <td>21508154</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>17</td>\n",
       "      <td>27543</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>222</th>\n",
       "      <td>chr7</td>\n",
       "      <td>82815056</td>\n",
       "      <td>PCLO</td>\n",
       "      <td>rs2189246</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000186472.15</td>\n",
       "      <td>82731428</td>\n",
       "      <td>1.0</td>\n",
       "      <td>82731451</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>23</td>\n",
       "      <td>-83628</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>225</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111135343</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs73201076</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "      <td>56209</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>228</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "      <td>-67307</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>231</th>\n",
       "      <td>chr7</td>\n",
       "      <td>111258859</td>\n",
       "      <td>IMMP2L</td>\n",
       "      <td>rs13240464</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000184903.5</td>\n",
       "      <td>111191552</td>\n",
       "      <td>1.0</td>\n",
       "      <td>111191588</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>36</td>\n",
       "      <td>-67307</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>235</th>\n",
       "      <td>chr8</td>\n",
       "      <td>10392538</td>\n",
       "      <td>MSRA</td>\n",
       "      <td>rs12541491</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000175806.10</td>\n",
       "      <td>10307052</td>\n",
       "      <td>1.0</td>\n",
       "      <td>10307110</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>58</td>\n",
       "      <td>-85486</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>236</th>\n",
       "      <td>chr8</td>\n",
       "      <td>38390788</td>\n",
       "      <td>LETM2</td>\n",
       "      <td>rs112537273</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165046.8</td>\n",
       "      <td>38299881</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38299913</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>32</td>\n",
       "      <td>-90907</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>237</th>\n",
       "      <td>chr8</td>\n",
       "      <td>38390788</td>\n",
       "      <td>LETM2</td>\n",
       "      <td>rs112537273</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>ENSG00000165046.8</td>\n",
       "      <td>38304617</td>\n",
       "      <td>1.0</td>\n",
       "      <td>38304670</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>53</td>\n",
       "      <td>-86171</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>62 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chrom  snp.start gene.name         SNPS  \\\n",
       "4     chr1  155906822      RIT1    rs3856261   \n",
       "6     chr1  243381525   SDCCAG8    rs1538774   \n",
       "7     chr1  243445665   SDCCAG8    rs6703335   \n",
       "8     chr1  243391803   SDCCAG8   rs77149735   \n",
       "9     chr1  243391803   SDCCAG8   rs77149735   \n",
       "10    chr1    8329994   SLC45A1   rs12142465   \n",
       "15    chr1  204630167     LRRN2    rs7551222   \n",
       "21    chr1  155017821    ZBTB7B    rs2242194   \n",
       "22    chr1  150567705  ADAMTSL4   rs72700829   \n",
       "23    chr1  205066327     CNTN2      rs16937   \n",
       "24    chr1  205066327     CNTN2      rs16937   \n",
       "47   chr10    3779369      KLF6      rs17731   \n",
       "49   chr11   17163076   PIK3C2A    rs2008905   \n",
       "50   chr11   17163076   PIK3C2A    rs2008905   \n",
       "51   chr11   17138601   PIK3C2A    rs4356203   \n",
       "52   chr11   17138601   PIK3C2A    rs4356203   \n",
       "57   chr11    8877853       ST5    rs2455601   \n",
       "59   chr11  134111180      JAM3    rs1267813   \n",
       "67   chr12   29719385     TMTC1   rs16934812   \n",
       "93   chr16   71362758     CALB2   rs62055045   \n",
       "96   chr16   89806562     FANCA   rs12449000   \n",
       "97   chr16   68155437    NFATC3    rs8044995   \n",
       "98   chr16   68155437    NFATC3    rs8044995   \n",
       "102  chr16   89543010      SPG7   rs34753377   \n",
       "105  chr16   74533177      GLG1   rs71391092   \n",
       "107  chr16    4447450    DNAJA3    rs6500602   \n",
       "109  chr17    2305605       SRR    rs4523957   \n",
       "111  chr17    2305605       SRR    rs4523957   \n",
       "113  chr17    2305605       SRR    rs4523957   \n",
       "114  chr17   19250104      EPN2    rs4273100   \n",
       "..     ...        ...       ...          ...   \n",
       "157  chr22   19960814      COMT    rs4646312   \n",
       "161  chr22   20146827    ZDHHC8    rs1974653   \n",
       "162  chr22   20146827    ZDHHC8    rs1974653   \n",
       "169   chr3    2506102     CNTN4   rs17194490   \n",
       "171   chr3    2506102     CNTN4   rs17194490   \n",
       "177   chr3  124364323     KALRN     rs504918   \n",
       "178   chr3  124364323     KALRN     rs504918   \n",
       "184   chr3  128174008    EEFSEC    rs2811474   \n",
       "185   chr3   52359426     DNAH1    rs2306864   \n",
       "188   chr4    2393570   ZFYVE28     rs959770   \n",
       "197   chr6   28259826     NKAPL       rs1635   \n",
       "199   chr6   32203298    NOTCH4    rs3132935   \n",
       "200   chr6   32203298    NOTCH4    rs3132935   \n",
       "201   chr6   32203298    NOTCH4    rs3132935   \n",
       "202   chr6   32203298    NOTCH4    rs3132935   \n",
       "203   chr6   32205216    NOTCH4    rs3131296   \n",
       "204   chr6   32205216    NOTCH4    rs3131296   \n",
       "208   chr6   33773939     LEMD2    rs4711350   \n",
       "209   chr6   30197496    TRIM26    rs2523722   \n",
       "210   chr6   30206354    TRIM26    rs2021722   \n",
       "212   chr6   28338894   ZSCAN31   rs13217619   \n",
       "213   chr6   31635190    PRRC2A    rs1046089   \n",
       "221   chr7   21480594       SP4   rs73060317   \n",
       "222   chr7   82815056      PCLO    rs2189246   \n",
       "225   chr7  111135343    IMMP2L   rs73201076   \n",
       "228   chr7  111258859    IMMP2L   rs13240464   \n",
       "231   chr7  111258859    IMMP2L   rs13240464   \n",
       "235   chr8   10392538      MSRA   rs12541491   \n",
       "236   chr8   38390788     LETM2  rs112537273   \n",
       "237   chr8   38390788     LETM2  rs112537273   \n",
       "\n",
       "                                CONTEXT                gene  str.start  \\\n",
       "4                        intron_variant   ENSG00000143622.6  155915072   \n",
       "6                        intron_variant  ENSG00000054282.11  243394402   \n",
       "7                        intron_variant  ENSG00000054282.11  243394402   \n",
       "8                        intron_variant  ENSG00000054282.11  243394402   \n",
       "9                        intron_variant  ENSG00000054282.11  243394402   \n",
       "10                       intron_variant  ENSG00000162426.10    8373565   \n",
       "15                       intron_variant   ENSG00000170382.7  204564282   \n",
       "21                  3_prime_UTR_variant   ENSG00000160685.9  155027695   \n",
       "22                       intron_variant   ENSG00000143382.9  150552235   \n",
       "23                       intron_variant   ENSG00000184144.5  205049654   \n",
       "24                       intron_variant   ENSG00000184144.5  205094969   \n",
       "47                  3_prime_UTR_variant  ENSG00000067082.10    3809510   \n",
       "49                       intron_variant   ENSG00000011405.9   17086239   \n",
       "50                       intron_variant   ENSG00000011405.9   17197998   \n",
       "51                       intron_variant   ENSG00000011405.9   17086239   \n",
       "52                       intron_variant   ENSG00000011405.9   17197998   \n",
       "57                       intron_variant  ENSG00000166444.13    8839810   \n",
       "59                       intron_variant   ENSG00000166086.8  134082270   \n",
       "67                       intron_variant  ENSG00000133687.11   29645262   \n",
       "93                       intron_variant  ENSG00000172137.14   71335226   \n",
       "96                       intron_variant  ENSG00000187741.10   89856806   \n",
       "97                       intron_variant  ENSG00000072736.14   68202375   \n",
       "98                       intron_variant  ENSG00000072736.14   68202375   \n",
       "102                      intron_variant   ENSG00000197912.9   89528428   \n",
       "105                      intron_variant   ENSG00000090863.7   74624846   \n",
       "107  non_coding_transcript_exon_variant   ENSG00000103423.9    4533162   \n",
       "109                      intron_variant   ENSG00000167720.8    2282198   \n",
       "111                      intron_variant   ENSG00000167720.8    2282198   \n",
       "113                      intron_variant   ENSG00000167720.8    2282198   \n",
       "114                      intron_variant  ENSG00000072134.11   19310367   \n",
       "..                                  ...                 ...        ...   \n",
       "157                      intron_variant   ENSG00000093010.7   19903852   \n",
       "161                 3_prime_UTR_variant  ENSG00000099904.11   20063280   \n",
       "162                 3_prime_UTR_variant  ENSG00000099904.11   20070890   \n",
       "169                      intron_variant  ENSG00000144619.10    2492052   \n",
       "171                      intron_variant  ENSG00000144619.10    2492052   \n",
       "177                      intron_variant  ENSG00000160145.11  124393851   \n",
       "178                      intron_variant  ENSG00000160145.11  124299795   \n",
       "184                      intron_variant   ENSG00000132394.6  128087279   \n",
       "185                      intron_variant  ENSG00000114841.13   52374425   \n",
       "188                      intron_variant   ENSG00000159733.9    2448084   \n",
       "197                    missense_variant   ENSG00000189134.3   28168887   \n",
       "199  non_coding_transcript_exon_variant   ENSG00000204301.5   32182244   \n",
       "200  non_coding_transcript_exon_variant   ENSG00000204301.5   32271092   \n",
       "201  non_coding_transcript_exon_variant   ENSG00000204301.5   32182244   \n",
       "202  non_coding_transcript_exon_variant   ENSG00000204301.5   32271092   \n",
       "203                      intron_variant   ENSG00000204301.5   32182244   \n",
       "204                      intron_variant   ENSG00000204301.5   32271092   \n",
       "208                      intron_variant   ENSG00000161904.7   33801317   \n",
       "209                      intron_variant   ENSG00000234127.4   30189240   \n",
       "210                      intron_variant   ENSG00000234127.4   30189240   \n",
       "212                      intron_variant   ENSG00000235109.3   28280272   \n",
       "213                    missense_variant   ENSG00000204469.8   31577420   \n",
       "221                      intron_variant   ENSG00000105866.9   21508137   \n",
       "222                      intron_variant  ENSG00000186472.15   82731428   \n",
       "225                      intron_variant   ENSG00000184903.5  111191552   \n",
       "228                      intron_variant   ENSG00000184903.5  111191552   \n",
       "231                      intron_variant   ENSG00000184903.5  111191552   \n",
       "235                      intron_variant  ENSG00000175806.10   10307052   \n",
       "236                      intron_variant   ENSG00000165046.8   38299881   \n",
       "237                      intron_variant   ENSG00000165046.8   38304617   \n",
       "\n",
       "     E.tissues    str.end  period   motif  str.len  distance  within_100k  \n",
       "4          4.0  155915114       2      AC       42      8250            1  \n",
       "6         12.0  243394412       1       A       10     12877            1  \n",
       "7         12.0  243394412       1       A       10    -51263            1  \n",
       "8         12.0  243394412       1       A       10      2599            1  \n",
       "9         12.0  243394412       1       A       10      2599            1  \n",
       "10         1.0    8373584       1       A       19     43571            1  \n",
       "15         1.0  204564326       3     AAT       44    -65885            1  \n",
       "21         1.0  155027718       3     AAT       23      9874            1  \n",
       "22         1.0  150552290       6  CCCCGG       55    -15470            1  \n",
       "23         1.0  205049690       2      AC       36    -16673            1  \n",
       "24         2.0  205095016       2      AC       47     28642            1  \n",
       "47         1.0    3809548       2      AG       38     30141            1  \n",
       "49         1.0   17086310       4    AGAT       71    -76837            1  \n",
       "50         1.0   17198018       4    AAAT       20     34922            1  \n",
       "51         1.0   17086310       4    AGAT       71    -52362            1  \n",
       "52         1.0   17198018       4    AAAT       20     59397            1  \n",
       "57         2.0    8839832       3     AAT       22    -38043            1  \n",
       "59         2.0  134082304       2      AC       34    -28910            1  \n",
       "67         1.0   29645273       1       A       11    -74123            1  \n",
       "93         1.0   71335284       4    AGAT       58    -27532            1  \n",
       "96         5.0   89856849       4    AAAT       43     50244            1  \n",
       "97         1.0   68202403       3     AAT       28     46938            1  \n",
       "98         1.0   68202403       3     AAT       28     46938            1  \n",
       "102        2.0   89528511       6  AGAGGG       83    -14582            1  \n",
       "105        1.0   74624859       1       A       13     91669            1  \n",
       "107        3.0    4533225       2      AC       63     85712            1  \n",
       "109        1.0    2282280       6  AGCCTC       82    -23407            1  \n",
       "111        1.0    2282280       6  AGCCTC       82    -23407            1  \n",
       "113        1.0    2282280       6  AGCCTC       82    -23407            1  \n",
       "114        1.0   19310411       2      AC       44     60263            1  \n",
       "..         ...        ...     ...     ...      ...       ...          ...  \n",
       "157        1.0   19903868       1       A       16    -56962            1  \n",
       "161        1.0   20063291       1       A       11    -83547            1  \n",
       "162        1.0   20070904       2      AC       14    -75937            1  \n",
       "169        1.0    2492087       4    AAAT       35    -14050            1  \n",
       "171        1.0    2492087       4    AAAT       35    -14050            1  \n",
       "177        1.0  124393870       2      AT       19     29528            1  \n",
       "178        1.0  124299820       3     AAT       25    -64528            1  \n",
       "184        1.0  128087308       1       A       29    -86729            1  \n",
       "185        1.0   52374465       4    AAAT       40     14999            1  \n",
       "188        1.0    2448136       3     AAT       52     54514            1  \n",
       "197        2.0   28168972       4    AGAT       85    -90939            1  \n",
       "199        1.0   32182262       2      AT       18    -21054            1  \n",
       "200        2.0   32271102       1       A       10     67794            1  \n",
       "201        1.0   32182262       2      AT       18    -21054            1  \n",
       "202        2.0   32271102       1       A       10     67794            1  \n",
       "203        1.0   32182262       2      AT       18    -22972            1  \n",
       "204        2.0   32271102       1       A       10     65876            1  \n",
       "208        1.0   33801379       6  AGAGGG       62     27378            1  \n",
       "209        5.0   30189277       4    AAAT       37     -8256            1  \n",
       "210        5.0   30189277       4    AAAT       37    -17114            1  \n",
       "212        9.0   28280298       2      AC       26    -58622            1  \n",
       "213        1.0   31577462       2      AC       42    -57770            1  \n",
       "221        1.0   21508154       1       A       17     27543            1  \n",
       "222        1.0   82731451       4    AAAT       23    -83628            1  \n",
       "225        1.0  111191588       2      AC       36     56209            1  \n",
       "228        1.0  111191588       2      AC       36    -67307            1  \n",
       "231        1.0  111191588       2      AC       36    -67307            1  \n",
       "235        1.0   10307110       2      AC       58    -85486            1  \n",
       "236        1.0   38299913       2      AC       32    -90907            1  \n",
       "237        1.0   38304670       2      AC       53    -86171            1  \n",
       "\n",
       "[62 rows x 14 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data['distance']=data['str.start'] - data['snp.start']\n",
    "data['within_100k'] = data['distance'].apply(lambda x: 1 if abs(x)<100000 else 0)\n",
    "print(data.shape)\n",
    "print('There are %s eGenes with associates eSNPs for schizophrenia '%len(set(list(data['gene']))))\n",
    "data.loc[data['within_100k']==1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chrom</th>\n",
       "      <th>best.str.start</th>\n",
       "      <th>best.score</th>\n",
       "      <th>best.q</th>\n",
       "      <th>best.tissue</th>\n",
       "      <th>NumTissues</th>\n",
       "      <th>gene.name</th>\n",
       "      <th>top.variant</th>\n",
       "      <th>best.llq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ENSG00000172137.14</td>\n",
       "      <td>chr16</td>\n",
       "      <td>71335226.0</td>\n",
       "      <td>4.545600e-03</td>\n",
       "      <td>0.057028</td>\n",
       "      <td>Heart-LeftVentricle</td>\n",
       "      <td>17</td>\n",
       "      <td>CALB2</td>\n",
       "      <td>SNP_71308106</td>\n",
       "      <td>0.065267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>229</th>\n",
       "      <td>ENSG00000100731.11</td>\n",
       "      <td>chr14</td>\n",
       "      <td>71416109.0</td>\n",
       "      <td>4.541180e-03</td>\n",
       "      <td>0.002441</td>\n",
       "      <td>Esophagus-Mucosa</td>\n",
       "      <td>17</td>\n",
       "      <td>PCNX</td>\n",
       "      <td>SNP_71401112</td>\n",
       "      <td>0.001157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>331</th>\n",
       "      <td>ENSG00000182667.10</td>\n",
       "      <td>chr11</td>\n",
       "      <td>131925517.0</td>\n",
       "      <td>1.389730e-02</td>\n",
       "      <td>0.001758</td>\n",
       "      <td>Nerve-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>NTM</td>\n",
       "      <td>SNP_131991325</td>\n",
       "      <td>0.000833</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>524</th>\n",
       "      <td>ENSG00000100207.14</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42819557.0</td>\n",
       "      <td>2.421520e-01</td>\n",
       "      <td>0.030173</td>\n",
       "      <td>Nerve-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>TCF20</td>\n",
       "      <td>STR_42819557</td>\n",
       "      <td>0.008556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1184</th>\n",
       "      <td>ENSG00000140836.10</td>\n",
       "      <td>chr16</td>\n",
       "      <td>73089578.0</td>\n",
       "      <td>4.615650e-02</td>\n",
       "      <td>0.091874</td>\n",
       "      <td>Esophagus-Mucosa</td>\n",
       "      <td>17</td>\n",
       "      <td>ZFHX3</td>\n",
       "      <td>SNP_73089716</td>\n",
       "      <td>0.057791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1276</th>\n",
       "      <td>ENSG00000153707.11</td>\n",
       "      <td>chr9</td>\n",
       "      <td>9595947.0</td>\n",
       "      <td>6.680190e-02</td>\n",
       "      <td>0.011214</td>\n",
       "      <td>Nerve-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>PTPRD</td>\n",
       "      <td>STR_9595947</td>\n",
       "      <td>0.001006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1281</th>\n",
       "      <td>ENSG00000151422.8</td>\n",
       "      <td>chr5</td>\n",
       "      <td>108032282.0</td>\n",
       "      <td>5.620240e-02</td>\n",
       "      <td>0.092540</td>\n",
       "      <td>Muscle-Skeletal</td>\n",
       "      <td>17</td>\n",
       "      <td>FER</td>\n",
       "      <td>STR_108032282</td>\n",
       "      <td>0.021767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1331</th>\n",
       "      <td>ENSG00000134698.10</td>\n",
       "      <td>chr1</td>\n",
       "      <td>36270731.0</td>\n",
       "      <td>8.211240e-02</td>\n",
       "      <td>0.087322</td>\n",
       "      <td>Artery-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>AGO4</td>\n",
       "      <td>STR_36270731</td>\n",
       "      <td>0.057141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1770</th>\n",
       "      <td>ENSG00000166444.13</td>\n",
       "      <td>chr11</td>\n",
       "      <td>8839810.0</td>\n",
       "      <td>4.352600e-02</td>\n",
       "      <td>0.086315</td>\n",
       "      <td>Adipose-Subcutaneous</td>\n",
       "      <td>17</td>\n",
       "      <td>ST5</td>\n",
       "      <td>STR_8839810</td>\n",
       "      <td>0.088472</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1822</th>\n",
       "      <td>ENSG00000070366.9</td>\n",
       "      <td>chr17</td>\n",
       "      <td>2129570.0</td>\n",
       "      <td>5.552770e-02</td>\n",
       "      <td>0.090078</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>SMG6</td>\n",
       "      <td>SNP_2026417</td>\n",
       "      <td>0.043943</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2202</th>\n",
       "      <td>ENSG00000166743.5</td>\n",
       "      <td>chr16</td>\n",
       "      <td>20783721.0</td>\n",
       "      <td>5.081830e-03</td>\n",
       "      <td>0.001756</td>\n",
       "      <td>Nerve-Tibial</td>\n",
       "      <td>16</td>\n",
       "      <td>ACSM1</td>\n",
       "      <td>SNP_20800916</td>\n",
       "      <td>0.000849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2470</th>\n",
       "      <td>ENSG00000031081.6</td>\n",
       "      <td>chr3</td>\n",
       "      <td>119023742.0</td>\n",
       "      <td>5.614370e-02</td>\n",
       "      <td>0.012053</td>\n",
       "      <td>Skin-NotSunExposed</td>\n",
       "      <td>17</td>\n",
       "      <td>ARHGAP31</td>\n",
       "      <td>SNP_119016581</td>\n",
       "      <td>0.007138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2513</th>\n",
       "      <td>ENSG00000153561.8</td>\n",
       "      <td>chr2</td>\n",
       "      <td>87015001.0</td>\n",
       "      <td>7.217370e-02</td>\n",
       "      <td>0.084054</td>\n",
       "      <td>Brain-Cerebellum</td>\n",
       "      <td>17</td>\n",
       "      <td>RMND5A</td>\n",
       "      <td>STR_87015001</td>\n",
       "      <td>0.076962</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3159</th>\n",
       "      <td>ENSG00000118922.12</td>\n",
       "      <td>chr13</td>\n",
       "      <td>74625469.0</td>\n",
       "      <td>2.220620e-01</td>\n",
       "      <td>0.017356</td>\n",
       "      <td>Adipose-Subcutaneous</td>\n",
       "      <td>17</td>\n",
       "      <td>KLF12</td>\n",
       "      <td>STR_74625469</td>\n",
       "      <td>0.008266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3167</th>\n",
       "      <td>ENSG00000105866.9</td>\n",
       "      <td>chr7</td>\n",
       "      <td>21508137.0</td>\n",
       "      <td>1.761360e-01</td>\n",
       "      <td>0.042465</td>\n",
       "      <td>Brain-Cerebellum</td>\n",
       "      <td>17</td>\n",
       "      <td>SP4</td>\n",
       "      <td>STR_21508137</td>\n",
       "      <td>0.031620</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3274</th>\n",
       "      <td>ENSG00000139970.12</td>\n",
       "      <td>chr14</td>\n",
       "      <td>60150172.0</td>\n",
       "      <td>8.841270e-03</td>\n",
       "      <td>0.001820</td>\n",
       "      <td>Thyroid</td>\n",
       "      <td>17</td>\n",
       "      <td>RTN1</td>\n",
       "      <td>SNP_60151508</td>\n",
       "      <td>0.000851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3537</th>\n",
       "      <td>ENSG00000140564.6</td>\n",
       "      <td>chr15</td>\n",
       "      <td>91392830.0</td>\n",
       "      <td>7.967430e-03</td>\n",
       "      <td>0.077631</td>\n",
       "      <td>Artery-Aorta</td>\n",
       "      <td>17</td>\n",
       "      <td>FURIN</td>\n",
       "      <td>SNP_91428636</td>\n",
       "      <td>0.065570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3543</th>\n",
       "      <td>ENSG00000156218.8</td>\n",
       "      <td>chr15</td>\n",
       "      <td>84346691.0</td>\n",
       "      <td>8.241310e-02</td>\n",
       "      <td>0.002540</td>\n",
       "      <td>Thyroid</td>\n",
       "      <td>17</td>\n",
       "      <td>ADAMTSL3</td>\n",
       "      <td>SNP_84367968</td>\n",
       "      <td>0.001028</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3621</th>\n",
       "      <td>ENSG00000234127.4</td>\n",
       "      <td>chr6</td>\n",
       "      <td>30189240.0</td>\n",
       "      <td>2.934870e-02</td>\n",
       "      <td>0.005401</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>TRIM26</td>\n",
       "      <td>STR_30189240</td>\n",
       "      <td>0.010418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3916</th>\n",
       "      <td>ENSG00000189134.3</td>\n",
       "      <td>chr6</td>\n",
       "      <td>28168887.0</td>\n",
       "      <td>1.645210e-01</td>\n",
       "      <td>0.032278</td>\n",
       "      <td>Adipose-Visceral</td>\n",
       "      <td>17</td>\n",
       "      <td>NKAPL</td>\n",
       "      <td>STR_28168887</td>\n",
       "      <td>0.029273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3959</th>\n",
       "      <td>ENSG00000141293.11</td>\n",
       "      <td>chr17</td>\n",
       "      <td>46530385.0</td>\n",
       "      <td>1.530190e-02</td>\n",
       "      <td>0.001825</td>\n",
       "      <td>Thyroid</td>\n",
       "      <td>16</td>\n",
       "      <td>SKAP1</td>\n",
       "      <td>SNP_46530158</td>\n",
       "      <td>0.000851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4124</th>\n",
       "      <td>ENSG00000140948.7</td>\n",
       "      <td>chr16</td>\n",
       "      <td>87586949.0</td>\n",
       "      <td>3.689510e-02</td>\n",
       "      <td>0.016890</td>\n",
       "      <td>Artery-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>ZCCHC14</td>\n",
       "      <td>SNP_87575332</td>\n",
       "      <td>0.014343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4172</th>\n",
       "      <td>ENSG00000186472.15</td>\n",
       "      <td>chr7</td>\n",
       "      <td>82731428.0</td>\n",
       "      <td>8.108110e-03</td>\n",
       "      <td>0.098826</td>\n",
       "      <td>Heart-LeftVentricle</td>\n",
       "      <td>17</td>\n",
       "      <td>PCLO</td>\n",
       "      <td>SNP_82726655</td>\n",
       "      <td>0.030372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4236</th>\n",
       "      <td>ENSG00000084710.9</td>\n",
       "      <td>chr2</td>\n",
       "      <td>25384460.0</td>\n",
       "      <td>1.982340e-01</td>\n",
       "      <td>0.009288</td>\n",
       "      <td>WholeBlood</td>\n",
       "      <td>17</td>\n",
       "      <td>EFR3B</td>\n",
       "      <td>STR_25384460</td>\n",
       "      <td>0.004777</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4552</th>\n",
       "      <td>ENSG00000124299.9</td>\n",
       "      <td>chr19</td>\n",
       "      <td>33914002.0</td>\n",
       "      <td>9.143340e-02</td>\n",
       "      <td>0.082393</td>\n",
       "      <td>Artery-Aorta</td>\n",
       "      <td>17</td>\n",
       "      <td>PEPD</td>\n",
       "      <td>SNP_33907536</td>\n",
       "      <td>0.009879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4597</th>\n",
       "      <td>ENSG00000160685.9</td>\n",
       "      <td>chr1</td>\n",
       "      <td>155027695.0</td>\n",
       "      <td>6.735660e-02</td>\n",
       "      <td>0.059769</td>\n",
       "      <td>WholeBlood</td>\n",
       "      <td>17</td>\n",
       "      <td>ZBTB7B</td>\n",
       "      <td>SNP_155030557</td>\n",
       "      <td>0.065131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4946</th>\n",
       "      <td>ENSG00000187741.10</td>\n",
       "      <td>chr16</td>\n",
       "      <td>89856806.0</td>\n",
       "      <td>1.467660e-11</td>\n",
       "      <td>0.021236</td>\n",
       "      <td>Adipose-Subcutaneous</td>\n",
       "      <td>17</td>\n",
       "      <td>FANCA</td>\n",
       "      <td>SNP_89883148</td>\n",
       "      <td>0.040287</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5508</th>\n",
       "      <td>ENSG00000006744.14</td>\n",
       "      <td>chr17</td>\n",
       "      <td>12865398.0</td>\n",
       "      <td>8.782120e-09</td>\n",
       "      <td>0.018706</td>\n",
       "      <td>Muscle-Skeletal</td>\n",
       "      <td>17</td>\n",
       "      <td>ELAC2</td>\n",
       "      <td>SNP_12895984</td>\n",
       "      <td>0.020965</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5540</th>\n",
       "      <td>ENSG00000162909.13</td>\n",
       "      <td>chr1</td>\n",
       "      <td>223894939.0</td>\n",
       "      <td>8.428090e-02</td>\n",
       "      <td>0.002578</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>CAPN2</td>\n",
       "      <td>SNP_223892703</td>\n",
       "      <td>0.001293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5590</th>\n",
       "      <td>ENSG00000185634.7</td>\n",
       "      <td>chr15</td>\n",
       "      <td>49304331.0</td>\n",
       "      <td>2.050180e-02</td>\n",
       "      <td>0.002133</td>\n",
       "      <td>Artery-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>SHC4</td>\n",
       "      <td>SNP_49325289</td>\n",
       "      <td>0.001080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10969</th>\n",
       "      <td>ENSG00000100162.10</td>\n",
       "      <td>chr22</td>\n",
       "      <td>42351189.0</td>\n",
       "      <td>1.489140e-01</td>\n",
       "      <td>0.003300</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>CENPM</td>\n",
       "      <td>SNP_42306369</td>\n",
       "      <td>0.001358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10970</th>\n",
       "      <td>ENSG00000049167.9</td>\n",
       "      <td>chr5</td>\n",
       "      <td>60171714.0</td>\n",
       "      <td>1.836050e-02</td>\n",
       "      <td>0.017774</td>\n",
       "      <td>WholeBlood</td>\n",
       "      <td>17</td>\n",
       "      <td>ERCC8</td>\n",
       "      <td>STR_60171714</td>\n",
       "      <td>0.020880</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11059</th>\n",
       "      <td>ENSG00000157404.11</td>\n",
       "      <td>chr4</td>\n",
       "      <td>55530745.0</td>\n",
       "      <td>2.894120e-02</td>\n",
       "      <td>0.095151</td>\n",
       "      <td>Skin-NotSunExposed</td>\n",
       "      <td>17</td>\n",
       "      <td>KIT</td>\n",
       "      <td>STR_55530745</td>\n",
       "      <td>0.102304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11064</th>\n",
       "      <td>ENSG00000178342.4</td>\n",
       "      <td>chr18</td>\n",
       "      <td>77565864.0</td>\n",
       "      <td>1.506260e-01</td>\n",
       "      <td>0.035205</td>\n",
       "      <td>Esophagus-Mucosa</td>\n",
       "      <td>17</td>\n",
       "      <td>KCNG2</td>\n",
       "      <td>STR_77565864</td>\n",
       "      <td>0.010231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11228</th>\n",
       "      <td>ENSG00000173575.14</td>\n",
       "      <td>chr15</td>\n",
       "      <td>93453170.0</td>\n",
       "      <td>2.839220e-02</td>\n",
       "      <td>0.037131</td>\n",
       "      <td>Esophagus-Muscularis</td>\n",
       "      <td>17</td>\n",
       "      <td>CHD2</td>\n",
       "      <td>STR_93453170</td>\n",
       "      <td>0.065758</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11252</th>\n",
       "      <td>ENSG00000011405.9</td>\n",
       "      <td>chr11</td>\n",
       "      <td>17086239.0</td>\n",
       "      <td>7.846460e-02</td>\n",
       "      <td>0.099794</td>\n",
       "      <td>Lung</td>\n",
       "      <td>17</td>\n",
       "      <td>PIK3C2A</td>\n",
       "      <td>STR_17086239</td>\n",
       "      <td>0.065022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11921</th>\n",
       "      <td>ENSG00000151150.16</td>\n",
       "      <td>chr10</td>\n",
       "      <td>62540788.0</td>\n",
       "      <td>4.678650e-02</td>\n",
       "      <td>0.022319</td>\n",
       "      <td>Artery-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>ANK3</td>\n",
       "      <td>SNP_62060382</td>\n",
       "      <td>0.007233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12060</th>\n",
       "      <td>ENSG00000142949.12</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44025261.0</td>\n",
       "      <td>5.285810e-02</td>\n",
       "      <td>0.026748</td>\n",
       "      <td>Esophagus-Muscularis</td>\n",
       "      <td>17</td>\n",
       "      <td>PTPRF</td>\n",
       "      <td>SNP_44018604</td>\n",
       "      <td>0.015328</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12090</th>\n",
       "      <td>ENSG00000117707.11</td>\n",
       "      <td>chr1</td>\n",
       "      <td>214137474.0</td>\n",
       "      <td>4.566840e-02</td>\n",
       "      <td>0.079520</td>\n",
       "      <td>Thyroid</td>\n",
       "      <td>17</td>\n",
       "      <td>PROX1</td>\n",
       "      <td>STR_214137474</td>\n",
       "      <td>0.060766</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12144</th>\n",
       "      <td>ENSG00000050405.9</td>\n",
       "      <td>chr12</td>\n",
       "      <td>50701452.0</td>\n",
       "      <td>8.985610e-03</td>\n",
       "      <td>0.027447</td>\n",
       "      <td>Thyroid</td>\n",
       "      <td>17</td>\n",
       "      <td>LIMA1</td>\n",
       "      <td>SNP_50747005</td>\n",
       "      <td>0.017000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12256</th>\n",
       "      <td>ENSG00000213593.5</td>\n",
       "      <td>chr11</td>\n",
       "      <td>57522177.0</td>\n",
       "      <td>2.035050e-02</td>\n",
       "      <td>0.065600</td>\n",
       "      <td>Thyroid</td>\n",
       "      <td>17</td>\n",
       "      <td>TMX2</td>\n",
       "      <td>STR_57522177</td>\n",
       "      <td>0.067158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12291</th>\n",
       "      <td>ENSG00000103423.9</td>\n",
       "      <td>chr16</td>\n",
       "      <td>4533162.0</td>\n",
       "      <td>6.240460e-02</td>\n",
       "      <td>0.045218</td>\n",
       "      <td>Skin-NotSunExposed</td>\n",
       "      <td>17</td>\n",
       "      <td>DNAJA3</td>\n",
       "      <td>STR_4533162</td>\n",
       "      <td>0.080405</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12917</th>\n",
       "      <td>ENSG00000155511.13</td>\n",
       "      <td>chr5</td>\n",
       "      <td>152802415.0</td>\n",
       "      <td>8.726530e-02</td>\n",
       "      <td>0.037671</td>\n",
       "      <td>WholeBlood</td>\n",
       "      <td>17</td>\n",
       "      <td>GRIA1</td>\n",
       "      <td>SNP_153181546</td>\n",
       "      <td>0.016513</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13178</th>\n",
       "      <td>ENSG00000135250.12</td>\n",
       "      <td>chr7</td>\n",
       "      <td>104694906.0</td>\n",
       "      <td>1.501310e-01</td>\n",
       "      <td>0.062815</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>SRPK2</td>\n",
       "      <td>STR_104694906</td>\n",
       "      <td>0.020555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13262</th>\n",
       "      <td>ENSG00000163634.7</td>\n",
       "      <td>chr3</td>\n",
       "      <td>63728508.0</td>\n",
       "      <td>6.428890e-02</td>\n",
       "      <td>0.093804</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>THOC7</td>\n",
       "      <td>STR_63728508</td>\n",
       "      <td>0.045633</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13292</th>\n",
       "      <td>ENSG00000174373.11</td>\n",
       "      <td>chr14</td>\n",
       "      <td>36214605.0</td>\n",
       "      <td>2.235290e-03</td>\n",
       "      <td>0.098621</td>\n",
       "      <td>Brain-Cerebellum</td>\n",
       "      <td>17</td>\n",
       "      <td>RALGAPA1</td>\n",
       "      <td>SNP_35924686</td>\n",
       "      <td>0.051057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13367</th>\n",
       "      <td>ENSG00000184226.10</td>\n",
       "      <td>chr13</td>\n",
       "      <td>67462676.0</td>\n",
       "      <td>7.632440e-02</td>\n",
       "      <td>0.048658</td>\n",
       "      <td>Artery-Aorta</td>\n",
       "      <td>17</td>\n",
       "      <td>PCDH9</td>\n",
       "      <td>STR_67462676</td>\n",
       "      <td>0.007176</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13466</th>\n",
       "      <td>ENSG00000073711.6</td>\n",
       "      <td>chr3</td>\n",
       "      <td>135742058.0</td>\n",
       "      <td>3.633750e-02</td>\n",
       "      <td>0.004471</td>\n",
       "      <td>Adipose-Subcutaneous</td>\n",
       "      <td>17</td>\n",
       "      <td>PPP2R3A</td>\n",
       "      <td>SNP_135792313</td>\n",
       "      <td>0.002942</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13617</th>\n",
       "      <td>ENSG00000067082.10</td>\n",
       "      <td>chr10</td>\n",
       "      <td>3809510.0</td>\n",
       "      <td>1.372350e-02</td>\n",
       "      <td>0.089672</td>\n",
       "      <td>Skin-SunExposed</td>\n",
       "      <td>17</td>\n",
       "      <td>KLF6</td>\n",
       "      <td>SNP_3818058</td>\n",
       "      <td>0.087977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13801</th>\n",
       "      <td>ENSG00000214575.5</td>\n",
       "      <td>chr15</td>\n",
       "      <td>83320867.0</td>\n",
       "      <td>2.928750e-02</td>\n",
       "      <td>0.039830</td>\n",
       "      <td>Brain-Cerebellum</td>\n",
       "      <td>17</td>\n",
       "      <td>CPEB1</td>\n",
       "      <td>SNP_83299956</td>\n",
       "      <td>0.048112</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14110</th>\n",
       "      <td>ENSG00000144724.14</td>\n",
       "      <td>chr3</td>\n",
       "      <td>61718384.0</td>\n",
       "      <td>3.770970e-02</td>\n",
       "      <td>0.028341</td>\n",
       "      <td>Lung</td>\n",
       "      <td>17</td>\n",
       "      <td>PTPRG</td>\n",
       "      <td>SNP_61722126</td>\n",
       "      <td>0.012534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14277</th>\n",
       "      <td>ENSG00000261115.1</td>\n",
       "      <td>chr7</td>\n",
       "      <td>140676038.0</td>\n",
       "      <td>3.646160e-01</td>\n",
       "      <td>0.002133</td>\n",
       "      <td>Artery-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>TMEM178B</td>\n",
       "      <td>STR_140676038</td>\n",
       "      <td>0.001080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14303</th>\n",
       "      <td>ENSG00000099904.11</td>\n",
       "      <td>chr22</td>\n",
       "      <td>20063280.0</td>\n",
       "      <td>8.052310e-01</td>\n",
       "      <td>0.004226</td>\n",
       "      <td>Artery-Aorta</td>\n",
       "      <td>17</td>\n",
       "      <td>ZDHHC8</td>\n",
       "      <td>STR_20063280</td>\n",
       "      <td>0.002050</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14459</th>\n",
       "      <td>ENSG00000153317.10</td>\n",
       "      <td>chr8</td>\n",
       "      <td>131430786.0</td>\n",
       "      <td>5.432710e-02</td>\n",
       "      <td>0.005166</td>\n",
       "      <td>Cells-Transformedfibroblasts</td>\n",
       "      <td>17</td>\n",
       "      <td>ASAP1</td>\n",
       "      <td>SNP_131427202</td>\n",
       "      <td>0.001911</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14468</th>\n",
       "      <td>ENSG00000142599.13</td>\n",
       "      <td>chr1</td>\n",
       "      <td>8695997.0</td>\n",
       "      <td>1.193400e-02</td>\n",
       "      <td>0.038435</td>\n",
       "      <td>Lung</td>\n",
       "      <td>17</td>\n",
       "      <td>RERE</td>\n",
       "      <td>SNP_8481016</td>\n",
       "      <td>0.011876</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14556</th>\n",
       "      <td>ENSG00000138821.8</td>\n",
       "      <td>chr4</td>\n",
       "      <td>103237183.0</td>\n",
       "      <td>8.112980e-02</td>\n",
       "      <td>0.058480</td>\n",
       "      <td>Muscle-Skeletal</td>\n",
       "      <td>17</td>\n",
       "      <td>SLC39A8</td>\n",
       "      <td>SNP_103234413</td>\n",
       "      <td>0.024008</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14922</th>\n",
       "      <td>ENSG00000167720.8</td>\n",
       "      <td>chr17</td>\n",
       "      <td>2129570.0</td>\n",
       "      <td>5.449040e-05</td>\n",
       "      <td>0.010830</td>\n",
       "      <td>WholeBlood</td>\n",
       "      <td>17</td>\n",
       "      <td>SRR</td>\n",
       "      <td>SNP_2131580</td>\n",
       "      <td>0.007627</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14983</th>\n",
       "      <td>ENSG00000093010.7</td>\n",
       "      <td>chr22</td>\n",
       "      <td>19903852.0</td>\n",
       "      <td>4.274690e-01</td>\n",
       "      <td>0.024488</td>\n",
       "      <td>Adipose-Visceral</td>\n",
       "      <td>17</td>\n",
       "      <td>COMT</td>\n",
       "      <td>STR_19903852</td>\n",
       "      <td>0.009536</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14990</th>\n",
       "      <td>ENSG00000183166.6</td>\n",
       "      <td>chr7</td>\n",
       "      <td>71845653.0</td>\n",
       "      <td>5.768180e-02</td>\n",
       "      <td>0.091214</td>\n",
       "      <td>Skin-SunExposed</td>\n",
       "      <td>17</td>\n",
       "      <td>CALN1</td>\n",
       "      <td>SNP_71706809</td>\n",
       "      <td>0.015107</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15023</th>\n",
       "      <td>ENSG00000182541.13</td>\n",
       "      <td>chr22</td>\n",
       "      <td>31570366.0</td>\n",
       "      <td>4.590850e-02</td>\n",
       "      <td>0.092542</td>\n",
       "      <td>Nerve-Tibial</td>\n",
       "      <td>17</td>\n",
       "      <td>LIMK2</td>\n",
       "      <td>SNP_31577850</td>\n",
       "      <td>0.033836</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>84 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     gene  chrom  best.str.start    best.score    best.q  \\\n",
       "9      ENSG00000172137.14  chr16      71335226.0  4.545600e-03  0.057028   \n",
       "229    ENSG00000100731.11  chr14      71416109.0  4.541180e-03  0.002441   \n",
       "331    ENSG00000182667.10  chr11     131925517.0  1.389730e-02  0.001758   \n",
       "524    ENSG00000100207.14  chr22      42819557.0  2.421520e-01  0.030173   \n",
       "1184   ENSG00000140836.10  chr16      73089578.0  4.615650e-02  0.091874   \n",
       "1276   ENSG00000153707.11   chr9       9595947.0  6.680190e-02  0.011214   \n",
       "1281    ENSG00000151422.8   chr5     108032282.0  5.620240e-02  0.092540   \n",
       "1331   ENSG00000134698.10   chr1      36270731.0  8.211240e-02  0.087322   \n",
       "1770   ENSG00000166444.13  chr11       8839810.0  4.352600e-02  0.086315   \n",
       "1822    ENSG00000070366.9  chr17       2129570.0  5.552770e-02  0.090078   \n",
       "2202    ENSG00000166743.5  chr16      20783721.0  5.081830e-03  0.001756   \n",
       "2470    ENSG00000031081.6   chr3     119023742.0  5.614370e-02  0.012053   \n",
       "2513    ENSG00000153561.8   chr2      87015001.0  7.217370e-02  0.084054   \n",
       "3159   ENSG00000118922.12  chr13      74625469.0  2.220620e-01  0.017356   \n",
       "3167    ENSG00000105866.9   chr7      21508137.0  1.761360e-01  0.042465   \n",
       "3274   ENSG00000139970.12  chr14      60150172.0  8.841270e-03  0.001820   \n",
       "3537    ENSG00000140564.6  chr15      91392830.0  7.967430e-03  0.077631   \n",
       "3543    ENSG00000156218.8  chr15      84346691.0  8.241310e-02  0.002540   \n",
       "3621    ENSG00000234127.4   chr6      30189240.0  2.934870e-02  0.005401   \n",
       "3916    ENSG00000189134.3   chr6      28168887.0  1.645210e-01  0.032278   \n",
       "3959   ENSG00000141293.11  chr17      46530385.0  1.530190e-02  0.001825   \n",
       "4124    ENSG00000140948.7  chr16      87586949.0  3.689510e-02  0.016890   \n",
       "4172   ENSG00000186472.15   chr7      82731428.0  8.108110e-03  0.098826   \n",
       "4236    ENSG00000084710.9   chr2      25384460.0  1.982340e-01  0.009288   \n",
       "4552    ENSG00000124299.9  chr19      33914002.0  9.143340e-02  0.082393   \n",
       "4597    ENSG00000160685.9   chr1     155027695.0  6.735660e-02  0.059769   \n",
       "4946   ENSG00000187741.10  chr16      89856806.0  1.467660e-11  0.021236   \n",
       "5508   ENSG00000006744.14  chr17      12865398.0  8.782120e-09  0.018706   \n",
       "5540   ENSG00000162909.13   chr1     223894939.0  8.428090e-02  0.002578   \n",
       "5590    ENSG00000185634.7  chr15      49304331.0  2.050180e-02  0.002133   \n",
       "...                   ...    ...             ...           ...       ...   \n",
       "10969  ENSG00000100162.10  chr22      42351189.0  1.489140e-01  0.003300   \n",
       "10970   ENSG00000049167.9   chr5      60171714.0  1.836050e-02  0.017774   \n",
       "11059  ENSG00000157404.11   chr4      55530745.0  2.894120e-02  0.095151   \n",
       "11064   ENSG00000178342.4  chr18      77565864.0  1.506260e-01  0.035205   \n",
       "11228  ENSG00000173575.14  chr15      93453170.0  2.839220e-02  0.037131   \n",
       "11252   ENSG00000011405.9  chr11      17086239.0  7.846460e-02  0.099794   \n",
       "11921  ENSG00000151150.16  chr10      62540788.0  4.678650e-02  0.022319   \n",
       "12060  ENSG00000142949.12   chr1      44025261.0  5.285810e-02  0.026748   \n",
       "12090  ENSG00000117707.11   chr1     214137474.0  4.566840e-02  0.079520   \n",
       "12144   ENSG00000050405.9  chr12      50701452.0  8.985610e-03  0.027447   \n",
       "12256   ENSG00000213593.5  chr11      57522177.0  2.035050e-02  0.065600   \n",
       "12291   ENSG00000103423.9  chr16       4533162.0  6.240460e-02  0.045218   \n",
       "12917  ENSG00000155511.13   chr5     152802415.0  8.726530e-02  0.037671   \n",
       "13178  ENSG00000135250.12   chr7     104694906.0  1.501310e-01  0.062815   \n",
       "13262   ENSG00000163634.7   chr3      63728508.0  6.428890e-02  0.093804   \n",
       "13292  ENSG00000174373.11  chr14      36214605.0  2.235290e-03  0.098621   \n",
       "13367  ENSG00000184226.10  chr13      67462676.0  7.632440e-02  0.048658   \n",
       "13466   ENSG00000073711.6   chr3     135742058.0  3.633750e-02  0.004471   \n",
       "13617  ENSG00000067082.10  chr10       3809510.0  1.372350e-02  0.089672   \n",
       "13801   ENSG00000214575.5  chr15      83320867.0  2.928750e-02  0.039830   \n",
       "14110  ENSG00000144724.14   chr3      61718384.0  3.770970e-02  0.028341   \n",
       "14277   ENSG00000261115.1   chr7     140676038.0  3.646160e-01  0.002133   \n",
       "14303  ENSG00000099904.11  chr22      20063280.0  8.052310e-01  0.004226   \n",
       "14459  ENSG00000153317.10   chr8     131430786.0  5.432710e-02  0.005166   \n",
       "14468  ENSG00000142599.13   chr1       8695997.0  1.193400e-02  0.038435   \n",
       "14556   ENSG00000138821.8   chr4     103237183.0  8.112980e-02  0.058480   \n",
       "14922   ENSG00000167720.8  chr17       2129570.0  5.449040e-05  0.010830   \n",
       "14983   ENSG00000093010.7  chr22      19903852.0  4.274690e-01  0.024488   \n",
       "14990   ENSG00000183166.6   chr7      71845653.0  5.768180e-02  0.091214   \n",
       "15023  ENSG00000182541.13  chr22      31570366.0  4.590850e-02  0.092542   \n",
       "\n",
       "                        best.tissue  NumTissues gene.name    top.variant  \\\n",
       "9               Heart-LeftVentricle          17     CALB2   SNP_71308106   \n",
       "229                Esophagus-Mucosa          17      PCNX   SNP_71401112   \n",
       "331                    Nerve-Tibial          17       NTM  SNP_131991325   \n",
       "524                    Nerve-Tibial          17     TCF20   STR_42819557   \n",
       "1184               Esophagus-Mucosa          17     ZFHX3   SNP_73089716   \n",
       "1276                   Nerve-Tibial          17     PTPRD    STR_9595947   \n",
       "1281                Muscle-Skeletal          17       FER  STR_108032282   \n",
       "1331                  Artery-Tibial          17      AGO4   STR_36270731   \n",
       "1770           Adipose-Subcutaneous          17       ST5    STR_8839810   \n",
       "1822   Cells-Transformedfibroblasts          17      SMG6    SNP_2026417   \n",
       "2202                   Nerve-Tibial          16     ACSM1   SNP_20800916   \n",
       "2470             Skin-NotSunExposed          17  ARHGAP31  SNP_119016581   \n",
       "2513               Brain-Cerebellum          17    RMND5A   STR_87015001   \n",
       "3159           Adipose-Subcutaneous          17     KLF12   STR_74625469   \n",
       "3167               Brain-Cerebellum          17       SP4   STR_21508137   \n",
       "3274                        Thyroid          17      RTN1   SNP_60151508   \n",
       "3537                   Artery-Aorta          17     FURIN   SNP_91428636   \n",
       "3543                        Thyroid          17  ADAMTSL3   SNP_84367968   \n",
       "3621   Cells-Transformedfibroblasts          17    TRIM26   STR_30189240   \n",
       "3916               Adipose-Visceral          17     NKAPL   STR_28168887   \n",
       "3959                        Thyroid          16     SKAP1   SNP_46530158   \n",
       "4124                  Artery-Tibial          17   ZCCHC14   SNP_87575332   \n",
       "4172            Heart-LeftVentricle          17      PCLO   SNP_82726655   \n",
       "4236                     WholeBlood          17     EFR3B   STR_25384460   \n",
       "4552                   Artery-Aorta          17      PEPD   SNP_33907536   \n",
       "4597                     WholeBlood          17    ZBTB7B  SNP_155030557   \n",
       "4946           Adipose-Subcutaneous          17     FANCA   SNP_89883148   \n",
       "5508                Muscle-Skeletal          17     ELAC2   SNP_12895984   \n",
       "5540   Cells-Transformedfibroblasts          17     CAPN2  SNP_223892703   \n",
       "5590                  Artery-Tibial          17      SHC4   SNP_49325289   \n",
       "...                             ...         ...       ...            ...   \n",
       "10969  Cells-Transformedfibroblasts          17     CENPM   SNP_42306369   \n",
       "10970                    WholeBlood          17     ERCC8   STR_60171714   \n",
       "11059            Skin-NotSunExposed          17       KIT   STR_55530745   \n",
       "11064              Esophagus-Mucosa          17     KCNG2   STR_77565864   \n",
       "11228          Esophagus-Muscularis          17      CHD2   STR_93453170   \n",
       "11252                          Lung          17   PIK3C2A   STR_17086239   \n",
       "11921                 Artery-Tibial          17      ANK3   SNP_62060382   \n",
       "12060          Esophagus-Muscularis          17     PTPRF   SNP_44018604   \n",
       "12090                       Thyroid          17     PROX1  STR_214137474   \n",
       "12144                       Thyroid          17     LIMA1   SNP_50747005   \n",
       "12256                       Thyroid          17      TMX2   STR_57522177   \n",
       "12291            Skin-NotSunExposed          17    DNAJA3    STR_4533162   \n",
       "12917                    WholeBlood          17     GRIA1  SNP_153181546   \n",
       "13178  Cells-Transformedfibroblasts          17     SRPK2  STR_104694906   \n",
       "13262  Cells-Transformedfibroblasts          17     THOC7   STR_63728508   \n",
       "13292              Brain-Cerebellum          17  RALGAPA1   SNP_35924686   \n",
       "13367                  Artery-Aorta          17     PCDH9   STR_67462676   \n",
       "13466          Adipose-Subcutaneous          17   PPP2R3A  SNP_135792313   \n",
       "13617               Skin-SunExposed          17      KLF6    SNP_3818058   \n",
       "13801              Brain-Cerebellum          17     CPEB1   SNP_83299956   \n",
       "14110                          Lung          17     PTPRG   SNP_61722126   \n",
       "14277                 Artery-Tibial          17  TMEM178B  STR_140676038   \n",
       "14303                  Artery-Aorta          17    ZDHHC8   STR_20063280   \n",
       "14459  Cells-Transformedfibroblasts          17     ASAP1  SNP_131427202   \n",
       "14468                          Lung          17      RERE    SNP_8481016   \n",
       "14556               Muscle-Skeletal          17   SLC39A8  SNP_103234413   \n",
       "14922                    WholeBlood          17       SRR    SNP_2131580   \n",
       "14983              Adipose-Visceral          17      COMT   STR_19903852   \n",
       "14990               Skin-SunExposed          17     CALN1   SNP_71706809   \n",
       "15023                  Nerve-Tibial          17     LIMK2   SNP_31577850   \n",
       "\n",
       "       best.llq  \n",
       "9      0.065267  \n",
       "229    0.001157  \n",
       "331    0.000833  \n",
       "524    0.008556  \n",
       "1184   0.057791  \n",
       "1276   0.001006  \n",
       "1281   0.021767  \n",
       "1331   0.057141  \n",
       "1770   0.088472  \n",
       "1822   0.043943  \n",
       "2202   0.000849  \n",
       "2470   0.007138  \n",
       "2513   0.076962  \n",
       "3159   0.008266  \n",
       "3167   0.031620  \n",
       "3274   0.000851  \n",
       "3537   0.065570  \n",
       "3543   0.001028  \n",
       "3621   0.010418  \n",
       "3916   0.029273  \n",
       "3959   0.000851  \n",
       "4124   0.014343  \n",
       "4172   0.030372  \n",
       "4236   0.004777  \n",
       "4552   0.009879  \n",
       "4597   0.065131  \n",
       "4946   0.040287  \n",
       "5508   0.020965  \n",
       "5540   0.001293  \n",
       "5590   0.001080  \n",
       "...         ...  \n",
       "10969  0.001358  \n",
       "10970  0.020880  \n",
       "11059  0.102304  \n",
       "11064  0.010231  \n",
       "11228  0.065758  \n",
       "11252  0.065022  \n",
       "11921  0.007233  \n",
       "12060  0.015328  \n",
       "12090  0.060766  \n",
       "12144  0.017000  \n",
       "12256  0.067158  \n",
       "12291  0.080405  \n",
       "12917  0.016513  \n",
       "13178  0.020555  \n",
       "13262  0.045633  \n",
       "13292  0.051057  \n",
       "13367  0.007176  \n",
       "13466  0.002942  \n",
       "13617  0.087977  \n",
       "13801  0.048112  \n",
       "14110  0.012534  \n",
       "14277  0.001080  \n",
       "14303  0.002050  \n",
       "14459  0.001911  \n",
       "14468  0.011876  \n",
       "14556  0.024008  \n",
       "14922  0.007627  \n",
       "14983  0.009536  \n",
       "14990  0.015107  \n",
       "15023  0.033836  \n",
       "\n",
       "[84 rows x 10 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Causal_scores = pd.read_csv(\"/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/Merged_Best_causality.Table\", sep='\\t')\n",
    "All_gwas = Causal_scores.loc[Causal_scores['gene'].isin(list(set(list(data['gene']))))]\n",
    "All_gwas.loc[All_gwas['best.q']<=.1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "62 60\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>str.start</th>\n",
       "      <th>str.id</th>\n",
       "      <th>gene</th>\n",
       "      <th>E.tissues</th>\n",
       "      <th>ll.E.tissues</th>\n",
       "      <th>eSTR.Adipose.S</th>\n",
       "      <th>eSTR.Adipose.S.ll</th>\n",
       "      <th>qval.Adipose.S</th>\n",
       "      <th>llqval.Adipose.S</th>\n",
       "      <th>...</th>\n",
       "      <th>beta.Thyroid</th>\n",
       "      <th>eSTR.Blood</th>\n",
       "      <th>eSTR.Blood.ll</th>\n",
       "      <th>qval.Blood</th>\n",
       "      <th>llqval.Blood</th>\n",
       "      <th>beta.Blood</th>\n",
       "      <th>str.end</th>\n",
       "      <th>period</th>\n",
       "      <th>motif</th>\n",
       "      <th>gene.name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10377</th>\n",
       "      <td>chr12</td>\n",
       "      <td>7034264</td>\n",
       "      <td>STR_7034264</td>\n",
       "      <td>ENSG00000111676.10</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.002085</td>\n",
       "      <td>0.001238</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.401075</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.811180</td>\n",
       "      <td>-0.027276</td>\n",
       "      <td>7034322</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>ATN1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14660</th>\n",
       "      <td>chr14</td>\n",
       "      <td>92557796</td>\n",
       "      <td>STR_92557796</td>\n",
       "      <td>ENSG00000066427.17</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.002077</td>\n",
       "      <td>0.001046</td>\n",
       "      <td>...</td>\n",
       "      <td>0.260183</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.098919</td>\n",
       "      <td>0.159107</td>\n",
       "      <td>-0.154160</td>\n",
       "      <td>92557821</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAG</td>\n",
       "      <td>ATXN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32869</th>\n",
       "      <td>chr19</td>\n",
       "      <td>13729541</td>\n",
       "      <td>STR_13729541</td>\n",
       "      <td>ENSG00000141837.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.812012</td>\n",
       "      <td>...</td>\n",
       "      <td>0.135937</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.851923</td>\n",
       "      <td>-0.005792</td>\n",
       "      <td>13729573</td>\n",
       "      <td>2</td>\n",
       "      <td>AG</td>\n",
       "      <td>CACNA1A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7587</th>\n",
       "      <td>chr11</td>\n",
       "      <td>119087050</td>\n",
       "      <td>STR_119087050</td>\n",
       "      <td>ENSG00000110395.4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.788202</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.064029</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.721164</td>\n",
       "      <td>-0.054110</td>\n",
       "      <td>119087097</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>CBL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28256</th>\n",
       "      <td>chr18</td>\n",
       "      <td>72210730</td>\n",
       "      <td>STR_72210730</td>\n",
       "      <td>ENSG00000150656.10</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.342899</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.056906</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.851343</td>\n",
       "      <td>-0.005813</td>\n",
       "      <td>72210778</td>\n",
       "      <td>5</td>\n",
       "      <td>AAAAT</td>\n",
       "      <td>CNDP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16802</th>\n",
       "      <td>chr15</td>\n",
       "      <td>51620745</td>\n",
       "      <td>STR_51620745</td>\n",
       "      <td>ENSG00000137869.9</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.843324</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.025783</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.808066</td>\n",
       "      <td>0.028981</td>\n",
       "      <td>51620782</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>CYP19A1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11100</th>\n",
       "      <td>chr12</td>\n",
       "      <td>51123498</td>\n",
       "      <td>STR_51123498</td>\n",
       "      <td>ENSG00000066084.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.619114</td>\n",
       "      <td>...</td>\n",
       "      <td>0.360361</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.635046</td>\n",
       "      <td>0.329699</td>\n",
       "      <td>0.120274</td>\n",
       "      <td>51123531</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>DIP2B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33138</th>\n",
       "      <td>chr19</td>\n",
       "      <td>46265048</td>\n",
       "      <td>STR_46265048</td>\n",
       "      <td>ENSG00000104936.13</td>\n",
       "      <td>1.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.776608</td>\n",
       "      <td>...</td>\n",
       "      <td>0.290464</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.292570</td>\n",
       "      <td>0.122120</td>\n",
       "      <td>46265085</td>\n",
       "      <td>6</td>\n",
       "      <td>AGCTCC</td>\n",
       "      <td>DMPK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33140</th>\n",
       "      <td>chr19</td>\n",
       "      <td>46308960</td>\n",
       "      <td>STR_46308960</td>\n",
       "      <td>ENSG00000104936.13</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.260466</td>\n",
       "      <td>...</td>\n",
       "      <td>0.064932</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.836187</td>\n",
       "      <td>-0.014291</td>\n",
       "      <td>46308980</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAG</td>\n",
       "      <td>DMPK</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91616</th>\n",
       "      <td>chr9</td>\n",
       "      <td>71685708</td>\n",
       "      <td>STR_71685708</td>\n",
       "      <td>ENSG00000165060.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.706640</td>\n",
       "      <td>...</td>\n",
       "      <td>0.088532</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.736591</td>\n",
       "      <td>0.449685</td>\n",
       "      <td>0.105619</td>\n",
       "      <td>71685728</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>FXN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91617</th>\n",
       "      <td>chr9</td>\n",
       "      <td>71749839</td>\n",
       "      <td>STR_71749839</td>\n",
       "      <td>ENSG00000165060.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.053631</td>\n",
       "      <td>0.034169</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.152007</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.767412</td>\n",
       "      <td>-0.041312</td>\n",
       "      <td>71749912</td>\n",
       "      <td>2</td>\n",
       "      <td>AG</td>\n",
       "      <td>FXN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17959</th>\n",
       "      <td>chr16</td>\n",
       "      <td>11940252</td>\n",
       "      <td>STR_11940252</td>\n",
       "      <td>ENSG00000103342.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.824577</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.092369</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.746033</td>\n",
       "      <td>-0.050057</td>\n",
       "      <td>11940286</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>GSPT1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4341</th>\n",
       "      <td>chr1</td>\n",
       "      <td>154871688</td>\n",
       "      <td>STR_154871688</td>\n",
       "      <td>ENSG00000143603.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.214518</td>\n",
       "      <td>0.204313</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002995</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.857601</td>\n",
       "      <td>0.002003</td>\n",
       "      <td>154871733</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>KCNN3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85613</th>\n",
       "      <td>chr6</td>\n",
       "      <td>31428339</td>\n",
       "      <td>STR_31428339</td>\n",
       "      <td>ENSG00000204520.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.735983</td>\n",
       "      <td>0.482626</td>\n",
       "      <td>...</td>\n",
       "      <td>0.162490</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.371577</td>\n",
       "      <td>0.112800</td>\n",
       "      <td>31428384</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>MICA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85716</th>\n",
       "      <td>chr6</td>\n",
       "      <td>29671619</td>\n",
       "      <td>STR_29671619</td>\n",
       "      <td>ENSG00000204655.7</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.459158</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.126769</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.595641</td>\n",
       "      <td>-0.077660</td>\n",
       "      <td>29671691</td>\n",
       "      <td>4</td>\n",
       "      <td>ATCC</td>\n",
       "      <td>MOG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61114</th>\n",
       "      <td>chr20</td>\n",
       "      <td>46210941</td>\n",
       "      <td>STR_46210941</td>\n",
       "      <td>ENSG00000124151.14</td>\n",
       "      <td>3.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.404204</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.103459</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.041836</td>\n",
       "      <td>0.005397</td>\n",
       "      <td>-0.216156</td>\n",
       "      <td>46210962</td>\n",
       "      <td>4</td>\n",
       "      <td>AATT</td>\n",
       "      <td>NCOA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61115</th>\n",
       "      <td>chr20</td>\n",
       "      <td>46242650</td>\n",
       "      <td>STR_46242650</td>\n",
       "      <td>ENSG00000124151.14</td>\n",
       "      <td>1.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.248326</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.117563</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.018870</td>\n",
       "      <td>-0.212018</td>\n",
       "      <td>46242674</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>NCOA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43396</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41233240</td>\n",
       "      <td>STR_41233240</td>\n",
       "      <td>ENSG00000105245.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.374403</td>\n",
       "      <td>...</td>\n",
       "      <td>0.098156</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.683022</td>\n",
       "      <td>0.063493</td>\n",
       "      <td>41233274</td>\n",
       "      <td>5</td>\n",
       "      <td>AAAAT</td>\n",
       "      <td>NUMBL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43403</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41290402</td>\n",
       "      <td>STR_41290402</td>\n",
       "      <td>ENSG00000105245.5</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.628057</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.020114</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.019162</td>\n",
       "      <td>0.003410</td>\n",
       "      <td>-0.222150</td>\n",
       "      <td>41290429</td>\n",
       "      <td>3</td>\n",
       "      <td>AAC</td>\n",
       "      <td>NUMBL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16973</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89768996</td>\n",
       "      <td>STR_89768996</td>\n",
       "      <td>ENSG00000140521.7</td>\n",
       "      <td>2.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.008189</td>\n",
       "      <td>...</td>\n",
       "      <td>0.192659</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.977813</td>\n",
       "      <td>0.531201</td>\n",
       "      <td>0.089374</td>\n",
       "      <td>89769033</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>POLG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16974</th>\n",
       "      <td>chr15</td>\n",
       "      <td>89796744</td>\n",
       "      <td>STR_89796744</td>\n",
       "      <td>ENSG00000140521.7</td>\n",
       "      <td>9.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.002077</td>\n",
       "      <td>0.001011</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.185725</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.855621</td>\n",
       "      <td>-0.003142</td>\n",
       "      <td>89796777</td>\n",
       "      <td>4</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>POLG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81429</th>\n",
       "      <td>chr5</td>\n",
       "      <td>146292064</td>\n",
       "      <td>STR_146292064</td>\n",
       "      <td>ENSG00000156475.14</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.813493</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.024325</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.825180</td>\n",
       "      <td>-0.020544</td>\n",
       "      <td>146292094</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>PPP2R2B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91427</th>\n",
       "      <td>chr9</td>\n",
       "      <td>101954984</td>\n",
       "      <td>STR_101954984</td>\n",
       "      <td>ENSG00000106799.8</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.411373</td>\n",
       "      <td>0.472398</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.120397</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.767681</td>\n",
       "      <td>0.707550</td>\n",
       "      <td>0.060907</td>\n",
       "      <td>101955036</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>TGFBR1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92254</th>\n",
       "      <td>chr9</td>\n",
       "      <td>2525935</td>\n",
       "      <td>STR_2525935</td>\n",
       "      <td>ENSG00000147852.11</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.817336</td>\n",
       "      <td>0.726808</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.163054</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.045917</td>\n",
       "      <td>0.081772</td>\n",
       "      <td>-0.166710</td>\n",
       "      <td>2526000</td>\n",
       "      <td>2</td>\n",
       "      <td>AC</td>\n",
       "      <td>VLDLR</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24 rows × 95 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom  str.start         str.id                gene  E.tissues  \\\n",
       "10377  chr12    7034264    STR_7034264  ENSG00000111676.10        8.0   \n",
       "14660  chr14   92557796   STR_92557796  ENSG00000066427.17       10.0   \n",
       "32869  chr19   13729541   STR_13729541  ENSG00000141837.14        1.0   \n",
       "7587   chr11  119087050  STR_119087050   ENSG00000110395.4        2.0   \n",
       "28256  chr18   72210730   STR_72210730  ENSG00000150656.10        1.0   \n",
       "16802  chr15   51620745   STR_51620745   ENSG00000137869.9        1.0   \n",
       "11100  chr12   51123498   STR_51123498   ENSG00000066084.8        1.0   \n",
       "33138  chr19   46265048   STR_46265048  ENSG00000104936.13        1.0   \n",
       "33140  chr19   46308960   STR_46308960  ENSG00000104936.13        1.0   \n",
       "91616   chr9   71685708   STR_71685708   ENSG00000165060.7        1.0   \n",
       "91617   chr9   71749839   STR_71749839   ENSG00000165060.7        1.0   \n",
       "17959  chr16   11940252   STR_11940252   ENSG00000103342.8        1.0   \n",
       "4341    chr1  154871688  STR_154871688  ENSG00000143603.14        1.0   \n",
       "85613   chr6   31428339   STR_31428339   ENSG00000204520.8        1.0   \n",
       "85716   chr6   29671619   STR_29671619   ENSG00000204655.7        1.0   \n",
       "61114  chr20   46210941   STR_46210941  ENSG00000124151.14        3.0   \n",
       "61115  chr20   46242650   STR_46242650  ENSG00000124151.14        1.0   \n",
       "43396  chr19   41233240   STR_41233240   ENSG00000105245.5        1.0   \n",
       "43403  chr19   41290402   STR_41290402   ENSG00000105245.5        1.0   \n",
       "16973  chr15   89768996   STR_89768996   ENSG00000140521.7        2.0   \n",
       "16974  chr15   89796744   STR_89796744   ENSG00000140521.7        9.0   \n",
       "81429   chr5  146292064  STR_146292064  ENSG00000156475.14        2.0   \n",
       "91427   chr9  101954984  STR_101954984   ENSG00000106799.8        1.0   \n",
       "92254   chr9    2525935    STR_2525935  ENSG00000147852.11        2.0   \n",
       "\n",
       "       ll.E.tissues  eSTR.Adipose.S  eSTR.Adipose.S.ll  qval.Adipose.S  \\\n",
       "10377           8.0             1.0                1.0        0.002085   \n",
       "14660           9.0             1.0                1.0        0.002077   \n",
       "32869           1.0             0.0                0.0        0.000000   \n",
       "7587            2.0             0.0                0.0        1.000000   \n",
       "28256           1.0             0.0                0.0        0.000000   \n",
       "16802           1.0             0.0                0.0        1.000000   \n",
       "11100           1.0             0.0                0.0        0.000000   \n",
       "33138           4.0             0.0                0.0        0.000000   \n",
       "33140           3.0             0.0                0.0        0.000000   \n",
       "91616           1.0             0.0                0.0        0.000000   \n",
       "91617           1.0             1.0                1.0        0.053631   \n",
       "17959           1.0             0.0                0.0        0.000000   \n",
       "4341            1.0             0.0                0.0        0.214518   \n",
       "85613           1.0             0.0                0.0        0.735983   \n",
       "85716           1.0             0.0                0.0        0.000000   \n",
       "61114           5.0             0.0                0.0        0.000000   \n",
       "61115           5.0             0.0                0.0        1.000000   \n",
       "43396           1.0             0.0                0.0        0.000000   \n",
       "43403           1.0             0.0                0.0        0.000000   \n",
       "16973          11.0             0.0                1.0        0.000000   \n",
       "16974          11.0             1.0                1.0        0.002077   \n",
       "81429           2.0             0.0                0.0        0.000000   \n",
       "91427           0.0             0.0                0.0        0.411373   \n",
       "92254           1.0             0.0                0.0        0.817336   \n",
       "\n",
       "       llqval.Adipose.S    ...      beta.Thyroid  eSTR.Blood  eSTR.Blood.ll  \\\n",
       "10377          0.001238    ...         -0.401075         0.0            0.0   \n",
       "14660          0.001046    ...          0.260183         1.0            0.0   \n",
       "32869          0.812012    ...          0.135937         0.0            0.0   \n",
       "7587           0.788202    ...         -0.064029         0.0            0.0   \n",
       "28256          0.342899    ...         -0.056906         0.0            0.0   \n",
       "16802          0.843324    ...         -0.025783         0.0            0.0   \n",
       "11100          0.619114    ...          0.360361         0.0            0.0   \n",
       "33138          0.776608    ...          0.290464         0.0            0.0   \n",
       "33140          0.260466    ...          0.064932         0.0            0.0   \n",
       "91616          0.706640    ...          0.088532         0.0            0.0   \n",
       "91617          0.034169    ...         -0.152007         0.0            0.0   \n",
       "17959          0.824577    ...         -0.092369         0.0            0.0   \n",
       "4341           0.204313    ...          0.002995         0.0            0.0   \n",
       "85613          0.482626    ...          0.162490         0.0            0.0   \n",
       "85716          0.459158    ...         -0.126769         0.0            0.0   \n",
       "61114          0.404204    ...         -0.103459         1.0            1.0   \n",
       "61115          0.248326    ...         -0.117563         0.0            1.0   \n",
       "43396          0.374403    ...          0.098156         0.0            0.0   \n",
       "43403          0.628057    ...         -0.020114         1.0            1.0   \n",
       "16973          0.008189    ...          0.192659         0.0            0.0   \n",
       "16974          0.001011    ...         -0.185725         0.0            0.0   \n",
       "81429          0.813493    ...         -0.024325         0.0            0.0   \n",
       "91427          0.472398    ...         -0.120397         0.0            0.0   \n",
       "92254          0.726808    ...         -0.163054         1.0            1.0   \n",
       "\n",
       "       qval.Blood  llqval.Blood  beta.Blood    str.end  period   motif  \\\n",
       "10377    1.000000      0.811180   -0.027276    7034322       2      AC   \n",
       "14660    0.098919      0.159107   -0.154160   92557821       4    AAAG   \n",
       "32869         NaN      0.851923   -0.005792   13729573       2      AG   \n",
       "7587          NaN      0.721164   -0.054110  119087097       4    AAAT   \n",
       "28256         NaN      0.851343   -0.005813   72210778       5   AAAAT   \n",
       "16802    1.000000      0.808066    0.028981   51620782       2      AC   \n",
       "11100    0.635046      0.329699    0.120274   51123531       1       A   \n",
       "33138         NaN      0.292570    0.122120   46265085       6  AGCTCC   \n",
       "33140         NaN      0.836187   -0.014291   46308980       4    AAAG   \n",
       "91616    0.736591      0.449685    0.105619   71685728       1       A   \n",
       "91617         NaN      0.767412   -0.041312   71749912       2      AG   \n",
       "17959    1.000000      0.746033   -0.050057   11940286       1       A   \n",
       "4341          NaN      0.857601    0.002003  154871733       4    AAAT   \n",
       "85613         NaN      0.371577    0.112800   31428384       4    AAAT   \n",
       "85716         NaN      0.595641   -0.077660   29671691       4    ATCC   \n",
       "61114    0.041836      0.005397   -0.216156   46210962       4    AATT   \n",
       "61115         NaN      0.018870   -0.212018   46242674       1       A   \n",
       "43396         NaN      0.683022    0.063493   41233274       5   AAAAT   \n",
       "43403    0.019162      0.003410   -0.222150   41290429       3     AAC   \n",
       "16973    0.977813      0.531201    0.089374   89769033       4    AAAT   \n",
       "16974         NaN      0.855621   -0.003142   89796777       4    AAAT   \n",
       "81429         NaN      0.825180   -0.020544  146292094       1       A   \n",
       "91427    0.767681      0.707550    0.060907  101955036       2      AC   \n",
       "92254    0.045917      0.081772   -0.166710    2526000       2      AC   \n",
       "\n",
       "       gene.name  \n",
       "10377       ATN1  \n",
       "14660      ATXN3  \n",
       "32869    CACNA1A  \n",
       "7587         CBL  \n",
       "28256      CNDP1  \n",
       "16802    CYP19A1  \n",
       "11100      DIP2B  \n",
       "33138       DMPK  \n",
       "33140       DMPK  \n",
       "91616        FXN  \n",
       "91617        FXN  \n",
       "17959      GSPT1  \n",
       "4341       KCNN3  \n",
       "85613       MICA  \n",
       "85716        MOG  \n",
       "61114      NCOA3  \n",
       "61115      NCOA3  \n",
       "43396      NUMBL  \n",
       "43403      NUMBL  \n",
       "16973       POLG  \n",
       "16974       POLG  \n",
       "81429    PPP2R2B  \n",
       "91427     TGFBR1  \n",
       "92254      VLDLR  \n",
       "\n",
       "[24 rows x 95 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#STR gwas hits\n",
    "print(len(STR_gwas_list), len(set(STR_gwas_list)))\n",
    "merged_t = merged_t.merge(names)\n",
    "Hits_gtex = merged_t.loc[merged_t['gene.name'].isin(STR_gwas_list)]\n",
    "Hits_gtex.loc[Hits_gtex['E.tissues']>=1].sort_values('gene.name')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
