{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "\"\"\"This code will crate a master table for each tissue\"\"\"\n",
    "#Set up\n",
    "COLORS = {\n",
    "    \"Artery-Aorta\":\"salmon\",\n",
    "    \"Artery-Tibial\": \"red\",\n",
    "    \"Adipose-Subcutaneous\": \"darkorange\",    \n",
    "    \"Adipose-Visceral\":\"orange\",\n",
    "    \"Brain-Caudate\":\"lemonchiffon\"   , \n",
    "    \"Brain-Cerebellum\":\"yellow\",\n",
    "    \"Cells-Transformedfibroblasts\": \"skyblue\",\n",
    "    \"Esophagus-Mucosa\": \"sienna\",\n",
    "    \"Esophagus-Muscularis\":\"burlywood\",\n",
    "    \"Heart-LeftVentricle\":\"darkviolet\",\n",
    "    \"Lung\": \"greenyellow\",\n",
    "    \"Muscle-Skeletal\": \"mediumslateblue\",\n",
    "    \"Nerve-Tibial\":\"gold\",\n",
    "    \"Skin-NotSunExposed\":\"blue\",\n",
    "    \"Skin-SunExposed\":\"cornflowerblue\",\n",
    "    \"Thyroid\":\"green\",\n",
    "    \"WholeBlood\": \"m\"\n",
    "        }\n",
    "\n",
    "SHORTEN = {\n",
    "    \"Artery-Aorta\":\"Artery A.\"     ,\n",
    "    \"Artery-Tibial\": \"Artery T.\",\n",
    "    \"Adipose-Subcutaneous\": \"Adipose S.\",    \n",
    "    \"Adipose-Visceral\":\"Adipose V.\",\n",
    "    \"Brain-Caudate\":\"Caudate\"   , \n",
    "    \"Brain-Cerebellum\":\"Cerebellum\",\n",
    "    \"Cells-Transformedfibroblasts\": \"Fibroblast\",\n",
    "    \"Esophagus-Mucosa\": \"E. Mucosa\",\n",
    "    \"Esophagus-Muscularis\":\"E Muscularis\",\n",
    "    \"Heart-LeftVentricle\":\"Ventricule\",\n",
    "    \"Lung\": \"Lung\",\n",
    "    \"Muscle-Skeletal\": \"Muscle\",\n",
    "    \"Nerve-Tibial\":\"Nerve\",\n",
    "    \"Skin-NotSunExposed\": \"Skin Unexposed\",\n",
    "    \"Skin-SunExposed\":\"Skin Leg\",\n",
    "    \"Thyroid\":\"Thyroid\",\n",
    "    \"WholeBlood\": \"Blood\"\n",
    "}\n",
    "\n",
    "import math\n",
    "import numpy as np\n",
    "import os\n",
    "import pandas as pd\n",
    "import random\n",
    "import sys\n",
    "from scipy import stats, integrate\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "sns.set(color_codes=True)\n",
    "\n",
    "tissu = sorted(COLORS.keys())\n",
    "CHR = [1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22]\n",
    "Gene_table = pd.read_csv('/storage/szfeupe/Runs/GTEx_estr/FEATURES/Genes_only_table', sep='\\t')\n",
    "Motifs = pd.read_csv('/storage/resources/dbase/human/hg19/hg19.hipstr_reference_withmotif.bed', sep='\\t', header=None)\n",
    "Motifs.columns = ['chrom','str.start','str.end','motif.len','motif']\n",
    "path='/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(514, 9)\n",
      "(193, 9)\n",
      "(261, 9)\n",
      "(594, 9)\n",
      "(30, 9)\n",
      "(96, 9)\n",
      "(446, 9)\n",
      "(491, 9)\n",
      "(439, 9)\n",
      "(221, 9)\n",
      "(360, 9)\n",
      "(510, 9)\n",
      "(704, 9)\n",
      "(271, 9)\n",
      "(468, 9)\n",
      "(650, 9)\n",
      "(386, 9)\n"
     ]
    }
   ],
   "source": [
    "A=[]\n",
    "B=[]\n",
    "All_eSTRs=[]\n",
    "for Tissue in tissu[0:]: \n",
    "    CAUSE=path+Tissue+'/HH/CAVIAR_analysis.table'\n",
    "    SCORE=path+Tissue+'/HH/strs_scorechr'\n",
    "    LR = path+Tissue+'/PQValues'\n",
    "    ANOV = path+Tissue+'/HH/Adusted_anova_wg.csv'\n",
    "#LR\n",
    "    lr = pd.read_csv(LR,sep='\\t')\n",
    "#Anova\n",
    "    anova = pd.read_csv(ANOV, sep='\\t')\n",
    "    anova = anova[['chrom','gene','str.start','anova_qval','estr_fdr','esnp_fdr','delta_bic','delta_aic','anova_pval']]\n",
    "    print(anova.loc[anova['anova_qval']<0.05].shape)\n",
    "    #merge 1\n",
    "    lr = pd.merge(lr, anova, on=['chrom','gene','str.start'], how='left')\n",
    "#Causal \n",
    "    causal = pd.read_csv(CAUSE, sep='\\t')\n",
    "    causal['chrom']=causal['CHROM']\n",
    "#Scores\n",
    "    score_items=[]\n",
    "    for i in CHR:\n",
    "        C=pd.read_csv(SCORE+str(i), sep='\\t', header=None)\n",
    "        score_items.append(C)\n",
    "    scores=score_items[0]\n",
    "    for i in range(1, len(score_items)):\n",
    "        scores=scores.append(score_items[i]) \n",
    "    scores.columns=['chrom', 'gene', 'str.id', 'caviar.score']\n",
    "    CA = scores; CA.index = CA['gene']        \n",
    "#Merge all in table\n",
    "    Table = pd.merge(lr, CA, on=['chrom','gene','str.id'], how='left')\n",
    "    Table.index = Table['gene']\n",
    "    Master = pd.merge(Table,causal, on=['chrom','gene'], how='left')\n",
    "    #print('With Caviar: ',Master.shape)\n",
    "#Gene names\n",
    "    genes=Gene_table.loc[Gene_table['gene'].isin(list(Master['gene']))]\n",
    "    genes=genes[['gene','chrom','gene.name']] \n",
    "    Master=pd.merge(Master, genes, on=['chrom','gene'])\n",
    "#motifs\n",
    "    Master=pd.merge(Master, Motifs [['chrom','str.start','motif','str.end']], on=['chrom','str.start'])\n",
    "#Additional columns\n",
    "    Master['best.str'] = np.where(Master['qvalue']<=1, 1,0)\n",
    "    Master['top.str.rank']=Master['str.rank']\n",
    "    Master['top.snp.score']=Master['top_snp_score']\n",
    "    Master['significant']= np.where(Master['qvalue']<=0.1, 1,0)\n",
    "    Master['llsignificant'] = Master['llsignif'].astype(int)\n",
    "    Master['top_str'] = Master['top_str'].astype(str)\n",
    "    Master['top.variant'] = np.where(Master['top.str.score']>Master['top.snp.score'], Master['top_str'], Master['top_snp'])\n",
    "    Master['best.str.start'] = Master['top_str'].astype(str).apply(lambda x: int(x.split('_')[1]) if 'STR' in x else '' )\n",
    "#Summarize and savw\n",
    "    Header = ['gene','chrom','str.start','str.id','best.str','best.str.start','top.variant','caviar.score','beta','beta.se','p.wald','qvalue','num.strs.in.top5','llqvalue','motif','top_str', 'top.str.score','top_snp', 'top.snp.score','top.str.rank','delta_bic','delta_aic','anova_pval', 'anova_qval','estr_fdr','esnp_fdr','llsignificant','significant', 'gene.name','str.end']\n",
    "    Master = Master[Header].sort_values(['chrom','gene','str.start'])\n",
    "    Master.to_csv('/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/'+Tissue+'/Master.table', sep='\\t',  index=None)\n",
    "    #print('With gene name and motif : ',Master.shape, Tissue)\n",
    "    #break\n",
    "##   RUN The following\n",
    "##   python ~/projects/GTEX_eSTRs/gtex-estrs/Scripts/STRs-vs-SNPs_effects/merge_causality_scores.py /storage/szfeupe/Runs/GTEx_estr/Analysis_by_Tissue/ WholeBlood,Cells-Transformedfibroblasts,Muscle-Skeletal,Lung,Artery-Tibial,Adipose-Subcutaneous,Esophagus-Mucosa causality > Merged_Best_causality.Table\n",
    "##   To merge N select best by tissue"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
