{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import vcf\n",
    "#\n",
    "\n",
    "#    \"Thyroid\": \"green\",\n",
    "SHORTEN = {\n",
    "    \"Artery-Aorta\":\"Artery.A\"     ,\n",
    "    \"Artery-Tibial\": \"Artery.T\",\n",
    "    \"Adipose-Subcutaneous\": \"Adipose.S\",    \n",
    "    \"Adipose-Visceral(Omentum)\":\"Adipose.V\",\n",
    "    \"Brain-Caudate(basalganglia)\":\"Caudate\"   , \n",
    "    \"Brain-Cerebellum\":\"Cerebellum\",\n",
    "    \"Cells-Transformedfibroblasts\": \"Fibroblast\",\n",
    "    \"Esophagus-Mucosa\": \"Mucosa\",\n",
    "    \"Esophagus-Muscularis\":\"Muscularis\",\n",
    "    \"Heart-LeftVentricle\":\"Heart\",\n",
    "    \"Lung\": \"Lung\",\n",
    "    \"Muscle-Skeletal\": \"Muscle\",\n",
    "    \"Nerve-Tibial\":\"Nerve\",\n",
    "    \"Skin-NotSunExposed(Suprapubic)\": \"SkinUnexposed\",\n",
    "    \"Skin-SunExposed(Lowerleg)\":\"SkinLeg\",\n",
    "    \"Thyroid\":\"Thyroid\",\n",
    "    \"WholeBlood\": \"Blood\",\n",
    "    \"permuted\":\"Permuted\",\n",
    "}\n",
    "\n",
    "GENE = '/storage/szfeupe/Runs/GTEx_estr/FEATURES/Genes_only_table'\n",
    "PATH= \"/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/\"\n",
    "TISSUES = sorted([item for item in list(SHORTEN.keys()) if item != \"permuted\"])\n",
    "motif='/storage/resources/dbase/human/hg19/hg19.hipstr_reference_withmotif.bed.gz'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "281318 \t (286612, 4) \t 3596 \t Adipose-Subcutaneous\n",
      "281223 \t (290076, 4) \t 1842 \t Adipose-Visceral(Omentum)\n",
      "277044 \t (292265, 4) \t 2401 \t Artery-Aorta\n",
      "275588 \t (292752, 4) \t 3462 \t Artery-Tibial\n",
      "278997 \t (299484, 4) \t 682 \t Brain-Caudate(basalganglia)\n",
      "277449 \t (301908, 4) \t 1555 \t Brain-Cerebellum\n",
      "267909 \t (302822, 4) \t 3096 \t Cells-Transformedfibroblasts\n",
      "281861 \t (304085, 4) \t 3468 \t Esophagus-Mucosa\n",
      "280638 \t (304403, 4) \t 2831 \t Esophagus-Muscularis\n",
      "271269 \t (304722, 4) \t 1733 \t Heart-LeftVentricle\n",
      "286473 \t (305270, 4) \t 2992 \t Lung\n",
      "271694 \t (305429, 4) \t 3373 \t Muscle-Skeletal\n",
      "284926 \t (305757, 4) \t 4364 \t Nerve-Tibial\n",
      "285850 \t (306652, 4) \t 2393 \t Skin-NotSunExposed(Suprapubic)\n",
      "285435 \t (306693, 4) \t 3836 \t Skin-SunExposed(Lowerleg)\n",
      "285470 \t (307187, 4) \t 4113 \t Thyroid\n",
      "261714 \t (307187, 4) \t 2587 \t WholeBlood\n",
      "Adipose-Subcutaneous\n",
      "Adipose-Visceral(Omentum)\n",
      "Artery-Aorta\n",
      "Artery-Tibial\n",
      "Brain-Caudate(basalganglia)\n",
      "Brain-Cerebellum\n",
      "Cells-Transformedfibroblasts\n",
      "Esophagus-Mucosa\n",
      "Esophagus-Muscularis\n",
      "Heart-LeftVentricle\n",
      "Lung\n",
      "Muscle-Skeletal\n",
      "Nerve-Tibial\n",
      "Skin-NotSunExposed(Suprapubic)\n",
      "Skin-SunExposed(Lowerleg)\n",
      "Thyroid\n",
      "WholeBlood\n",
      "Number of egenes...  (25619, 74)\n",
      "Nbr of eGenes in more than 1 tissues  (8152, 74)\n",
      "Nbr of eGenes in at least 5 tissues  (1910, 74)\n",
      "Nbr of eGenes in at least 10 tissues  (468, 74)\n",
      "(1620037, 5)\n",
      "(307187, 77) (10250, 77)\n"
     ]
    }
   ],
   "source": [
    "#Linear regression QVALUES\n",
    "#\n",
    "GENES = pd.read_csv(GENE, sep='\\t')\n",
    "col=['chrom','str.start','str.id','gene']\n",
    "SUMMARY = pd.read_csv(PATH +'WholeBlood/PQValues', sep = '\\t')[col]\n",
    "estrs = []\n",
    "SUM = {}\n",
    "Z = 0\n",
    "#Merging by tissues \n",
    "for T in TISSUES:\n",
    "    table = pd.read_csv(PATH +T+'/PQValues', sep = '\\t')\n",
    "    #table['eSTR.'+SHORTEN[T]+'.ll'] = np.where(table['llqvalue']<=0.1, 1, 0)  #10%fdr at locus level\n",
    "    table['eSTR.'+SHORTEN[T]] = np.where(table['qvalue']<=0.1, 1, 0)\n",
    "    table['qval.'+SHORTEN[T]] = table['qvalue'].astype(float)\n",
    "    table['llqval.'+SHORTEN[T]] = table['llqvalue'].astype(float)\n",
    "    table['beta.'+SHORTEN[T]] = table['beta'].astype(float)\n",
    "    table = table[col + ['eSTR.'+SHORTEN[T], 'qval.'+SHORTEN[T],'llqval.'+SHORTEN[T],'beta.'+SHORTEN[T]]]\n",
    "    table['str.start'].astype(int)\n",
    "    table['chrom'].astype(str)\n",
    "    #estrs.append('eSTR.'+SHORTEN[T])\n",
    "    SUM[T]=table\n",
    "    table['eSTR.'+SHORTEN[T]] = table['eSTR.'+SHORTEN[T]].astype(int)\n",
    "    SUMMARY = pd.merge(SUMMARY,table[col], on=col, how='outer').drop_duplicates().reset_index(drop=True)\n",
    "    print(table.shape[0],'\\t',SUMMARY.shape,'\\t',table.loc[table['eSTR.'+SHORTEN[T]]==1].shape[0],'\\t', T)\n",
    "    Z=Z+table.shape[0]\n",
    "SUMMARY = SUMMARY.sort_values(['chrom','gene', 'str.start']).reset_index(drop=True)\n",
    "SUMMARY['E.tissues']=[0]*SUMMARY.shape[0]\n",
    "SUMMARY['ll.E.tissues']=[0]*SUMMARY.shape[0]\n",
    "for T in TISSUES :\n",
    "    print (T)\n",
    "    SUMMARY = pd.merge(SUMMARY.fillna(0), SUM[T], how='left',on=col)\n",
    "    SUMMARY['E.tissues'] = SUMMARY['E.tissues'] + SUMMARY['eSTR.'+SHORTEN[T]].fillna(0).astype(int)\n",
    "\n",
    "\n",
    "print('Number of egenes... ',SUMMARY.loc[SUMMARY['E.tissues']>=1].shape)\n",
    "print('Nbr of eGenes in more than 1 tissues ',SUMMARY.loc[SUMMARY['E.tissues']>=2].shape)\n",
    "print('Nbr of eGenes in at least 5 tissues ',SUMMARY.loc[SUMMARY['E.tissues']>=5].shape)\n",
    "print('Nbr of eGenes in at least 10 tissues ',SUMMARY.loc[SUMMARY['E.tissues']>=10].shape)\n",
    "\n",
    "add_info = pd.read_csv(motif, sep='\\t', header=None)\n",
    "add_info.columns=['chrom','str.start','str.end','period','motif']\n",
    "print(add_info.shape)\n",
    "SUMMARY = pd.merge(SUMMARY, add_info, on=['chrom','str.start'], how='inner')\n",
    "e_homopolymers = SUMMARY.loc[(SUMMARY['E.tissues']>=1) & (SUMMARY['period']==1)]\n",
    "print(SUMMARY.shape, e_homopolymers.shape)\n",
    "\n",
    "SUMMARY.to_csv('/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/LR_SummaryTest_Table.tsv', sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "31.82013349467192"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#SUMMARY.loc[SUMMARY['E.tissues']>=1]\n",
    "8152*100/25619"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unit.size</th>\n",
       "      <th>all.tissues</th>\n",
       "      <th>Adipose.S</th>\n",
       "      <th>Adipose.V</th>\n",
       "      <th>Artery.A</th>\n",
       "      <th>Artery.T</th>\n",
       "      <th>Caudate</th>\n",
       "      <th>Cerebellum</th>\n",
       "      <th>Fibroblast</th>\n",
       "      <th>Mucosa</th>\n",
       "      <th>Muscularis</th>\n",
       "      <th>Heart</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Nerve</th>\n",
       "      <th>SkinUnexposed</th>\n",
       "      <th>SkinLeg</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Blood</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>10250</td>\n",
       "      <td>1357</td>\n",
       "      <td>713</td>\n",
       "      <td>971</td>\n",
       "      <td>1371</td>\n",
       "      <td>269</td>\n",
       "      <td>588</td>\n",
       "      <td>1246</td>\n",
       "      <td>1363</td>\n",
       "      <td>1149</td>\n",
       "      <td>685</td>\n",
       "      <td>1199</td>\n",
       "      <td>1318</td>\n",
       "      <td>1690</td>\n",
       "      <td>956</td>\n",
       "      <td>1491</td>\n",
       "      <td>1619</td>\n",
       "      <td>1033</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>7037</td>\n",
       "      <td>959</td>\n",
       "      <td>493</td>\n",
       "      <td>625</td>\n",
       "      <td>902</td>\n",
       "      <td>179</td>\n",
       "      <td>441</td>\n",
       "      <td>807</td>\n",
       "      <td>900</td>\n",
       "      <td>728</td>\n",
       "      <td>458</td>\n",
       "      <td>801</td>\n",
       "      <td>861</td>\n",
       "      <td>1142</td>\n",
       "      <td>632</td>\n",
       "      <td>1014</td>\n",
       "      <td>1099</td>\n",
       "      <td>634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>1609</td>\n",
       "      <td>258</td>\n",
       "      <td>132</td>\n",
       "      <td>149</td>\n",
       "      <td>238</td>\n",
       "      <td>49</td>\n",
       "      <td>114</td>\n",
       "      <td>209</td>\n",
       "      <td>240</td>\n",
       "      <td>193</td>\n",
       "      <td>114</td>\n",
       "      <td>217</td>\n",
       "      <td>225</td>\n",
       "      <td>316</td>\n",
       "      <td>161</td>\n",
       "      <td>293</td>\n",
       "      <td>280</td>\n",
       "      <td>191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>4690</td>\n",
       "      <td>679</td>\n",
       "      <td>351</td>\n",
       "      <td>429</td>\n",
       "      <td>637</td>\n",
       "      <td>118</td>\n",
       "      <td>280</td>\n",
       "      <td>553</td>\n",
       "      <td>676</td>\n",
       "      <td>526</td>\n",
       "      <td>314</td>\n",
       "      <td>519</td>\n",
       "      <td>671</td>\n",
       "      <td>842</td>\n",
       "      <td>438</td>\n",
       "      <td>731</td>\n",
       "      <td>781</td>\n",
       "      <td>485</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>1331</td>\n",
       "      <td>228</td>\n",
       "      <td>106</td>\n",
       "      <td>152</td>\n",
       "      <td>206</td>\n",
       "      <td>48</td>\n",
       "      <td>90</td>\n",
       "      <td>183</td>\n",
       "      <td>179</td>\n",
       "      <td>151</td>\n",
       "      <td>117</td>\n",
       "      <td>179</td>\n",
       "      <td>183</td>\n",
       "      <td>259</td>\n",
       "      <td>138</td>\n",
       "      <td>204</td>\n",
       "      <td>214</td>\n",
       "      <td>156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>702</td>\n",
       "      <td>115</td>\n",
       "      <td>47</td>\n",
       "      <td>75</td>\n",
       "      <td>108</td>\n",
       "      <td>19</td>\n",
       "      <td>42</td>\n",
       "      <td>98</td>\n",
       "      <td>110</td>\n",
       "      <td>84</td>\n",
       "      <td>45</td>\n",
       "      <td>77</td>\n",
       "      <td>115</td>\n",
       "      <td>115</td>\n",
       "      <td>68</td>\n",
       "      <td>103</td>\n",
       "      <td>120</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unit.size  all.tissues  Adipose.S  Adipose.V  Artery.A  Artery.T  Caudate  \\\n",
       "1          1        10250       1357        713       971      1371      269   \n",
       "2          2         7037        959        493       625       902      179   \n",
       "3          3         1609        258        132       149       238       49   \n",
       "4          4         4690        679        351       429       637      118   \n",
       "5          5         1331        228        106       152       206       48   \n",
       "6          6          702        115         47        75       108       19   \n",
       "\n",
       "   Cerebellum  Fibroblast  Mucosa  Muscularis  Heart  Lung  Muscle  Nerve  \\\n",
       "1         588        1246    1363        1149    685  1199    1318   1690   \n",
       "2         441         807     900         728    458   801     861   1142   \n",
       "3         114         209     240         193    114   217     225    316   \n",
       "4         280         553     676         526    314   519     671    842   \n",
       "5          90         183     179         151    117   179     183    259   \n",
       "6          42          98     110          84     45    77     115    115   \n",
       "\n",
       "   SkinUnexposed  SkinLeg  Thyroid  Blood  \n",
       "1            956     1491     1619   1033  \n",
       "2            632     1014     1099    634  \n",
       "3            161      293      280    191  \n",
       "4            438      731      781    485  \n",
       "5            138      204      214    156  \n",
       "6             68      103      120     88  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Number of eSTRs by STR unit size\n",
    "sumtable = {}\n",
    "sumtable['Unit.size'] = [x for x in range(1,7,1)]\n",
    "sumtable['all.tissues'] = [ SUMMARY.loc[(SUMMARY['E.tissues']>=1) & (SUMMARY['period']==i)].shape[0] for i in range(1,7,1)]\n",
    "\n",
    "for T in TISSUES:\n",
    "    sumtable[SHORTEN[T]] = [ SUMMARY.loc[(SUMMARY['eSTR.'+SHORTEN[T]]>=1) & (SUMMARY['period']==i)].shape[0] for i in range(1,7,1)]\n",
    "    \n",
    "estr_by_unit = pd.DataFrame(sumtable)[['Unit.size', 'all.tissues']+ [SHORTEN[T] for T in TISSUES]]\n",
    "estr_by_unit.index = list(estr_by_unit['Unit.size'])\n",
    "estr_by_unit['Unit.size']\n",
    "\n",
    "estr_by_unit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total # eSTRs 25619 \n",
      "Unique to tissues 17467 \n",
      "Shared eSTRs 8152\n",
      "Total # of Unique to tissue eSTRs 17467 \n",
      "Number of homopolymers  7013 \n",
      " Shared homopolymers eSTRs 3237\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(10250, 77)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "E = SUMMARY.loc[SUMMARY['E.tissues']>=1]\n",
    "U = SUMMARY.loc[SUMMARY['E.tissues']==1]\n",
    "S = SUMMARY.loc[SUMMARY['E.tissues']>1]\n",
    "E1 = E.loc[E['period']==1]\n",
    "print('Total # eSTRs',E.shape[0],'\\nUnique to tissues', U.shape[0],'\\nShared eSTRs', S.shape[0])\n",
    "H = U.loc[U['period']==1]; S1 = S.loc[S['period']==1]\n",
    "print('Total # of Unique to tissue eSTRs',U.shape[0],'\\nNumber of homopolymers ', H.shape[0],'\\n Shared homopolymers eSTRs', S1.shape[0])\n",
    "\n",
    "E1.shape"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
