{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "# Libraries\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import vcf\n",
    "#\n",
    "\n",
    "#    \"Thyroid\": \"green\",\n",
    "SHORTEN = {\n",
    "    \"Artery-Aorta\":\"Artery.A\"     ,\n",
    "    \"Artery-Tibial\": \"Artery.T\",\n",
    "    \"Adipose-Subcutaneous\": \"Adipose.S\",    \n",
    "    \"Adipose-Visceral(Omentum)\":\"Adipose.V\",\n",
    "    \"Brain-Caudate(basalganglia)\":\"Caudate\"   , \n",
    "    \"Brain-Cerebellum\":\"Cerebellum\",\n",
    "    \"Cells-Transformedfibroblasts\": \"Fibroblast\",\n",
    "    \"Esophagus-Mucosa\": \"Mucosa\",\n",
    "    \"Esophagus-Muscularis\":\"Muscularis\",\n",
    "    \"Heart-LeftVentricle\":\"Heart\",\n",
    "    \"Lung\": \"Lung\",\n",
    "    \"Muscle-Skeletal\": \"Muscle\",\n",
    "    \"Nerve-Tibial\":\"Nerve\",\n",
    "    \"Skin-NotSunExposed(Suprapubic)\": \"SkinUnexposed\",\n",
    "    \"Skin-SunExposed(Lowerleg)\":\"SkinLeg\",\n",
    "    \"Thyroid\":\"Thyroid\",\n",
    "    \"WholeBlood\": \"Blood\",\n",
    "    \"permuted\":\"Permuted\",\n",
    "}\n",
    "\n",
    "GENE = '/storage/szfeupe/Runs/GTEx_estr/FEATURES/Genes_only_table'\n",
    "PATH= \"/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/\"\n",
    "TISSUES = sorted([item for item in list(SHORTEN.keys()) if item != \"permuted\"])\n",
    "motif='/storage/resources/dbase/human/hg19/hg19.hipstr_reference_withmotif.bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of egenes...  (9446, 91)\n",
      "Number of egenes by locus...  (19566, 91)\n",
      "\n",
      "Nbr of eGenes in more than 1 tissues  (3663, 91)\n",
      "Nbr of eGenes by locus in more than 1 tissues  (8768, 91)\n",
      "\n",
      "egenes in at least 10 tissues  (319, 91)\n",
      "egenes in at least 10 tissues (locus) (1090, 91)\n",
      "(1620037, 5)\n",
      "(98671, 94) (3416, 94)\n"
     ]
    }
   ],
   "source": [
    "#Linear regression QVALUES\n",
    "#\n",
    "GENES = pd.read_csv(GENE, sep='\\t')\n",
    "col=['chrom','str.start','str.id','gene']\n",
    "SUMMARY = pd.read_csv(PATH +'WholeBlood/PQValues', sep = '\\t')[col]\n",
    "estrs = []\n",
    "SUM = {}\n",
    "#Merging by tissues \n",
    "for T in TISSUES:\n",
    "    table = pd.read_csv(PATH +T+'/PQValues', sep = '\\t')\n",
    "    table['eSTR.'+SHORTEN[T]+'.ll'] = np.where(table['llqvalue']<=0.1, 1, 0)  #10%fdr at locus level\n",
    "    table['eSTR.'+SHORTEN[T]] = np.where(table['qvalue']<=0.1, 1, 0)\n",
    "    table['qval.'+SHORTEN[T]] = table['qvalue'].astype(float)\n",
    "    table['llqval.'+SHORTEN[T]] = table['llqvalue'].astype(float)\n",
    "    table['beta.'+SHORTEN[T]] = table['beta'].astype(float)\n",
    "    table = table[col + ['eSTR.'+SHORTEN[T], 'eSTR.'+SHORTEN[T]+'.ll', 'qval.'+SHORTEN[T],'llqval.'+SHORTEN[T],'beta.'+SHORTEN[T]]]\n",
    "    table['str.start'].astype(int)\n",
    "    table['chrom'].astype(str)\n",
    "    #estrs.append('eSTR.'+SHORTEN[T])\n",
    "    SUM[T]=table\n",
    "    table['eSTR.'+SHORTEN[T]] = table['eSTR.'+SHORTEN[T]].astype(int)\n",
    "    SUMMARY = pd.merge(SUMMARY,table[col], on=col, how='outer').drop_duplicates().reset_index(drop=True)\n",
    "    #print(table.shape[0],'\\t',SUMMARY.shape,'\\t',table.loc[table['eSTR.'+SHORTEN[T]]==1].shape[0],'\\t', T)\n",
    "\n",
    "SUMMARY = SUMMARY.sort_values(['chrom','gene', 'str.start']).reset_index(drop=True)\n",
    "SUMMARY['E.tissues']=[0]*SUMMARY.shape[0]\n",
    "SUMMARY['ll.E.tissues']=[0]*SUMMARY.shape[0]\n",
    "for T in TISSUES :\n",
    "    SUMMARY = pd.merge(SUMMARY.fillna(0), SUM[T], how='left',on=col)\n",
    "    SUMMARY['E.tissues'] = SUMMARY['E.tissues'].fillna(0) + SUMMARY['eSTR.'+SHORTEN[T]]\n",
    "    SUMMARY['ll.E.tissues'] = SUMMARY['ll.E.tissues'].fillna(0) + SUMMARY['eSTR.'+SHORTEN[T]+'.ll']\n",
    "\n",
    "\n",
    "print('Number of egenes... ',SUMMARY.loc[SUMMARY['E.tissues']>=1].shape)\n",
    "print('Number of egenes by locus... ',SUMMARY.loc[SUMMARY['ll.E.tissues']>=1].shape)\n",
    "print('\\nNbr of eGenes in more than 1 tissues ',SUMMARY.loc[SUMMARY['E.tissues']>=2].shape)\n",
    "print('Nbr of eGenes by locus in more than 1 tissues ',SUMMARY.loc[SUMMARY['ll.E.tissues']>=2].shape)\n",
    "print('\\negenes in at least 10 tissues ',SUMMARY.loc[SUMMARY['E.tissues']>=10].shape)\n",
    "print('egenes in at least 10 tissues (locus)',SUMMARY.loc[SUMMARY['ll.E.tissues']>=10].shape)\n",
    "\n",
    "add_info = pd.read_csv(motif, sep='\\t', header=None)\n",
    "add_info.columns=['chrom','str.start','str.end','period','motif']\n",
    "print(add_info.shape)\n",
    "\n",
    "SUMMARY = pd.merge(SUMMARY, add_info, on=['chrom','str.start'], how='inner')\n",
    "\n",
    "e_homopolymers = SUMMARY.loc[(SUMMARY['E.tissues']>=1) & (SUMMARY['period']==1)]\n",
    "\n",
    "print(SUMMARY.shape, e_homopolymers.shape)\n",
    "\n",
    "SUMMARY.to_csv('/storage/szfeupe/Runs/650GTEx_estr/Analysis_by_Tissue/LR_SummaryTest_Table.tsv', sep='\\t', index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unit.size</th>\n",
       "      <th>all.tissues</th>\n",
       "      <th>Adipose.S</th>\n",
       "      <th>Adipose.V</th>\n",
       "      <th>Artery.A</th>\n",
       "      <th>Artery.T</th>\n",
       "      <th>Caudate</th>\n",
       "      <th>Cerebellum</th>\n",
       "      <th>Fibroblast</th>\n",
       "      <th>Mucosa</th>\n",
       "      <th>Muscularis</th>\n",
       "      <th>Heart</th>\n",
       "      <th>Lung</th>\n",
       "      <th>Muscle</th>\n",
       "      <th>Nerve</th>\n",
       "      <th>SkinUnexposed</th>\n",
       "      <th>SkinLeg</th>\n",
       "      <th>Thyroid</th>\n",
       "      <th>Blood</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>3416</td>\n",
       "      <td>647</td>\n",
       "      <td>328</td>\n",
       "      <td>437</td>\n",
       "      <td>610</td>\n",
       "      <td>97</td>\n",
       "      <td>217</td>\n",
       "      <td>532</td>\n",
       "      <td>571</td>\n",
       "      <td>501</td>\n",
       "      <td>302</td>\n",
       "      <td>530</td>\n",
       "      <td>601</td>\n",
       "      <td>788</td>\n",
       "      <td>405</td>\n",
       "      <td>712</td>\n",
       "      <td>733</td>\n",
       "      <td>476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>2907</td>\n",
       "      <td>531</td>\n",
       "      <td>275</td>\n",
       "      <td>355</td>\n",
       "      <td>521</td>\n",
       "      <td>73</td>\n",
       "      <td>214</td>\n",
       "      <td>455</td>\n",
       "      <td>506</td>\n",
       "      <td>438</td>\n",
       "      <td>230</td>\n",
       "      <td>425</td>\n",
       "      <td>479</td>\n",
       "      <td>682</td>\n",
       "      <td>343</td>\n",
       "      <td>583</td>\n",
       "      <td>621</td>\n",
       "      <td>344</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>594</td>\n",
       "      <td>135</td>\n",
       "      <td>60</td>\n",
       "      <td>71</td>\n",
       "      <td>105</td>\n",
       "      <td>19</td>\n",
       "      <td>36</td>\n",
       "      <td>117</td>\n",
       "      <td>120</td>\n",
       "      <td>98</td>\n",
       "      <td>54</td>\n",
       "      <td>96</td>\n",
       "      <td>107</td>\n",
       "      <td>145</td>\n",
       "      <td>80</td>\n",
       "      <td>140</td>\n",
       "      <td>135</td>\n",
       "      <td>86</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>1767</td>\n",
       "      <td>324</td>\n",
       "      <td>157</td>\n",
       "      <td>175</td>\n",
       "      <td>318</td>\n",
       "      <td>42</td>\n",
       "      <td>118</td>\n",
       "      <td>283</td>\n",
       "      <td>329</td>\n",
       "      <td>228</td>\n",
       "      <td>158</td>\n",
       "      <td>242</td>\n",
       "      <td>321</td>\n",
       "      <td>402</td>\n",
       "      <td>184</td>\n",
       "      <td>369</td>\n",
       "      <td>379</td>\n",
       "      <td>220</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>508</td>\n",
       "      <td>107</td>\n",
       "      <td>46</td>\n",
       "      <td>67</td>\n",
       "      <td>93</td>\n",
       "      <td>16</td>\n",
       "      <td>47</td>\n",
       "      <td>88</td>\n",
       "      <td>95</td>\n",
       "      <td>72</td>\n",
       "      <td>45</td>\n",
       "      <td>75</td>\n",
       "      <td>106</td>\n",
       "      <td>128</td>\n",
       "      <td>68</td>\n",
       "      <td>104</td>\n",
       "      <td>114</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>254</td>\n",
       "      <td>45</td>\n",
       "      <td>22</td>\n",
       "      <td>34</td>\n",
       "      <td>42</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "      <td>43</td>\n",
       "      <td>59</td>\n",
       "      <td>37</td>\n",
       "      <td>20</td>\n",
       "      <td>44</td>\n",
       "      <td>51</td>\n",
       "      <td>46</td>\n",
       "      <td>21</td>\n",
       "      <td>47</td>\n",
       "      <td>51</td>\n",
       "      <td>39</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Unit.size  all.tissues  Adipose.S  Adipose.V  Artery.A  Artery.T  Caudate  \\\n",
       "1          1         3416        647        328       437       610       97   \n",
       "2          2         2907        531        275       355       521       73   \n",
       "3          3          594        135         60        71       105       19   \n",
       "4          4         1767        324        157       175       318       42   \n",
       "5          5          508        107         46        67        93       16   \n",
       "6          6          254         45         22        34        42        4   \n",
       "\n",
       "   Cerebellum  Fibroblast  Mucosa  Muscularis  Heart  Lung  Muscle  Nerve  \\\n",
       "1         217         532     571         501    302   530     601    788   \n",
       "2         214         455     506         438    230   425     479    682   \n",
       "3          36         117     120          98     54    96     107    145   \n",
       "4         118         283     329         228    158   242     321    402   \n",
       "5          47          88      95          72     45    75     106    128   \n",
       "6          11          43      59          37     20    44      51     46   \n",
       "\n",
       "   SkinUnexposed  SkinLeg  Thyroid  Blood  \n",
       "1            405      712      733    476  \n",
       "2            343      583      621    344  \n",
       "3             80      140      135     86  \n",
       "4            184      369      379    220  \n",
       "5             68      104      114     78  \n",
       "6             21       47       51     39  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Number of eSTRs by STR unit size\n",
    "sumtable = {}\n",
    "sumtable['Unit.size'] = [x for x in range(1,7,1)]\n",
    "sumtable['all.tissues'] = [ SUMMARY.loc[(SUMMARY['E.tissues']>=1) & (SUMMARY['period']==i)].shape[0] for i in range(1,7,1)]\n",
    "\n",
    "for T in TISSUES:\n",
    "    sumtable[SHORTEN[T]] = [ SUMMARY.loc[(SUMMARY['eSTR.'+SHORTEN[T]]>=1) & (SUMMARY['period']==i)].shape[0] for i in range(1,7,1)]\n",
    "    \n",
    "estr_by_unit = pd.DataFrame(sumtable)[['Unit.size', 'all.tissues']+ [SHORTEN[T] for T in TISSUES]]\n",
    "estr_by_unit.index = list(estr_by_unit['Unit.size'])\n",
    "estr_by_unit['Unit.size']\n",
    "\n",
    "estr_by_unit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Total # eSTRs 9446 \n",
      "Unique to tissues 5783 \n",
      "Shared eSTRs 3663\n",
      "Total # of Unique to tissue eSTRs 5783 \n",
      "Number of homopolymers  2089 \n",
      " Shared homopolymers eSTRs 1327\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(3416, 94)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "E = SUMMARY.loc[SUMMARY['E.tissues']>=1]\n",
    "U = SUMMARY.loc[SUMMARY['E.tissues']==1]\n",
    "S = SUMMARY.loc[SUMMARY['E.tissues']>1]\n",
    "E1 = E.loc[E['period']==1]\n",
    "print('Total # eSTRs',E.shape[0],'\\nUnique to tissues', U.shape[0],'\\nShared eSTRs', S.shape[0])\n",
    "H = U.loc[U['period']==1]; S1 = S.loc[S['period']==1]\n",
    "print('Total # of Unique to tissue eSTRs',U.shape[0],'\\nNumber of homopolymers ', H.shape[0],'\\n Shared homopolymers eSTRs', S1.shape[0])\n",
    "\n",
    "E1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "36.16345543087021"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "100*3416/9446"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "##This code summarises the Anova results and prepare file for anovapvalue qqplot \n",
    "##\n",
    "import matplotlib.pyplot as plt\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import scipy.stats as ss\n",
    "\n",
    "Tissue=\"WholeBlood\"\n",
    "##delta_bic = snp_bic - snpstr_bic        ##delta_aic = snp_aic - snpstr_aic \n",
    "##              the lower tht aic or bic the closer the data is to the model\n",
    "print('Summary for '+Tissue+' ...\\n')\n",
    "\n",
    "#FDR \n",
    "pval = pd.read_csv(\"/storage/szfeupe/Runs/GTEx_estr/Analysis_by_Tissue/\"+Tissue+\"/PQValues.txt\", sep='\\t')\n",
    "pval = pval.loc[pval['qvalue']<=0.05]  ##eSTRs at 10% FDR\n",
    "print(pval.shape[0],' eSTRs at 10% FDR')\n",
    "\n",
    "#Anova output \n",
    "hh = pd.read_csv(\"/storage/szfeupe/Runs/GTEx_estr/Analysis_by_Tissue/\"+Tissue+\"/HH/Anova_wg_old.csv\", sep=',')\n",
    "hh1 = hh.loc[hh['gene'].isin(pval['gene'])] \n",
    "print(hh1.shape[0], ' eSTRs and their anova p-values')\n",
    "\n",
    "hh1['st1']=[0]*len(hh1)\n",
    "hh1['st2']=[0]*len(hh1)\n",
    "hh1['st1'][hh1['delta_aic']>0] = 1\n",
    "hh1['st2'][hh1['delta_bic']>0] = 2\n",
    "hh1['strprefer']=hh1['st1']+hh1['st2']\n",
    "\n",
    "\n",
    "hh1.to_csv(\"/storage/szfeupe/Runs/GTEx_estr/Analysis_by_Tissue/\"+Tissue+\"/HH/ESTRs_Anova\", sep='\\t', index=False)\n",
    "\n",
    "hh1"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.4.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
